

<div class='row' id="top">





<section class="col-md-4">
    <aside class="well">
        <a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
    </aside>
    <aside class="well">
        <h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandManipulationTools">Variant Evaluation and Manipulation Tools</a>
							<div id="VariantEvaluationandManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ValidationUtilities">Validation Utilities</a>
							<div id="ValidationUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_GenotypeAndValidate.html">GenotypeAndValidate</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#RODCodecs">ROD Codecs</a>
							<div id="RODCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Uncategorized">Uncategorized</a>
							<div id="Uncategorized"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_AnnotatePopulationAFWalker.html">AnnotatePopulationAFWalker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_AssignSomaticStatus.html">AssignSomaticStatus</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantAnnotations">Variant Annotations</a>
							<div id="VariantAnnotations"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
    </aside>


</section>

<div class="col-md-8">      <h1><small>Showing docs for version 3.5-0 </small></h1>     <hr />

    <h1>GenotypeAndValidate</h1>

    <p class="lead">Genotype and validate a dataset and the calls of another dataset using the Unified Genotyper</p>
        <h3>Category
            <small> Validation Utilities</small>
        </h3>
		<h3>Traversal
			<small>LocusWalker</small>
		</h3>
		<h3>PartitionBy
			<small>LOCUS</small>
		</h3>

	<hr>
    <h2>Overview</h2>
    

  <h4>Note that this is an old tool that makes use of the UnifiedGenotyper, which has since been
  deprecated in favor of the HaplotypeCaller.</h4>
  <p>
     Genotype and Validate is a tool to evaluate the quality of a dataset for calling SNPs
     and Indels given a secondary (validation) data source. The data sources are BAM or VCF
     files. You can use them interchangeably (i.e. a BAM to validate calls in a VCF or a VCF
     to validate calls on a BAM).
  </p>

  <p>
     The simplest scenario is when you have a VCF of hand annotated SNPs and Indels, and you
     want to know how well a particular technology performs calling these snps. With a
     dataset (BAM file) generated by the technology in test, and the hand annotated VCF, you
     can run GenotypeAndValidate to asses the accuracy of the calls with the new technology's
     dataset.
  </p>

  <p>
     Another option is to validate the calls on a VCF file, using a deep coverage BAM file
     that you trust the calls on. The GenotypeAndValidate walker will make calls using the
     reads in the BAM file and take them as truth, then compare to the calls in the VCF file
     and produce a truth table.
  </p>


 <h3>Input</h3>
  <p>
      A BAM file to make calls on and a VCF file to use as truth validation dataset.

      You also have the option to invert the roles of the files using the command line options listed below.
  </p>

 <h3>Output</h3>
  <p>
      GenotypeAndValidate has two outputs. The truth table and the optional VCF file. The truth table is a
      2x2 table correlating what was called in the dataset with the truth of the call (whether it's a true
      positive or a false positive). The table should look like this:
  </p>
 <center>
 <table id="description-table">
     <tr>
         <th></th>
         <th>ALT</th>
         <th>REF</th>
         <th>Predictive Value</th>
     </tr>
     <tr>
         <td><b>called alt</b></td>
         <td>True Positive (TP)</td>
         <td>False Positive (FP)</td>
         <td>Positive PV</td>
     </tr>
     <tr>
         <td><b>called ref</b></td>
         <td>False Negative (FN)</td>
         <td>True Negative (TN)</td>
         <td>Negative PV</td>
     </tr>
 </table>
 </center>

  <p>
      The <b>positive predictive value (PPV)</b> is the proportion of subjects with positive test results
      who are correctly diagnosed.
  </p>
  <p>
      The <b>negative predictive value (NPV)</b> is the proportion of subjects with a negative test result
      who are correctly diagnosed.
  </p>
  <p>
      The VCF file will contain only the variants that were called or not called, excluding the ones that
      were uncovered or didn't pass the filters. This file is useful if you are trying to compare
      the PPV and NPV of two different technologies on the exact same sites (so you can compare apples to
      apples).
  </p>

  <p>
      Here is an example of an annotated VCF file (info field clipped for clarity)

 <pre>
 #CHROM  POS ID  REF ALT QUAL    FILTER  INFO    FORMAT  NA12878
 1   20568807    .   C   T   0    HapMapHet        AC=1;AF=0.50;AN=2;DP=0;GV=T  GT  0/1
 1   22359922    .   T   C   282  WG-CG-HiSeq      AC=2;AF=0.50;GV=T;AN=4;DP=42 GT:AD:DP:GL:GQ  1/0 ./. 0/1:20,22:39:-72.79,-11.75,-67.94:99    ./.
 13  102391461   .   G   A   341  Indel;SnpCluster AC=1;GV=F;AF=0.50;AN=2;DP=45 GT:AD:DP:GL:GQ  ./. ./. 0/1:32,13:45:-50.99,-13.56,-112.17:99   ./.
 1   175516757   .   C   G   655  SnpCluster,WG    AC=1;AF=0.50;AN=2;GV=F;DP=74 GT:AD:DP:GL:GQ  ./. ./. 0/1:52,22:67:-89.02,-20.20,-191.27:99   ./.
 </pre>

  </p>

  <h3>Additional Details</h3>
  <ul>
      <li>
          You should always use -L on your VCF track, so that the GATK only looks at the sites on the VCF file.
          This speeds up the process a lot.
      </li>
      <li>
          The total number of visited bases may be greater than the number of variants in the original
          VCF file because of extended indels, as they trigger one call per new insertion or deletion.
          (i.e. ACTG/- will count as 4 genotyper calls, but it's only one line in the VCF).
      </li>
  </ul>

 <h3>Usage examples</h3>
 <h4>Genotypes BAM file from new technology using the VCF as a truth dataset</h4>
 <pre>
  java
      -jar GenomeAnalysisTK.jar \
      -T  GenotypeAndValidate \
      -R reference.fasta \
      -I myNewTechReads.bam \
      -alleles handAnnotatedVCF.vcf \
      -L handAnnotatedVCF.vcf \
      -o output.vcf
 </pre>

 <h4>Genotypes BAM file from new technology a BAM file as the truth dataset</h4>
 <pre>
  java
      -jar GenomeAnalysisTK.jar \
      -T  GenotypeAndValidate \
      -R reference.fasta \
      -I myTruthDataset.bam \
      -alleles callsToValidate.vcf \
      -L callsToValidate.vcf \
      -bt \
      -o output.vcf
 </pre>
    
			<hr>
			<h2>Additional Information</h2>
			<p></p>
			<h3>Read filters</h3>
				<p>These Read Filters are automatically applied to the data by the Engine before processing by GenotypeAndValidate.</p>
			<ul>
					<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>	
			</ul>
			<h3>Parallelism options</h3>
				<p>This tool can be run in multi-threaded mode using this option.</p>
			<ul>
					<li><a href="http://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_engine_CommandLineGATK.html#-nt">TreeReducible (-nt)</a></li>
			</ul>
			<h3>Window size</h3>
			<p>This tool uses a sliding window on the reference.</p>
			<ul>
				<li>Window start: <b>-200</b> bp before the locus</li>
				<li>Window stop: <b>200</b> bp after the locus</li>
			</ul> 
			<hr>
			<h2>Command-line Arguments</h2>
			<p></p>
			<h3>Engine arguments</h3>
			<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
				used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
				engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
				certain read filters to exclude some of the data from the analysis.</p>
			<ul>
					<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
			</ul>
   
		
				<h3>GenotypeAndValidate specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<!--<th>Short name</th> -->
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
        <tr>
            <th colspan="4" id="row-divider">Required Inputs</th>
        </tr>
            <tr>
                <td><a href="#--alleles">--alleles</a><br />
                </td>
                <!--<td>RodBinding[VariantContext]</td> -->
                <td>NA</td>
                <td>The set of alleles at which to genotype</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Outputs</th>
        </tr>
            <tr>
                <td><a href="#--out">--out</a><br />
                			&nbsp;<em>-o</em>
                </td>
                <!--<td>VariantContextWriter</td> -->
                <td>NA</td>
                <td>Output VCF file with annotated variants</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Parameters</th>
        </tr>
            <tr>
                <td><a href="#--condition_on_depth">--condition_on_depth</a><br />
                			&nbsp;<em>-depth</em>
                </td>
                <!--<td>int</td> -->
                <td>-1</td>
                <td>Condition validation on a minimum depth of coverage by the reads</td>
            </tr>
            <tr>
                <td><a href="#--maximum_deletion_fraction">--maximum_deletion_fraction</a><br />
                			&nbsp;<em>-deletions</em>
                </td>
                <!--<td>double</td> -->
                <td>-1.0</td>
                <td>Maximum deletion fraction for calling a genotype</td>
            </tr>
            <tr>
                <td><a href="#--minimum_base_quality_score">--minimum_base_quality_score</a><br />
                			&nbsp;<em>-mbq</em>
                </td>
                <!--<td>int</td> -->
                <td>-1</td>
                <td>Minimum base quality score for calling a genotype</td>
            </tr>
            <tr>
                <td><a href="#--standard_min_confidence_threshold_for_calling">--standard_min_confidence_threshold_for_calling</a><br />
                			&nbsp;<em>-stand_call_conf</em>
                </td>
                <!--<td>double</td> -->
                <td>-1.0</td>
                <td>the minimum phred-scaled Qscore threshold to separate high confidence from low confidence calls</td>
            </tr>
            <tr>
                <td><a href="#--standard_min_confidence_threshold_for_emitting">--standard_min_confidence_threshold_for_emitting</a><br />
                			&nbsp;<em>-stand_emit_conf</em>
                </td>
                <!--<td>double</td> -->
                <td>-1.0</td>
                <td>the minimum phred-scaled Qscore threshold to emit low confidence calls</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Flags</th>
        </tr>
            <tr>
                <td><a href="#--set_bam_truth">--set_bam_truth</a><br />
                			&nbsp;<em>-bt</em>
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>Use the calls on the reads (bam file) as the truth dataset and validate the calls on the VCF</td>
            </tr>
					</tbody>
				</table>
	
						<h3>Argument details</h3>
						<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--alleles">--alleles </a>
		 / <small>-alleles</small>
	</h3>
	<p class="args">
		<b>The set of alleles at which to genotype</b><br />
		The callset to be used as truth (default) or validated (if BAM file is set to truth).
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>required</p> 
	-->
		<p>--alleles binds reference ordered data. This argument supports ROD files of the following types: <a href=htsjdk_variant_bcf2_BCF2Codec.html>BCF2</a>, <a href=htsjdk_variant_vcf_VCFCodec.html>VCF</a>, <a href=htsjdk_variant_vcf_VCF3Codec.html>VCF3</a></p>
	<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">RodBinding[VariantContext]</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--condition_on_depth">--condition_on_depth </a>
		 / <small>-depth</small>
	</h3>
	<p class="args">
		<b>Condition validation on a minimum depth of coverage by the reads</b><br />
		Only validate sites that have at least a given depth
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">int</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--maximum_deletion_fraction">--maximum_deletion_fraction </a>
		 / <small>-deletions</small>
	</h3>
	<p class="args">
		<b>Maximum deletion fraction for calling a genotype</b><br />
		The maximum deletion fraction allowed in a site for calling a genotype. This argument is passed to the Unified Genotyper.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">double</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--minimum_base_quality_score">--minimum_base_quality_score </a>
		 / <small>-mbq</small>
	</h3>
	<p class="args">
		<b>Minimum base quality score for calling a genotype</b><br />
		The minimum base quality score necessary for a base to be considered when calling a genotype. This argument is passed to the Unified Genotyper.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">int</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--out">--out </a>
		 / <small>-o</small>
	</h3>
	<p class="args">
		<b>Output VCF file with annotated variants</b><br />
		The optional output file that will have all the variants used in the Genotype and Validation essay.
 The new annotation `callStatus` will carry the value called in the validation VCF or BAM file."
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">VariantContextWriter</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--set_bam_truth">--set_bam_truth </a>
		 / <small>-bt</small>
	</h3>
	<p class="args">
		<b>Use the calls on the reads (bam file) as the truth dataset and validate the calls on the VCF</b><br />
		Makes the Unified Genotyper calls to the BAM file the truth dataset and validates the alleles ROD binding callset.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--standard_min_confidence_threshold_for_calling">--standard_min_confidence_threshold_for_calling </a>
		 / <small>-stand_call_conf</small>
	</h3>
	<p class="args">
		<b>the minimum phred-scaled Qscore threshold to separate high confidence from low confidence calls</b><br />
		the minimum phred-scaled Qscore threshold to separate high confidence from low confidence calls. This argument is passed to the Unified Genotyper.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">double</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--standard_min_confidence_threshold_for_emitting">--standard_min_confidence_threshold_for_emitting </a>
		 / <small>-stand_emit_conf</small>
	</h3>
	<p class="args">
		<b>the minimum phred-scaled Qscore threshold to emit low confidence calls</b><br />
		the minimum phred-scaled Qscore threshold to emit low confidence calls. This argument is passed to the Unified Genotyper.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">double</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
		




</div>

</div>  