

<div class='row' id="top">





<section class="col-md-4">
    <aside class="well">
        <a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
    </aside>
    <aside class="well">
        <h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandManipulationTools">Variant Evaluation and Manipulation Tools</a>
							<div id="VariantEvaluationandManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ValidationUtilities">Validation Utilities</a>
							<div id="ValidationUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_GenotypeAndValidate.html">GenotypeAndValidate</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#RODCodecs">ROD Codecs</a>
							<div id="RODCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Uncategorized">Uncategorized</a>
							<div id="Uncategorized"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_AnnotatePopulationAFWalker.html">AnnotatePopulationAFWalker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_AssignSomaticStatus.html">AssignSomaticStatus</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantAnnotations">Variant Annotations</a>
							<div id="VariantAnnotations"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
    </aside>


</section>

<div class="col-md-8">      <h1><small>Showing docs for version 3.5-0 </small></h1>     <hr />

    <h1>DiffObjects</h1>

    <p class="lead">A generic engine for comparing tree-structured objects</p>
        <h3>Category
            <small> Diagnostics and Quality Control Tools</small>
        </h3>
		<h3>Traversal
			<small>LocusWalker</small>
		</h3>
		<h3>PartitionBy
			<small>LOCUS</small>
		</h3>

	<hr>
    <h2>Overview</h2>
    

 <p>
      This tool compares two record-oriented files, itemizing specific difference between equivalent
      records in the two files.  Reports both itemized and summarized differences.
 </p>

 <h3>What are the summarized differences and the DiffObjectsWalker?</h3>

 <p>
     The GATK contains a summarizing difference engine that compares hierarchical data structures to emit:
      <ul>
          <li>A list of specific differences between the two data structures.  This is similar to saying the value in field A in record 1 in file F differences from the value in field A in record 1 in file G.</li>
          <li>A summarized list of differences ordered by frequency of the difference.  This output is similar to saying field A in 50 records in files F and G differed.</li>
      </ul>
 </p>

 <p>
      The GATK contains a private walker DiffObjects that allows you access to the DiffEngine capabilities on the command line.  Simply provide the walker with the master and test files and it will emit summarized differences for you.
 </p>

 <h3>Why?</h3>

 <p>
      The reason for this system is that it allows you to compare two structured files -- such as BAMs and VCFs -- for common differences among them.  This is primarily useful in regression testing or optimization, where you want to ensure that the differences are those that you expect and not any others.
 </p>

 <h3>Input</h3>
 <p>
      The DiffObjectsWalker works with BAM or VCF files.
 </p>

 <h3>Output</h3>
 <p>
      The DiffEngine system compares to two hierarchical data structures for specific differences in the values of named
      nodes.  Suppose I have two trees:
 <pre>
     Tree1=(A=1 B=(C=2 D=3))
     Tree2=(A=1 B=(C=3 D=3 E=4))
     Tree3=(A=1 B=(C=4 D=3 E=4))
 </pre>
 <p>
     where every node in the tree is named, or is a raw value (here all leaf values are integers).  The DiffEngine
      traverses these data structures by name, identifies equivalent nodes by fully qualified names
      (Tree1.A is distinct from Tree2.A, and determines where their values are equal (Tree1.A=1, Tree2.A=1, so they are).
      These itemized differences are listed as:
 <pre>
     Tree1.B.C=2 != Tree2.B.C=3
     Tree1.B.C=2 != Tree3.B.C=4
     Tree2.B.C=3 != Tree3.B.C=4
     Tree1.B.E=MISSING != Tree2.B.E=4
 </pre>

 <p>
      This conceptually very similar to the output of the unix command line tool diff.  What's nice about DiffEngine though
      is that it computes similarity among the itemized differences and displays the count of differences names
      in the system.  In the above example, the field C is not equal three times, while the missing E in Tree1 occurs
      only once.  So the summary is:

 <pre>
     *.B.C : 3
     *.B.E : 1
 </pre>

 <p>
      where the * operator indicates that any named field matches.  This output is sorted by counts, and provides an
      immediate picture of the commonly occurring differences among the files.
 <p>
      Below is a detailed example of two VCF fields that differ because of a bug in the AC, AF, and AN counting routines,
      detected by the integrationtest integration (more below).  You can see that in the although there are many specific
      instances of these differences between the two files, the summarized differences provide an immediate picture that
      the AC, AF, and AN fields are the major causes of the differences.
 <p>

 <pre>
 [testng] path                                                             count
 [testng] *.*.*.AC                                                         6
 [testng] *.*.*.AF                                                         6
 [testng] *.*.*.AN                                                         6
 [testng] 64b991fd3850f83614518f7d71f0532f.integrationtest.20:10000000.AC  1
 [testng] 64b991fd3850f83614518f7d71f0532f.integrationtest.20:10000000.AF  1
 [testng] 64b991fd3850f83614518f7d71f0532f.integrationtest.20:10000000.AN  1
 [testng] 64b991fd3850f83614518f7d71f0532f.integrationtest.20:10000117.AC  1
 [testng] 64b991fd3850f83614518f7d71f0532f.integrationtest.20:10000117.AF  1
 [testng] 64b991fd3850f83614518f7d71f0532f.integrationtest.20:10000117.AN  1
 [testng] 64b991fd3850f83614518f7d71f0532f.integrationtest.20:10000211.AC  1
 [testng] 64b991fd3850f83614518f7d71f0532f.integrationtest.20:10000211.AF  1
 [testng] 64b991fd3850f83614518f7d71f0532f.integrationtest.20:10000211.AN  1
 [testng] 64b991fd3850f83614518f7d71f0532f.integrationtest.20:10000598.AC  1
 </pre>

 <h3>Caveat</h3>
 <p>Because this is a walker, it requires that you pass a reference file. However the reference is not actually used, so it does not matter what you pass as reference.</p>
    
			<hr>
			<h2>Additional Information</h2>
			<p></p>
			<h3>Read filters</h3>
				<p>These Read Filters are automatically applied to the data by the Engine before processing by DiffObjects.</p>
			<ul>
					<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>	
			</ul>
			<hr>
			<h2>Command-line Arguments</h2>
			<p></p>
			<h3>Engine arguments</h3>
			<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
				used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
				engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
				certain read filters to exclude some of the data from the analysis.</p>
			<ul>
					<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
			</ul>
   
		
				<h3>DiffObjects specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<!--<th>Short name</th> -->
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
        <tr>
            <th colspan="4" id="row-divider">Required Inputs</th>
        </tr>
            <tr>
                <td><a href="#--master">--master</a><br />
                			&nbsp;<em>-m</em>
                </td>
                <!--<td>File</td> -->
                <td>NA</td>
                <td>Master file: expected results</td>
            </tr>
            <tr>
                <td><a href="#--test">--test</a><br />
                			&nbsp;<em>-t</em>
                </td>
                <!--<td>File</td> -->
                <td>NA</td>
                <td>Test file: new results to compare to the master file</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Outputs</th>
        </tr>
            <tr>
                <td><a href="#--out">--out</a><br />
                			&nbsp;<em>-o</em>
                </td>
                <!--<td>PrintStream</td> -->
                <td>NA</td>
                <td>File to which results should be written</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Parameters</th>
        </tr>
            <tr>
                <td><a href="#--iterations">--iterations</a><br />
                </td>
                <!--<td>int</td> -->
                <td>1</td>
                <td>Number of iterations to perform, should be 1 unless you are doing memory testing</td>
            </tr>
            <tr>
                <td><a href="#--maxCount1Diffs">--maxCount1Diffs</a><br />
                			&nbsp;<em>-M1</em>
                </td>
                <!--<td>int</td> -->
                <td>0</td>
                <td>Max. number of diffs occuring exactly once in the file to process</td>
            </tr>
            <tr>
                <td><a href="#--maxDiffs">--maxDiffs</a><br />
                			&nbsp;<em>-M</em>
                </td>
                <!--<td>int</td> -->
                <td>0</td>
                <td>Max. number of diffs to process</td>
            </tr>
            <tr>
                <td><a href="#--maxObjectsToRead">--maxObjectsToRead</a><br />
                			&nbsp;<em>-motr</em>
                </td>
                <!--<td>int</td> -->
                <td>-1</td>
                <td>Max. number of objects to read from the files.  -1 [default] means unlimited</td>
            </tr>
            <tr>
                <td><a href="#--maxRawDiffsToSummarize">--maxRawDiffsToSummarize</a><br />
                </td>
                <!--<td>int</td> -->
                <td>-1</td>
                <td>Max. number of differences to include in the summary.  -1 [default] means unlimited</td>
            </tr>
            <tr>
                <td><a href="#--minCountForDiff">--minCountForDiff</a><br />
                			&nbsp;<em>-MCFD</em>
                </td>
                <!--<td>int</td> -->
                <td>1</td>
                <td>Min number of observations for a records to display</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Flags</th>
        </tr>
            <tr>
                <td><a href="#--doPairwise">--doPairwise</a><br />
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>If provided, we will compute the minimum pairwise differences to summary, which can be extremely expensive</td>
            </tr>
            <tr>
                <td><a href="#--showItemizedDifferences">--showItemizedDifferences</a><br />
                			&nbsp;<em>-SID</em>
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>Should we enumerate all differences between the files?</td>
            </tr>
					</tbody>
				</table>
	
						<h3>Argument details</h3>
						<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--doPairwise">--doPairwise </a>
		 / <small>-doPairwise</small>
	</h3>
	<p class="args">
		<b>If provided, we will compute the minimum pairwise differences to summary, which can be extremely expensive</b><br />
		
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--iterations">--iterations </a>
		 / <small>NA</small>
	</h3>
	<p class="args">
		<b>Number of iterations to perform, should be 1 unless you are doing memory testing</b><br />
		
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">int</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--master">--master </a>
		 / <small>-m</small>
	</h3>
	<p class="args">
		<b>Master file: expected results</b><br />
		The master file against which we will compare test.  This is one of the two required
 files to do the comparison.  Conceptually master is the original file contained the expected
 results, but this doesn't currently have an impact on the calculations, but might in the future.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>required</p> 
	-->
	<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--maxCount1Diffs">--maxCount1Diffs </a>
		 / <small>-M1</small>
	</h3>
	<p class="args">
		<b>Max. number of diffs occuring exactly once in the file to process</b><br />
		The maximum number of singleton (occurs exactly once between the two files) to display when writing out
 the summary.  Only applies if maxDiffs hasn't been exceeded.  For example, if maxDiffs is 10 and maxCount1Diffs
 is 2 and there are 20 diffs with count > 1, then only 10 are shown, all of which have count above 1.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">int</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--maxDiffs">--maxDiffs </a>
		 / <small>-M</small>
	</h3>
	<p class="args">
		<b>Max. number of diffs to process</b><br />
		The max number of differences to display when summarizing.  For example, if there are 10M differences, but
 maxDiffs is 10, then the comparison aborts after first ten summarized differences are shown.  Note that
 the system shows differences sorted by frequency, so these 10 would be the most common between the two files.
 A value of 0 means show all possible differences.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">int</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--maxObjectsToRead">--maxObjectsToRead </a>
		 / <small>-motr</small>
	</h3>
	<p class="args">
		<b>Max. number of objects to read from the files.  -1 [default] means unlimited</b><br />
		The engine will read at most this number of objects from each of master and test files.  This reduces
 the memory requirements for DiffObjects but does limit you to comparing at most this number of objects
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">int</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--maxRawDiffsToSummarize">--maxRawDiffsToSummarize </a>
		 / <small>-maxRawDiffsToSummarize</small>
	</h3>
	<p class="args">
		<b>Max. number of differences to include in the summary.  -1 [default] means unlimited</b><br />
		
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">int</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--minCountForDiff">--minCountForDiff </a>
		 / <small>-MCFD</small>
	</h3>
	<p class="args">
		<b>Min number of observations for a records to display</b><br />
		Only differences that occur more than minCountForDiff are displayed.  For example, if minCountForDiff is 10, then
 a difference must occur at least 10 times between the two files to be shown.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">int</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--out">--out </a>
		 / <small>-o</small>
	</h3>
	<p class="args">
		<b>File to which results should be written</b><br />
		Writes out a file of the DiffEngine format:

     See http://www.broadinstitute.org/gatk/guide/article?id=1299 for details.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">PrintStream</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--showItemizedDifferences">--showItemizedDifferences </a>
		 / <small>-SID</small>
	</h3>
	<p class="args">
		<b>Should we enumerate all differences between the files?</b><br />
		If provided, the system will write out the summarized, individual differences.  May lead to enormous outputs,
 depending on how many differences are found.  Note these are not sorted in any way, so if you have 10M
 common differences in the files, you will see 10M records, whereas the final summarize will just list the
 difference and its count of 10M.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--test">--test </a>
		 / <small>-t</small>
	</h3>
	<p class="args">
		<b>Test file: new results to compare to the master file</b><br />
		The test file against which we will compare to the master.  This is one of the two required
 files to do the comparison.  Conceptually test is the derived file from master, but this
 doesn't currently have an impact on the calculations, but might in the future.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>required</p> 
	-->
	<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">File</span> 
	</p>
		




</div>

</div>  