

<div class='row' id="top">





<section class="col-md-4">
    <aside class="well">
        <a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
    </aside>
    <aside class="well">
        <h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandManipulationTools">Variant Evaluation and Manipulation Tools</a>
							<div id="VariantEvaluationandManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ValidationUtilities">Validation Utilities</a>
							<div id="ValidationUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_GenotypeAndValidate.html">GenotypeAndValidate</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#RODCodecs">ROD Codecs</a>
							<div id="RODCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Uncategorized">Uncategorized</a>
							<div id="Uncategorized"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_AnnotatePopulationAFWalker.html">AnnotatePopulationAFWalker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_AssignSomaticStatus.html">AssignSomaticStatus</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantAnnotations">Variant Annotations</a>
							<div id="VariantAnnotations"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
    </aside>


</section>

<div class="col-md-8">      <h1><small>Showing docs for version 3.5-0 </small></h1>     <hr />

    <h1>VariantFiltration</h1>

    <p class="lead">Filter variant calls based on INFO and FORMAT annotations</p>
        <h3>Category
            <small> Variant Evaluation and Manipulation Tools</small>
        </h3>
		<h3>Traversal
			<small>LocusWalker</small>
		</h3>
		<h3>PartitionBy
			<small>LOCUS</small>
		</h3>

	<hr>
    <h2>Overview</h2>
    

 <p>
 This tool is designed for hard-filtering variant calls based on certain criteria.
 Records are hard-filtered by changing the value in the FILTER field to something other than PASS. Filtered records
 will be preserved in the output unless their removal is requested in the command line. </p>

 <h3>Input</h3>
 <p>
 A variant set to filter.
 </p>

 <h3>Output</h3>
 <p>
 A filtered VCF.
 </p>

 <h3>Usage example</h3>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T VariantFiltration \
   -R reference.fasta \
   -o output.vcf \
   --variant input.vcf \
   --filterExpression "AB < 0.2 || MQ0 > 50" \
   --filterName "Nov09filters" \
   --mask mask.vcf \
   --maskName InDel
 </pre>
    
			<hr>
			<h2>Additional Information</h2>
			<p></p>
			<h3>Read filters</h3>
				<p>These Read Filters are automatically applied to the data by the Engine before processing by VariantFiltration.</p>
			<ul>
					<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>	
			</ul>
			<h3>Window size</h3>
			<p>This tool uses a sliding window on the reference.</p>
			<ul>
				<li>Window start: <b>-50</b> bp before the locus</li>
				<li>Window stop: <b>50</b> bp after the locus</li>
			</ul> 
			<hr>
			<h2>Command-line Arguments</h2>
			<p></p>
			<h3>Engine arguments</h3>
			<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
				used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
				engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
				certain read filters to exclude some of the data from the analysis.</p>
			<ul>
					<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
			</ul>
   
		
				<h3>VariantFiltration specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<!--<th>Short name</th> -->
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
        <tr>
            <th colspan="4" id="row-divider">Required Inputs</th>
        </tr>
            <tr>
                <td><a href="#--variant">--variant</a><br />
                			&nbsp;<em>-V</em>
                </td>
                <!--<td>RodBinding[VariantContext]</td> -->
                <td>NA</td>
                <td>Input VCF file</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Inputs</th>
        </tr>
            <tr>
                <td><a href="#--mask">--mask</a><br />
                </td>
                <!--<td>RodBinding[Feature]</td> -->
                <td>NA</td>
                <td>Input ROD mask</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Outputs</th>
        </tr>
            <tr>
                <td><a href="#--out">--out</a><br />
                			&nbsp;<em>-o</em>
                </td>
                <!--<td>VariantContextWriter</td> -->
                <td>NA</td>
                <td>File to which variants should be written</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Parameters</th>
        </tr>
            <tr>
                <td><a href="#--clusterSize">--clusterSize</a><br />
                			&nbsp;<em>-cluster</em>
                </td>
                <!--<td>Integer</td> -->
                <td>3</td>
                <td>The number of SNPs which make up a cluster</td>
            </tr>
            <tr>
                <td><a href="#--clusterWindowSize">--clusterWindowSize</a><br />
                			&nbsp;<em>-window</em>
                </td>
                <!--<td>Integer</td> -->
                <td>0</td>
                <td>The window size (in bases) in which to evaluate clustered SNPs</td>
            </tr>
            <tr>
                <td><a href="#--filterExpression">--filterExpression</a><br />
                			&nbsp;<em>-filter</em>
                </td>
                <!--<td>ArrayList[String]</td> -->
                <td>NA</td>
                <td>One or more expression used with INFO fields to filter</td>
            </tr>
            <tr>
                <td><a href="#--filterName">--filterName</a><br />
                </td>
                <!--<td>ArrayList[String]</td> -->
                <td>NA</td>
                <td>Names to use for the list of filters</td>
            </tr>
            <tr>
                <td><a href="#--genotypeFilterExpression">--genotypeFilterExpression</a><br />
                			&nbsp;<em>-G_filter</em>
                </td>
                <!--<td>ArrayList[String]</td> -->
                <td>NA</td>
                <td>One or more expression used with FORMAT (sample/genotype-level) fields to filter (see documentation guide for more info)</td>
            </tr>
            <tr>
                <td><a href="#--genotypeFilterName">--genotypeFilterName</a><br />
                			&nbsp;<em>-G_filterName</em>
                </td>
                <!--<td>ArrayList[String]</td> -->
                <td>NA</td>
                <td>Names to use for the list of sample/genotype filters (must be a 1-to-1 mapping); this name is put in the FILTER field for variants that get filtered</td>
            </tr>
            <tr>
                <td><a href="#--maskExtension">--maskExtension</a><br />
                			&nbsp;<em>-maskExtend</em>
                </td>
                <!--<td>Integer</td> -->
                <td>0</td>
                <td>How many bases beyond records from a provided 'mask' rod should variants be filtered</td>
            </tr>
            <tr>
                <td><a href="#--maskName">--maskName</a><br />
                </td>
                <!--<td>String</td> -->
                <td>NA</td>
                <td>The text to put in the FILTER field if a 'mask' rod is provided and overlaps with a variant call</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Flags</th>
        </tr>
            <tr>
                <td><a href="#--filterNotInMask">--filterNotInMask</a><br />
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>Filter records NOT in given input mask.</td>
            </tr>
            <tr>
                <td><a href="#--invalidatePreviousFilters">--invalidatePreviousFilters</a><br />
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>Remove previous filters applied to the VCF</td>
            </tr>
            <tr>
                <td><a href="#--invertFilterExpression">--invertFilterExpression</a><br />
                			&nbsp;<em>-invfilter</em>
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>Invert the selection criteria for --filterExpression</td>
            </tr>
            <tr>
                <td><a href="#--invertGenotypeFilterExpression">--invertGenotypeFilterExpression</a><br />
                			&nbsp;<em>-invG_filter</em>
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>Invert the selection criteria for --genotypeFilterExpression</td>
            </tr>
            <tr>
                <td><a href="#--missingValuesInExpressionsShouldEvaluateAsFailing">--missingValuesInExpressionsShouldEvaluateAsFailing</a><br />
                </td>
                <!--<td>Boolean</td> -->
                <td>NA</td>
                <td>When evaluating the JEXL expressions, missing values should be considered failing the expression</td>
            </tr>
            <tr>
                <td><a href="#--setFilteredGtToNocall">--setFilteredGtToNocall</a><br />
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>Set filtered genotypes to no-call</td>
            </tr>
					</tbody>
				</table>
	
						<h3>Argument details</h3>
						<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--clusterSize">--clusterSize </a>
		 / <small>-cluster</small>
	</h3>
	<p class="args">
		<b>The number of SNPs which make up a cluster</b><br />
		Works together with the --clusterWindowSize argument.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">Integer</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--clusterWindowSize">--clusterWindowSize </a>
		 / <small>-window</small>
	</h3>
	<p class="args">
		<b>The window size (in bases) in which to evaluate clustered SNPs</b><br />
		Works together with the --clusterSize argument.  To disable the clustered SNP filter, set this value to less than 1.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">Integer</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--filterExpression">--filterExpression </a>
		 / <small>-filter</small>
	</h3>
	<p class="args">
		<b>One or more expression used with INFO fields to filter</b><br />
		VariantFiltration accepts any number of JEXL expressions (so you can have two named filters by using
 --filterName One --filterExpression "X < 1" --filterName Two --filterExpression "X > 2").
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">ArrayList[String]</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--filterName">--filterName </a>
		 / <small>-filterName</small>
	</h3>
	<p class="args">
		<b>Names to use for the list of filters</b><br />
		This name is put in the FILTER field for variants that get filtered.  Note that there must be a 1-to-1 mapping between filter expressions and filter names.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">ArrayList[String]</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--filterNotInMask">--filterNotInMask </a>
		 / <small>-filterNotInMask</small>
	</h3>
	<p class="args">
		<b>Filter records NOT in given input mask.</b><br />
		By default, if the -mask argument is used, any variant falling in a mask will be filtered.
 If this argument is used, logic is reversed, and variants falling outside a given mask will be filtered.
 Use case is, for example, if we have an interval list or BED file with "good" sites.
 Note that it is up to the user to adapt the name of the mask to make it clear that the reverse logic was used
 (e.g. if masking against Hapmap, use -maskName=hapmap for the normal masking and -maskName=not_hapmap for the reverse masking).
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--genotypeFilterExpression">--genotypeFilterExpression </a>
		 / <small>-G_filter</small>
	</h3>
	<p class="args">
		<b>One or more expression used with FORMAT (sample/genotype-level) fields to filter (see documentation guide for more info)</b><br />
		Similar to the INFO field based expressions, but used on the FORMAT (genotype) fields instead.
 VariantFiltration will add the sample-level FT tag to the FORMAT field of filtered samples (this does not affect the record's FILTER tag).
 One can filter normally based on most fields (e.g. "GQ < 5.0"), but the GT (genotype) field is an exception. We have put in convenience
 methods so that one can now filter out hets ("isHet == 1"), refs ("isHomRef == 1"), or homs ("isHomVar == 1"). Also available are
 expressions isCalled, isNoCall, isMixed, and isAvailable, in accordance with the methods of the Genotype object.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">ArrayList[String]</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--genotypeFilterName">--genotypeFilterName </a>
		 / <small>-G_filterName</small>
	</h3>
	<p class="args">
		<b>Names to use for the list of sample/genotype filters (must be a 1-to-1 mapping); this name is put in the FILTER field for variants that get filtered</b><br />
		Similar to the INFO field based expressions, but used on the FORMAT (genotype) fields instead.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">ArrayList[String]</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--invalidatePreviousFilters">--invalidatePreviousFilters </a>
		 / <small>NA</small>
	</h3>
	<p class="args">
		<b>Remove previous filters applied to the VCF</b><br />
		Invalidate previous filters applied to the VariantContext, applying only the filters here
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--invertFilterExpression">--invertFilterExpression </a>
		 / <small>-invfilter</small>
	</h3>
	<p class="args">
		<b>Invert the selection criteria for --filterExpression</b><br />
		Invert the selection criteria for --filterExpression
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--invertGenotypeFilterExpression">--invertGenotypeFilterExpression </a>
		 / <small>-invG_filter</small>
	</h3>
	<p class="args">
		<b>Invert the selection criteria for --genotypeFilterExpression</b><br />
		Invert the selection criteria for --genotypeFilterExpression
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--mask">--mask </a>
		 / <small>-mask</small>
	</h3>
	<p class="args">
		<b>Input ROD mask</b><br />
		Any variant which overlaps entries from the provided mask rod will be filtered. If the user wants logic to be reversed,
 i.e. filter variants that do not overlap with provided mask, then argument -filterNotInMask can be used.
 Note that it is up to the user to adapt the name of the mask to make it clear that the reverse logic was used
 (e.g. if masking against Hapmap, use -maskName=hapmap for the normal masking and -maskName=not_hapmap for the reverse masking).
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
		<p>--mask binds reference ordered data. This argument supports ROD files of the following types: <a href=htsjdk_variant_bcf2_BCF2Codec.html>BCF2</a>, <a href=org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html>BEAGLE</a>, <a href=htsjdk_tribble_bed_BEDCodec.html>BED</a>, <a href=org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html>BEDTABLE</a>, <a href=htsjdk_tribble_example_ExampleBinaryCodec.html>EXAMPLEBINARY</a>, <a href=htsjdk_tribble_gelitext_GeliTextCodec.html>GELITEXT</a>, <a href=org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html>RAWHAPMAP</a>, <a href=org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html>REFSEQ</a>, <a href=org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html>SAMPILEUP</a>, <a href=org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html>SAMREAD</a>, <a href=org_broadinstitute_gatk_utils_codecs_table_TableCodec.html>TABLE</a>, <a href=htsjdk_variant_vcf_VCFCodec.html>VCF</a>, <a href=htsjdk_variant_vcf_VCF3Codec.html>VCF3</a></p>
	<p>
		<span class="label label-info ">RodBinding[Feature]</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--maskExtension">--maskExtension </a>
		 / <small>-maskExtend</small>
	</h3>
	<p class="args">
		<b>How many bases beyond records from a provided 'mask' rod should variants be filtered</b><br />
		
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">Integer</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--maskName">--maskName </a>
		 / <small>-maskName</small>
	</h3>
	<p class="args">
		<b>The text to put in the FILTER field if a 'mask' rod is provided and overlaps with a variant call</b><br />
		When using the -mask argument, the maskName will be annotated in the variant record.
 Note that when using the -filterNotInMask argument to reverse the masking logic,
 it is up to the user to adapt the name of the mask to make it clear that the reverse logic was used
 (e.g. if masking against Hapmap, use -maskName=hapmap for the normal masking and -maskName=not_hapmap for the reverse masking).
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">String</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--missingValuesInExpressionsShouldEvaluateAsFailing">--missingValuesInExpressionsShouldEvaluateAsFailing </a>
		 / <small>NA</small>
	</h3>
	<p class="args">
		<b>When evaluating the JEXL expressions, missing values should be considered failing the expression</b><br />
		By default, if JEXL cannot evaluate your expression for a particular record because one of the annotations is not present, the whole expression evaluates as PASSing.
 Use this argument to have it evaluate as failing filters instead for these cases.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">Boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--out">--out </a>
		 / <small>-o</small>
	</h3>
	<p class="args">
		<b>File to which variants should be written</b><br />
		
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">VariantContextWriter</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--setFilteredGtToNocall">--setFilteredGtToNocall </a>
		 / <small>NA</small>
	</h3>
	<p class="args">
		<b>Set filtered genotypes to no-call</b><br />
		If this argument is provided, set filtered genotypes to no-call (./.).
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--variant">--variant </a>
		 / <small>-V</small>
	</h3>
	<p class="args">
		<b>Input VCF file</b><br />
		Variants from this VCF file are used by this tool as input.
 The file must at least contain the standard VCF header lines, but
 can be empty (i.e., no variants are contained in the file).
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>required</p> 
	-->
		<p>--variant binds reference ordered data. This argument supports ROD files of the following types: <a href=htsjdk_variant_bcf2_BCF2Codec.html>BCF2</a>, <a href=htsjdk_variant_vcf_VCFCodec.html>VCF</a>, <a href=htsjdk_variant_vcf_VCF3Codec.html>VCF3</a></p>
	<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">RodBinding[VariantContext]</span> 
	</p>
		




</div>

</div>  