

<div class='row' id="top">





<section class="col-md-4">
    <aside class="well">
        <a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
    </aside>
    <aside class="well">
        <h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationandManipulationTools">Variant Evaluation and Manipulation Tools</a>
							<div id="VariantEvaluationandManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ValidationUtilities">Validation Utilities</a>
							<div id="ValidationUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_GenotypeAndValidate.html">GenotypeAndValidate</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#RODCodecs">ROD Codecs</a>
							<div id="RODCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#Uncategorized">Uncategorized</a>
							<div id="Uncategorized"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_AnnotatePopulationAFWalker.html">AnnotatePopulationAFWalker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_AssignSomaticStatus.html">AssignSomaticStatus</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantAnnotations">Variant Annotations</a>
							<div id="VariantAnnotations"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
    </aside>


</section>

<div class="col-md-8">      <h1><small>Showing docs for version 3.5-0 </small></h1>     <hr />

    <h1>ValidationSiteSelector</h1>

    <p class="lead">Randomly select variant records according to specified options</p>
        <h3>Category
            <small> Validation Utilities</small>
        </h3>
		<h3>Traversal
			<small>LocusWalker</small>
		</h3>
		<h3>PartitionBy
			<small>LOCUS</small>
		</h3>

	<hr>
    <h2>Overview</h2>
    

 <p>
 This tool is intended for use in experiments where we sample data randomly from a set of variants, for example
 in order to choose sites for a follow-up validation study.</p>

 <p>Sites are selected randomly but within certain restrictions. There are two main sources of restrictions:</p>
 <ul>
     <li><b>Sample restrictions:</b> A user can specify a set of samples, and we will only consider sites which are
     polymorphic within the given sample subset. These sample restrictions can be given as a set of individual
     samples, a text file (each line containing a sample name), or a regular expression. A user can additionally
     specify whether samples will be considered based on their genotypes (a non-reference genotype means that the
     sample is polymorphic in that variant, and hence that variant will be considered for inclusion in set), or
     based on their PLs.</li>
     <li><b>Sampling methods:</b>
          <ol>
              <li>Uniform sampling will just sample uniformly from variants that are polymorphic in selected samples</li>
              <li>Sampling based on Allele Frequency spectrum will ensure that output sites have the same AF distribution as the input set</li>
          </ol>
     </li>
     <li>Variant type (SNP, Indel, etc.)</li>
 </ul>

 <h3>Input</h3>
 <p>
 One or more variant sets to choose from.
 </p>

 <h3>Output</h3>
 <p>
 A sites-only VCF with the desired number of randomly selected sites.
 </p>

 <h3>Usage examples</h3>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T ValidationSiteSelectorWalker \
   -R reference.fasta \
   -V input1.vcf \
   -V input2.vcf \
   -sn NA12878 \
   -o output.vcf \
   --numValidationSites 200   \
   -sampleMode POLY_BASED_ON_GT \
   -freqMode KEEP_AF_SPECTRUM
 </pre>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T ValidationSiteSelectorWalker \
   -R reference.fasta \
   -V:foo input1.vcf \
   -V:bar input2.vcf \
   --numValidationSites 200 \
   -sf samples.txt \
   -o output.vcf \
   -sampleMode  POLY_BASED_ON_GT \
   -freqMode UNIFORM \
   -selectType INDEL
 </pre>
    
			<hr>
			<h2>Additional Information</h2>
			<p></p>
			<h3>Read filters</h3>
				<p>These Read Filters are automatically applied to the data by the Engine before processing by ValidationSiteSelector.</p>
			<ul>
					<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>	
					<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>	
			</ul>
			<hr>
			<h2>Command-line Arguments</h2>
			<p></p>
			<h3>Engine arguments</h3>
			<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
				used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
				engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
				certain read filters to exclude some of the data from the analysis.</p>
			<ul>
					<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
			</ul>
   
		
				<h3>ValidationSiteSelector specific arguments</h3>
				<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
				<table class="table table-striped table-bordered table-condensed">
					<thead>
					<tr>
						<th>Argument name(s)</th>
						<!--<th>Short name</th> -->
						<th>Default value</th>
						<th>Summary</th>
					</tr>
					</thead>
					<tbody>
        <tr>
            <th colspan="4" id="row-divider">Required Inputs</th>
        </tr>
            <tr>
                <td><a href="#--variant">--variant</a><br />
                			&nbsp;<em>-V</em>
                </td>
                <!--<td>List[RodBinding[VariantContext]]</td> -->
                <td>NA</td>
                <td>Input VCF file, can be specified multiple times</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Required Parameters</th>
        </tr>
            <tr>
                <td><a href="#--numValidationSites">--numValidationSites</a><br />
                			&nbsp;<em>-numSites</em>
                </td>
                <!--<td>int</td> -->
                <td>0</td>
                <td>Number of output validation sites</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Inputs</th>
        </tr>
            <tr>
                <td><a href="#--sample_file">--sample_file</a><br />
                			&nbsp;<em>-sf</em>
                </td>
                <!--<td>Set[File]</td> -->
                <td>NA</td>
                <td>File containing a list of samples (one per line) to include. Can be specified multiple times</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Outputs</th>
        </tr>
            <tr>
                <td><a href="#--out">--out</a><br />
                			&nbsp;<em>-o</em>
                </td>
                <!--<td>VariantContextWriter</td> -->
                <td>NA</td>
                <td>File to which variants should be written</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Parameters</th>
        </tr>
            <tr>
                <td><a href="#--frequencySelectionMode">--frequencySelectionMode</a><br />
                			&nbsp;<em>-freqMode</em>
                </td>
                <!--<td>AF_COMPUTATION_MODE</td> -->
                <td>NA</td>
                <td>Allele Frequency selection mode</td>
            </tr>
            <tr>
                <td><a href="#--sample_expressions">--sample_expressions</a><br />
                			&nbsp;<em>-se</em>
                </td>
                <!--<td>Set[String]</td> -->
                <td>NA</td>
                <td>Regular expression to select many samples from the ROD tracks provided. Can be specified multiple times</td>
            </tr>
            <tr>
                <td><a href="#--sample_name">--sample_name</a><br />
                			&nbsp;<em>-sn</em>
                </td>
                <!--<td>Set[String]</td> -->
                <td>NA</td>
                <td>Include genotypes from this sample. Can be specified multiple times</td>
            </tr>
            <tr>
                <td><a href="#--sampleMode">--sampleMode</a><br />
                </td>
                <!--<td>SAMPLE_SELECTION_MODE</td> -->
                <td>NA</td>
                <td>Sample selection mode</td>
            </tr>
            <tr>
                <td><a href="#--samplePNonref">--samplePNonref</a><br />
                </td>
                <!--<td>double</td> -->
                <td>0.99</td>
                <td>GL-based selection mode only: the probability that a site is non-reference in the samples for which to include the site</td>
            </tr>
            <tr>
                <td><a href="#--selectTypeToInclude">--selectTypeToInclude</a><br />
                			&nbsp;<em>-selectType</em>
                </td>
                <!--<td>List[Type]</td> -->
                <td>NA</td>
                <td>Select only a certain type of variants from the input file. Valid types are INDEL, SNP, MIXED, MNP, SYMBOLIC, NO_VARIATION. Can be specified multiple times</td>
            </tr>
        <tr>
            <th colspan="4" id="row-divider">Optional Flags</th>
        </tr>
            <tr>
                <td><a href="#--ignoreGenotypes">--ignoreGenotypes</a><br />
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>If true, will ignore genotypes in VCF, will take AC,AF from annotations and will make no sample selection</td>
            </tr>
            <tr>
                <td><a href="#--ignorePolymorphicStatus">--ignorePolymorphicStatus</a><br />
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>If true, will ignore polymorphic status in VCF, and will take VCF record directly without pre-selection</td>
            </tr>
            <tr>
                <td><a href="#--includeFilteredSites">--includeFilteredSites</a><br />
                			&nbsp;<em>-ifs</em>
                </td>
                <!--<td>boolean</td> -->
                <td>NA</td>
                <td>If true, will include filtered sites in set to choose variants from</td>
            </tr>
					</tbody>
				</table>
	
						<h3>Argument details</h3>
						<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--frequencySelectionMode">--frequencySelectionMode </a>
		 / <small>-freqMode</small>
	</h3>
	<p class="args">
		<b>Allele Frequency selection mode</b><br />
		This argument selects allele frequency selection mode. See the wiki for more information.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
		<p>
		The --frequencySelectionMode argument is an enumerated type (AF_COMPUTATION_MODE), which can have one of the following values:
		<dl class="enum">
				<dt class="enum">KEEP_AF_SPECTRUM</dt>
				<dd class="enum"></dd>
				<dt class="enum">UNIFORM</dt>
				<dd class="enum"></dd>
		</dl>
		</p>
	<p>
		<span class="label label-info ">AF_COMPUTATION_MODE</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--ignoreGenotypes">--ignoreGenotypes </a>
		 / <small>-ignoreGenotypes</small>
	</h3>
	<p class="args">
		<b>If true, will ignore genotypes in VCF, will take AC,AF from annotations and will make no sample selection</b><br />
		Argument for the frequency selection mode. (AC/AF/AN) are taken from VCF info field, not recalculated. Typically specified for sites-only VCFs that still have AC/AF/AN information.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--ignorePolymorphicStatus">--ignorePolymorphicStatus </a>
		 / <small>-ignorePolymorphicStatus</small>
	</h3>
	<p class="args">
		<b>If true, will ignore polymorphic status in VCF, and will take VCF record directly without pre-selection</b><br />
		Argument for the frequency selection mode. Allows reference (non-polymorphic) sites to be included in the validation set.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--includeFilteredSites">--includeFilteredSites </a>
		 / <small>-ifs</small>
	</h3>
	<p class="args">
		<b>If true, will include filtered sites in set to choose variants from</b><br />
		Do not exclude filtered sites (e.g. not PASS or .) from consideration for validation
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">boolean</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--numValidationSites">--numValidationSites </a>
		 / <small>-numSites</small>
	</h3>
	<p class="args">
		<b>Number of output validation sites</b><br />
		The number of sites in your validation set
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>required</p> 
	-->
	<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">int</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--out">--out </a>
		 / <small>-o</small>
	</h3>
	<p class="args">
		<b>File to which variants should be written</b><br />
		The output VCF file
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">VariantContextWriter</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--sample_expressions">--sample_expressions </a>
		 / <small>-se</small>
	</h3>
	<p class="args">
		<b>Regular expression to select many samples from the ROD tracks provided. Can be specified multiple times</b><br />
		Sample regexps to subset the input VCF to, prior to selecting variants. -sn NA12* subsets to all samples with prefix NA12
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">Set[String]</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--sample_file">--sample_file </a>
		 / <small>-sf</small>
	</h3>
	<p class="args">
		<b>File containing a list of samples (one per line) to include. Can be specified multiple times</b><br />
		File containing a list of sample names to subset the input vcf to. Equivalent to specifying the contents of the file separately with -sn
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">Set[File]</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--sample_name">--sample_name </a>
		 / <small>-sn</small>
	</h3>
	<p class="args">
		<b>Include genotypes from this sample. Can be specified multiple times</b><br />
		Sample name(s) to subset the input VCF to, prior to selecting variants. -sn A -sn B subsets to samples A and B.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">Set[String]</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--sampleMode">--sampleMode </a>
		 / <small>-sampleMode</small>
	</h3>
	<p class="args">
		<b>Sample selection mode</b><br />
		A mode for selecting sites based on sample-level data. See the wiki documentation for more information.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
		<p>
		The --sampleMode argument is an enumerated type (SAMPLE_SELECTION_MODE), which can have one of the following values:
		<dl class="enum">
				<dt class="enum">NONE</dt>
				<dd class="enum"></dd>
				<dt class="enum">POLY_BASED_ON_GT</dt>
				<dd class="enum"></dd>
				<dt class="enum">POLY_BASED_ON_GL</dt>
				<dd class="enum"></dd>
		</dl>
		</p>
	<p>
		<span class="label label-info ">SAMPLE_SELECTION_MODE</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--samplePNonref">--samplePNonref </a>
		 / <small>-samplePNonref</small>
	</h3>
	<p class="args">
		<b>GL-based selection mode only: the probability that a site is non-reference in the samples for which to include the site</b><br />
		An P[nonref] threshold for SAMPLE_SELECTION_MODE=POLY_BASED_ON_GL. See the wiki documentation for more information.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">double</span> 
			&nbsp;<span class="label label-warning">[ [ -∞</span>
			&nbsp;<span class="label label-warning">∞ ] ]</span>
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--selectTypeToInclude">--selectTypeToInclude </a>
		 / <small>-selectType</small>
	</h3>
	<p class="args">
		<b>Select only a certain type of variants from the input file. Valid types are INDEL, SNP, MIXED, MNP, SYMBOLIC, NO_VARIATION. Can be specified multiple times</b><br />
		This argument selects particular kinds of variants (i.e. SNP, INDEL) out of a list. If left unspecified, all types are considered.
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>NA</p> 
	-->
	<p>
		<span class="label label-info ">List[Type]</span> 
	</p>
	<hr style="border-bottom: dotted 1px #C0C0C0;" />
	<h3><a name="--variant">--variant </a>
		 / <small>-V</small>
	</h3>
	<p class="args">
		<b>Input VCF file, can be specified multiple times</b><br />
		The input VCF file
	</p>
	<!-- Not sure what this can contain apart from "required" and "deprecated" (which args are @hidden) so removing it provisionally -->
	<!--
		<p>required</p> 
	-->
		<p>--variant binds reference ordered data. This argument supports ROD files of the following types: <a href=htsjdk_variant_bcf2_BCF2Codec.html>BCF2</a>, <a href=htsjdk_variant_vcf_VCFCodec.html>VCF</a>, <a href=htsjdk_variant_vcf_VCF3Codec.html>VCF3</a></p>
	<p>
				<span class="badge badge-important">R</span>
		<span class="label label-info ">List[RodBinding[VariantContext]]</span> 
	</p>
		




</div>

</div>  