## Away Workshop (Brussels) 2014 June 24-26: Best Practices for Variant Calling with GATK

By Geraldine_VdAuwera

<p>Laura Gauthier, David Roazen and Geraldine Van der Auwera presented this workshop in Brussels, Belgium, upon invitation from the <a rel="nofollow" href="https://www.naturalsciences.be/">Royal Belgian Institute for Natural Sciences</a>.</p>

<p>This workshop included two modules:</p>

<ul><li><p><strong>Best Practices for Variant Calling with the GATK</strong></p>

<p>The core steps involved in calling variants with the Broad’s Genome Analysis Toolkit, using the “Best Practices” developed by the GATK team. The presentation materials describe why each step is essential to the calling process, what are the key operations performed on the data at each step, and how to use the GATK tools to get the most accurate and reliable results out of your dataset.</p></li>
<li><p><strong>Beyond the Best Practices</strong></p>

<p>Additional considerations such as calling variants in RNAseq data and calling cohorts efficiently, as well as dealing with non-human data, RNAseq data, whole-genome vs. exome, basic quality control, and performance.</p></li>
</ul><p>This was complemented by a set of hands-on exercises aiming to teach basic GATK usage to new users.</p>

<p>The workshop materials are available <a rel="nofollow" href="http://www.broadinstitute.org/gatk/guide/presentations?id=4765">at this link</a> if you're viewing this post in the forum, or below if you are viewing the presentation page already.</p>

<p><img src="https://us.v-cdn.net/5019796/uploads/FileUpload/aa/2f7372db24ef605a2a7490b084ea2a.png" alt="image" class="embedImage-img importedEmbed-img"></img></p>
