

<div class='row'>

    

    <div class='col-md-9'>

        <div class="row">
            <div class="col-md-6">
                <h1 id="top"> Tool Documentation Index </h1>
            </div>
            <div class="col-md-6">
                <div class="btn-group pull-right" style='margin-top:14px;'>
                    <a class="btn btn-warning dropdown-toggle" data-toggle="dropdown" href="#">
                        Version 3.6-0
                        <span class="caret"></span>
                    </a>
                    
                </div>
            </div>
        </div>

        <hr />
        <em>Showing docs for 3.6-0 </em>
        <hr />

<div class="accordion" id="index">
		<br />
    <div class="accordion-group">
        <div class="accordion-heading">
            <a class="accordion-toggle" data-toggle="collapse" data-parent="#index" href="#EngineParametersavailabletoalltools">
                <h4>Engine Parameters (available to all tools)</h4>
            </a>
        </div>
        <div class="accordion-body collapse" id="EngineParametersavailabletoalltools">
            <div class="accordion-inner">
                <p class="lead"></p>
                <table class="table table-striped table-bordered table-condensed">
                    <tr>
                        <th>Name</th>
                        <th>Summary</th>
                    </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></td>
                                <td>Command line parameters accepted by most if not all tools in the GATK</td>
                            </tr>
                </table>
            </div>
        </div>
    </div>
		<br />
    <div class="accordion-group">
        <div class="accordion-heading">
            <a class="accordion-toggle" data-toggle="collapse" data-parent="#index" href="#DiagnosticsandQualityControlTools">
                <h4>Diagnostics and Quality Control Tools</h4>
            </a>
        </div>
        <div class="accordion-body collapse" id="DiagnosticsandQualityControlTools">
            <div class="accordion-inner">
                <p class="lead"></p>
                <table class="table table-striped table-bordered table-condensed">
                    <tr>
                        <th>Name</th>
                        <th>Summary</th>
                    </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a></td>
                                <td>Calculate read counts per allele for allele-specific expression analysis</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a></td>
                                <td>Create plots to visualize base recalibration results</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a></td>
                                <td>Collect statistics on callable, uncallable, poorly mapped, and other parts of the genome</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a></td>
                                <td>Compare GATK's internal pileup to a reference Samtools pileup</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a></td>
                                <td>Compare callability statistics</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a></td>
                                <td>Estimate cross-sample contamination</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a></td>
                                <td>Count the number of bases in a set of reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a></td>
                                <td>Count contiguous regions in an interval list</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a></td>
                                <td>Count the total number of covered loci</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a></td>
                                <td>Count the number of reads seen from male samples</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a></td>
                                <td>Count the number of ROD objects encountered</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a></td>
                                <td>Count the number of ROD objects encountered along the reference</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a></td>
                                <td>Count the number of read events</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a></td>
                                <td>Count the number of reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a></td>
                                <td>Count the number of reads ending in insertions, deletions or soft-clips</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a></td>
                                <td>Assess sequence coverage by a wide array of metrics, partitioned by sample, read group, or library</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a></td>
                                <td>Analyze coverage distribution and validate read mates per interval and per sample</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a></td>
                                <td>A generic engine for comparing tree-structured objects</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a></td>
                                <td>Compute the read error rate per position</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a></td>
                                <td>Calculate basic statistics about the reference sequence itself</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a></td>
                                <td>Outputs a list of intervals that are covered to or above a given threshold</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a></td>
                                <td>Collect statistics about sequence reads based on their SAM flags</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a></td>
                                <td>Calculates the GC content of the reference sequence for each interval</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a></td>
                                <td>Print read alignments in Pileup-style format</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a></td>
                                <td>Print out all of the RODs in the input data set</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a></td>
                                <td>Collect quality metrics for a set of intervals</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a></td>
                                <td>Collect read clipping statistics</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a></td>
                                <td>Collect statistics about read groups and their properties</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a></td>
                                <td>Collect read length statistics</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a></td>
                                <td>Generate simulated reads for variants</td>
                            </tr>
                </table>
            </div>
        </div>
    </div>
    <div class="accordion-group">
        <div class="accordion-heading">
            <a class="accordion-toggle" data-toggle="collapse" data-parent="#index" href="#SequenceDataProcessingTools">
                <h4>Sequence Data Processing Tools</h4>
            </a>
        </div>
        <div class="accordion-body collapse" id="SequenceDataProcessingTools">
            <div class="accordion-inner">
                <p class="lead"></p>
                <table class="table table-striped table-bordered table-condensed">
                    <tr>
                        <th>Name</th>
                        <th>Summary</th>
                    </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a></td>
                                <td>Detect systematic errors in base quality scores</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a></td>
                                <td>Read clipping based on quality, position or sequence matching</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a></td>
                                <td>Perform local realignment of reads around indels</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a></td>
                                <td>Left-align indels within reads in a bam file</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a></td>
                                <td>Write out sequence read data (for filtering, merging, subsetting etc)</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a></td>
                                <td>Define intervals to target for local realignment</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a></td>
                                <td>Splits reads that contain Ns in their CIGAR string</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a></td>
                                <td>Split a BAM file by sample</td>
                            </tr>
                </table>
            </div>
        </div>
    </div>
    <div class="accordion-group">
        <div class="accordion-heading">
            <a class="accordion-toggle" data-toggle="collapse" data-parent="#index" href="#VariantDiscoveryTools">
                <h4>Variant Discovery Tools</h4>
            </a>
        </div>
        <div class="accordion-body collapse" id="VariantDiscoveryTools">
            <div class="accordion-inner">
                <p class="lead"></p>
                <table class="table table-striped table-bordered table-condensed">
                    <tr>
                        <th>Name</th>
                        <th>Summary</th>
                    </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a></td>
                                <td>Apply a score cutoff to filter variants based on a recalibration table</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a></td>
                                <td>Calculate genotype posterior likelihoods given panel data</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_examples_GATKPaperGenotyper.html">GATKPaperGenotyper</a></td>
                                <td>Simple Bayesian genotyper used in the original GATK paper</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a></td>
                                <td>Perform joint genotyping on gVCF files produced by HaplotypeCaller</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a></td>
                                <td>Call germline SNPs and indels via local re-assembly of haplotypes</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a></td>
                                <td>Call somatic SNPs and indels via local re-assembly of haplotypes</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a></td>
                                <td>Compute the most likely genotype combination and phasing for trios and parent/child pairs</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a></td>
                                <td>Regenotypes the variants from a VCF containing PLs or GLs.</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a></td>
                                <td>Call SNPs and indels on a per-locus basis</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a></td>
                                <td>Build a recalibration model to score variant quality for filtering purposes</td>
                            </tr>
                </table>
            </div>
        </div>
    </div>
    <div class="accordion-group">
        <div class="accordion-heading">
            <a class="accordion-toggle" data-toggle="collapse" data-parent="#index" href="#VariantEvaluationTools">
                <h4>Variant Evaluation Tools</h4>
            </a>
        </div>
        <div class="accordion-body collapse" id="VariantEvaluationTools">
            <div class="accordion-inner">
                <p class="lead"></p>
                <table class="table table-striped table-bordered table-condensed">
                    <tr>
                        <th>Name</th>
                        <th>Summary</th>
                    </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a></td>
                                <td>Genotype concordance between two callsets</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a></td>
                                <td>Validate a VCF file with an extra strict set of criteria</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a></td>
                                <td>General-purpose tool for variant evaluation (% in dbSNP, genotype concordance, Ti/Tv ratios, and a lot more)</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a></td>
                                <td>Filter variant calls based on INFO and FORMAT annotations</td>
                            </tr>
                </table>
            </div>
        </div>
    </div>
    <div class="accordion-group">
        <div class="accordion-heading">
            <a class="accordion-toggle" data-toggle="collapse" data-parent="#index" href="#VariantManipulationTools">
                <h4>Variant Manipulation Tools</h4>
            </a>
        </div>
        <div class="accordion-body collapse" id="VariantManipulationTools">
            <div class="accordion-inner">
                <p class="lead"></p>
                <table class="table table-striped table-bordered table-condensed">
                    <tr>
                        <th>Name</th>
                        <th>Summary</th>
                    </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a></td>
                                <td>Concatenate VCF files of non-overlapping genome intervals, all with the same set of samples</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a></td>
                                <td>Combine per-sample gVCF files produced by HaplotypeCaller into a multi-sample gVCF file</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a></td>
                                <td>Combine variant records from different sources</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a></td>
                                <td>Haplotype-based resolution of variants in separate callsets.</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a></td>
                                <td>Left-align indels in a variant callset</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a></td>
                                <td>Randomly split variants into different sets</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a></td>
                                <td>Annotate physical phasing information</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a></td>
                                <td>Selects headers from a VCF source</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a></td>
                                <td>Select a subset of variants from a larger callset</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a></td>
                                <td>Randomly select variant records according to specified options</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a></td>
                                <td>Annotate variant calls with context information</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a></td>
                                <td>Simplify multi-nucleotide variants (MNPs) into more basic/primitive alleles.</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a></td>
                                <td>Convert VCF to binary pedigree file</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a></td>
                                <td>Extract specific fields from a VCF file to a tab-delimited table</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a></td>
                                <td>Convert variants from other file formats to VCF format</td>
                            </tr>
                </table>
            </div>
        </div>
    </div>
		<br />
    <div class="accordion-group">
        <div class="accordion-heading">
            <a class="accordion-toggle" data-toggle="collapse" data-parent="#index" href="#AnnotationModules">
                <h4>Annotation Modules</h4>
            </a>
        </div>
        <div class="accordion-body collapse" id="AnnotationModules">
            <div class="accordion-inner">
                <p class="lead">Annotations available to VariantAnnotator and the variant callers (some restrictions apply)</p>
                <table class="table table-striped table-bordered table-condensed">
                    <tr>
                        <th>Name</th>
                        <th>Summary</th>
                    </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a></td>
                                <td>Allele-specific rank Sum Test of REF versus ALT base quality scores</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a></td>
                                <td>Allele-specific strand bias estimated using Fisher's Exact Test

 *</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a></td>
                                <td>Allele-specific likelihood-based test for the inbreeding among samples</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InsertSizeRankSum.html">AS_InsertSizeRankSum</a></td>
                                <td>Allele specific Rank Sum Test for insert sizes of REF versus ALT reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MQMateRankSumTest.html">AS_MQMateRankSumTest</a></td>
                                <td>Allele specific Rank Sum Test for mate's mapping qualities of REF versus ALT reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a></td>
                                <td>Allele specific Rank Sum Test for mapping qualities of REF versus ALT reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a></td>
                                <td>Allele-specific call confidence normalized by depth of sample reads supporting the allele</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a></td>
                                <td>Allele-specific Root Mean Square of the mapping quality of reads across all samples.</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a></td>
                                <td>Allele-specific Rank Sum Test for relative positioning of REF versus ALT allele within reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a></td>
                                <td>Allele-specific strand bias estimated by the Symmetric Odds Ratio test</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a></td>
                                <td>Allele balance across all samples</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a></td>
                                <td>Allele balance per sample</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a></td>
                                <td>Allele count and frequency expectation per sample</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a></td>
                                <td>Count of A, C, G, T bases across all samples</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a></td>
                                <td>Count of A, C, G, T bases for each sample</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a></td>
                                <td>Rank Sum Test of REF versus ALT base quality scores</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a></td>
                                <td>Sum of evidence in reads supporting each allele for each sample</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a></td>
                                <td>Counts and frequency of alleles in called genotypes</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a></td>
                                <td>Rank Sum Test for hard-clipped bases on REF versus ALT reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a></td>
                                <td>Total depth of coverage per sample and over all samples.</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a></td>
                                <td>Depth of coverage of each allele per sample</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a></td>
                                <td>Depth of informative coverage for each sample.</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a></td>
                                <td>Phred-scaled p-value for exact test of excess heterozygosity</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a></td>
                                <td>Strand bias estimated using Fisher's Exact Test</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a></td>
                                <td>The fraction of reads deemed informative over the entire cohort</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a></td>
                                <td>GC content of the reference around the given site</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a></td>
                                <td>Summarize genotype statistics from all samples at the site level</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a></td>
                                <td>Consistency of the site with strictly two segregating haplotypes</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a></td>
                                <td>Hardy-Weinberg test for transmission disequilibrium</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a></td>
                                <td>Largest contiguous homopolymer run of the variant allele</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a></td>
                                <td>Likelihood-based test for the inbreeding among samples</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a></td>
                                <td>Rank Sum Test of per-read likelihoods of REF versus ALT reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a></td>
                                <td>Proportion of low quality reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a></td>
                                <td>Likelihood of being a Mendelian Violation</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a></td>
                                <td>Rank Sum Test for mapping qualities of REF versus ALT reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a></td>
                                <td>Count of all reads with MAPQ = 0 across all samples</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a></td>
                                <td>Count of reads with mapping quality zero for each sample</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a></td>
                                <td>Percentage of N bases</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a></td>
                                <td>Count of read pairs in the F1R2 and F2R1 configurations supporting the reference and alternate alleles</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a></td>
                                <td>Existence of a de novo mutation in at least one of the given families</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a></td>
                                <td>Variant call confidence normalized by depth of sample reads supporting a variant</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a></td>
                                <td>Root Mean Square of the mapping quality of reads across all samples.</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a></td>
                                <td>Rank Sum Test for relative positioning of REF versus ALT alleles within reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a></td>
                                <td>List samples that are non-reference at a given site</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a></td>
                                <td>Top effect from SnpEff functional predictions</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a></td>
                                <td>Fraction of reads containing spanning deletions</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a></td>
                                <td>Number of forward and reverse reads that support each allele</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a></td>
                                <td>Number of forward and reverse reads that support REF and ALT alleles</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a></td>
                                <td>Strand bias estimated by the Symmetric Odds Ratio test</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a></td>
                                <td>Tandem repeat unit composition and counts per allele</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a></td>
                                <td>Wittkowski transmission disequilibrium test</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a></td>
                                <td>General category of variant</td>
                            </tr>
                </table>
            </div>
        </div>
    </div>
    <div class="accordion-group">
        <div class="accordion-heading">
            <a class="accordion-toggle" data-toggle="collapse" data-parent="#index" href="#ReadFilters">
                <h4>Read Filters</h4>
            </a>
        </div>
        <div class="accordion-body collapse" id="ReadFilters">
            <div class="accordion-inner">
                <p class="lead">GATK Engine arguments that filter or transfer incoming SAM/BAM data files</p>
                <table class="table table-striped table-bordered table-condensed">
                    <tr>
                        <th>Name</th>
                        <th>Summary</th>
                    </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></td>
                                <td>Filter out reads with wonky CIGAR strings</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a></td>
                                <td>Filter out reads whose mate maps to a different contig</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a></td>
                                <td></td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></td>
                                <td>Filter out duplicate reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></td>
                                <td>Filter out reads that fail the vendor quality check</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a></td>
                                <td>Filter out reads with low mapping qualities for HaplotypeCaller</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a></td>
                                <td>Only use reads from the specified library</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></td>
                                <td>Filter out malformed reads</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a></td>
                                <td>Filter out reads with low mapping qualities</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a></td>
                                <td>Filter out reads with no mapping quality information</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a></td>
                                <td>Filter out reads with mapping quality zero</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a></td>
                                <td>Filter out reads with bad pairing (and related) properties</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a></td>
                                <td>Filter out reads that exceed a given insert size</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a></td>
                                <td>Filter out reads without read group information</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a></td>
                                <td>Filter out reads that do not have an original quality quality score (OQ) tag</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></td>
                                <td>Filter out reads that are secondary alignments</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a></td>
                                <td>Filter out reads that are over-soft-clipped</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a></td>
                                <td>Filter out reads produced by 454 technology</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a></td>
                                <td>Filter out reads that were generated by a specific sequencing platform</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a></td>
                                <td>Filter out reads with blacklisted platform unit tags</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a></td>
                                <td>Filter out reads matching a read group tag value</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a></td>
                                <td>Filter out reads based on length</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a></td>
                                <td>Only use reads with this read name</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a></td>
                                <td>Filter out reads based on strand orientation</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a></td>
                                <td>Set the mapping quality of all reads to a given value</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a></td>
                                <td>Set the mapping quality of reads with a given value to another given value</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_ReassignOriginalMQAfterIndelRealignmentFilter.html">ReassignOriginalMQAfterIndelRealignmentFilter</a></td>
                                <td>Revert the MQ of reads that were modified by IndelRealigner</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a></td>
                                <td>Only use reads belonging to a specific sample</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a></td>
                                <td>Only use reads from the specified read group</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></td>
                                <td>Filter out unmapped reads</td>
                            </tr>
                </table>
            </div>
        </div>
    </div>
    <div class="accordion-group">
        <div class="accordion-heading">
            <a class="accordion-toggle" data-toggle="collapse" data-parent="#index" href="#ResourceFileCodecs">
                <h4>Resource File Codecs</h4>
            </a>
        </div>
        <div class="accordion-body collapse" id="ResourceFileCodecs">
            <div class="accordion-inner">
                <p class="lead">Codecs for reading resource files in reference ordered data (ROD) files such as BED</p>
                <table class="table table-striped table-bordered table-condensed">
                    <tr>
                        <th>Name</th>
                        <th>Summary</th>
                    </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a></td>
                                <td>Codec for Beagle imputation engine</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a></td>
                                <td>The standard table codec that expects loci as contig start stop, not contig:start-stop</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a></td>
                                <td>A codec for the file types produced by the HapMap consortium</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a></td>
                                <td>Allows for reading in RefSeq information</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a></td>
                                <td>Decoder for SAM pileup data</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a></td>
                                <td>Decodes a simple SAM text string</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a></td>
                                <td>Reads tab deliminated tabular text files</td>
                            </tr>
                </table>
            </div>
        </div>
    </div>
		<br />
    <div class="accordion-group">
        <div class="accordion-heading">
            <a class="accordion-toggle" data-toggle="collapse" data-parent="#index" href="#ReferenceUtilities">
                <h4>Reference Utilities</h4>
            </a>
        </div>
        <div class="accordion-body collapse" id="ReferenceUtilities">
            <div class="accordion-inner">
                <p class="lead"></p>
                <table class="table table-striped table-bordered table-condensed">
                    <tr>
                        <th>Name</th>
                        <th>Summary</th>
                    </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a></td>
                                <td>Generate an alternative reference sequence over the specified interval</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a></td>
                                <td>Create a subset of a FASTA reference sequence</td>
                            </tr>
                            <tr>
                                <td><a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a></td>
                                <td>Quality control for the reference fasta</td>
                            </tr>
                </table>
            </div>
        </div>
    </div>
</div>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="http://software.broadinstitute.org/gatk/documentation/">GATK Documentation Index</a> |
        	<a href="index.html">Tool Docs Index</a> |
        	<a href="http://gatkforums.broadinstitute.org/gatk">Support Forum</a>
        </p>

        <p class="version">GATK version 3.6-0-g89b7209 built at 2017/02/09 12:52:48.
        </p>

</div></div>


