

<div class='row' id="top">

	

	

	<section class="col-md-4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_examples_GATKPaperGenotyper.html">GATKPaperGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationTools">Variant Evaluation Tools</a>
							<div id="VariantEvaluationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulationTools">Variant Manipulation Tools</a>
							<div id="VariantManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#AnnotationModules">Annotation Modules</a>
							<div id="AnnotationModules"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InsertSizeRankSum.html">AS_InsertSizeRankSum</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MQMateRankSumTest.html">AS_MQMateRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOriginalMQAfterIndelRealignmentFilter.html">ReassignOriginalMQAfterIndelRealignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ResourceFileCodecs">Resource File Codecs</a>
							<div id="ResourceFileCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
		</aside>
		

	</section>

	<div class="col-md-8">           <h1><small>Showing docs for version 3.6-0 </small></h1>     <hr />

		<h1>VariantRecalibrator</h1>

		<p class="lead">Build a recalibration model to score variant quality for filtering purposes</p>

			<h3>Category
				<small> Variant Discovery Tools</small>
			</h3>
			<h3>Traversal
				<small>LocusWalker</small>
			</h3>
			<h3>PartitionBy
				<small>NONE</small>
			</h3>

		<hr>
		<h2>Overview</h2>
		

 <p>
 The purpose of variant recalibration is to assign a well-calibrated probability to each variant call in a call set.
 You can then create highly accurate call sets by filtering based on this single estimate for the accuracy of each call.
 The approach taken by variant quality score recalibration is to develop a continuous, covarying estimate of the relationship
 between SNP call annotations (such as QD, MQ, and ReadPosRankSum, for example) and the probability that a SNP is a true genetic
 variant versus a sequencing or data processing artifact. This model is determined adaptively based on "true sites" provided
 as input, typically HapMap 3 sites and those sites found to be polymorphic on the Omni 2.5M SNP chip array (in humans). This adaptive
 error model can then be applied to both known and novel variation discovered in the call set of interest to evaluate the
 probability that each call is real. The score that gets added to the INFO field of each variant is called the VQSLOD. It is
 the log odds of being a true variant versus being false under the trained Gaussian mixture model.
 </p>
 
 <p>
 This tool performs the first pass in a two-stage process called VQSR; the second pass is performed by the
 <a href='https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html'>ApplyRecalibration</a> tool.
 In brief, the first pass consists of creating a Gaussian mixture model by looking at the distribution of annotation
 values over a high quality subset of the input call set, and then scoring all input variants according to the model.
 The second pass consists of filtering variants based on score cutoffs identified in the first pass.
</p>

 <p>VQSR is probably the hardest part of the Best Practices to get right, so be sure to read the
 <a href='https://www.broadinstitute.org/gatk/guide/article?id=39'>method documentation</a>,
 <a href='https://www.broadinstitute.org/gatk/guide/article?id=1259'>parameter recommendations</a> and
 <a href='https://www.broadinstitute.org/gatk/guide/article?id=2805'>tutorial</a> to really understand what these
 tools and how to use them for best results on your own data.</p>

 <h3>Inputs</h3>
 <ul>
 <li>The input raw variants to be recalibrated. These variant calls must be annotated with the annotations that will be used for modeling. 
 If the calls come from multiple samples, they must have been obtained by joint calling the samples, either directly (running HaplotypeCaller 
 on all samples together) or via the GVCF workflow (HaplotypeCaller with -ERC GVCF per-sample then GenotypeGVCFs on the resulting gVCFs) 
 which is more scalable.</li>
 <li>Known, truth, and training sets to be used by the algorithm. See the method documentation for more details.</li>
 </ul>

 <h3>Output</h3>
 <ul>
 <li>A recalibration table file that will be used by the ApplyRecalibration tool.</li>
 <li>A tranches file which shows various metrics of the recalibration callset for slices of the data.</li>
 </ul>

 <h3>Usage example</h3>
 <p>Recalibrating SNPs in exome data:</p>
 <pre>
 java -Xmx4g -jar GenomeAnalysisTK.jar \
   -T VariantRecalibrator \
   -R reference.fasta \
   -input raw_variants.vcf \
   -resource:hapmap,known=false,training=true,truth=true,prior=15.0 hapmap_3.3.b37.sites.vcf \
   -resource:omni,known=false,training=true,truth=false,prior=12.0 1000G_omni2.5.b37.sites.vcf \
   -resource:1000G,known=false,training=true,truth=false,prior=10.0 1000G_phase1.snps.high_confidence.vcf
   -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 dbsnp_135.b37.vcf \
   -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR -an InbreedingCoeff \
   -mode SNP \
   -recalFile output.recal \
   -tranchesFile output.tranches \
   -rscriptFile output.plots.R
 </pre>

 <h3>Allele-specfic usage</h3>
 <pre>
 java -Xmx4g -jar GenomeAnalysisTK.jar \
   -T VariantRecalibrator \
   -R reference.fasta \
   -input raw_variants.withASannotations.vcf \
   -AS \
   -resource:hapmap,known=false,training=true,truth=true,prior=15.0 hapmap_3.3.b37.sites.vcf \
   -resource:omni,known=false,training=true,truth=false,prior=12.0 1000G_omni2.5.b37.sites.vcf \
   -resource:1000G,known=false,training=true,truth=false,prior=10.0 1000G_phase1.snps.high_confidence.vcf
   -resource:dbsnp,known=true,training=false,truth=false,prior=2.0 dbsnp_135.b37.vcf \
   -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR -an InbreedingCoeff \
   -mode SNP \
   -recalFile output.AS.recal \
   -tranchesFile output.AS.tranches \
   -rscriptFile output.plots.AS.R
 </pre>
 The input VCF must have been produced using allele-specific annotations in HaplotypeCaller.
 Note that each allele will have a separate line in the output .recal file with its own VQSLOD and culprit that will be transferred to the final VCF in ApplyRecalibration.

 <h3>Caveats</h3>

 <ul>
 <li>SNPs and indels must be recalibrated in separate runs (but it is not necessary to separate them into different files). Mixed records are treated as indels.</li>
 <li>The values used in the example above are only meant to show how the command lines are composed.
 They are not meant to be taken as specific recommendations of values to use in your own work, and they may be
 different from the values cited elsewhere in our documentation. For the latest and greatest recommendations on
 how to set parameter values for you own analyses, please read the Best Practices section of the documentation,
 especially the <a href='https://www.broadinstitute.org/gatk/guide/article?id=1259'>FAQ document</a> on VQSR parameters.</li>
 <li>Whole genomes and exomes take slightly different parameters, so make sure you adapt your commands accordingly! See the documents linked above for details.</li>
 <li>If you work with small datasets (e.g. targeted capture experiments or small number of exomes), you will run into problems. Read the docs linked above for advice on how to deal with those issues.</li>
 <li>In order to create the model reporting plots Rscript needs to be in your environment PATH (this is the scripting version of R, not the interactive version).
 See <a target="r-project" href="http://www.r-project.org">http://www.r-project.org</a> for more info on how to download and install R.</li>
 </ul>

				<hr>
				<h2>Additional Information</h2>
				<p></p>
				<h3>Read filters</h3>
					<p>These Read Filters are automatically applied to the data by the Engine before processing by VariantRecalibrator.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>
				</ul>
				<h3>Parallelism options</h3>
					<p>This tool can be run in multi-threaded mode using this option.</p>
				<ul>
						<li><a href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_engine_CommandLineGATK.html#-nt">TreeReducible (-nt)</a></li>
				</ul>
				<hr>
				<h2>Command-line Arguments</h2>
				<p></p>
				<h3>Engine arguments</h3>
				<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
					used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
					engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
					certain read filters to exclude some of the data from the analysis.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
				</ul>


					<h3>VariantRecalibrator specific arguments</h3>
					<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
					<table class="table table-striped table-bordered table-condensed">
						<thead>
						<tr>
							<th>Argument name(s)</th>
							<th>Default value</th>
							<th>Summary</th>
						</tr>
						</thead>
						<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Inputs</th>
			</tr>
				<tr>
					<td><a href="#--input">--input</a><br />
					</td>
					<!--<td>List[RodBindingCollection[VariantContext]]</td> -->
					<td>NA</td>
					<td>One or more VCFs of raw input variants to be recalibrated</td>
				</tr>
				<tr>
					<td><a href="#--resource">--resource</a><br />
					</td>
					<!--<td>List[RodBinding[VariantContext]]</td> -->
					<td>[]</td>
					<td>A list of sites for which to apply a prior probability of being correct but which aren't used by the algorithm (training and truth sets are required to run)</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Required Outputs</th>
			</tr>
				<tr>
					<td><a href="#--recal_file">--recal_file</a><br />
								&nbsp;<em>-recalFile</em>
					</td>
					<!--<td>VariantContextWriter</td> -->
					<td>NA</td>
					<td>The output recal file used by ApplyRecalibration</td>
				</tr>
				<tr>
					<td><a href="#--tranches_file">--tranches_file</a><br />
								&nbsp;<em>-tranchesFile</em>
					</td>
					<!--<td>File</td> -->
					<td>NA</td>
					<td>The output tranches file used by ApplyRecalibration</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Required Parameters</th>
			</tr>
				<tr>
					<td><a href="#--mode">--mode</a><br />
					</td>
					<!--<td>Mode</td> -->
					<td>SNP</td>
					<td>Recalibration mode to employ</td>
				</tr>
				<tr>
					<td><a href="#--use_annotation">--use_annotation</a><br />
								&nbsp;<em>-an</em>
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>The names of the annotations which should used for calculations</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Inputs</th>
			</tr>
				<tr>
					<td><a href="#--aggregate">--aggregate</a><br />
					</td>
					<!--<td>List[RodBinding[VariantContext]]</td> -->
					<td>NA</td>
					<td>Additional raw input variants to be used in building the model</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Outputs</th>
			</tr>
				<tr>
					<td><a href="#--model_file">--model_file</a><br />
								&nbsp;<em>-modelFile</em>
					</td>
					<!--<td>PrintStream</td> -->
					<td>stdout</td>
					<td>A GATKReport containing the positive and negative model fits</td>
				</tr>
				<tr>
					<td><a href="#--rscript_file">--rscript_file</a><br />
								&nbsp;<em>-rscriptFile</em>
					</td>
					<!--<td>File</td> -->
					<td>NA</td>
					<td>The output rscript file generated by the VQSR to aid in visualization of the input data and learned model</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Parameters</th>
			</tr>
				<tr>
					<td><a href="#--ignore_filter">--ignore_filter</a><br />
								&nbsp;<em>-ignoreFilter</em>
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>If specified, the variant recalibrator will also use variants marked as filtered by the specified filter name in the input VCF file</td>
				</tr>
				<tr>
					<td><a href="#--target_titv">--target_titv</a><br />
								&nbsp;<em>-titv</em>
					</td>
					<!--<td>double</td> -->
					<td>2.15</td>
					<td>The expected novel Ti/Tv ratio to use when calculating FDR tranches and for display on the optimization curve output figures. (approx 2.15 for whole genome experiments). ONLY USED FOR PLOTTING PURPOSES!</td>
				</tr>
				<tr>
					<td><a href="#--TStranche">--TStranche</a><br />
								&nbsp;<em>-tranche</em>
					</td>
					<!--<td>List[Double]</td> -->
					<td>[100.0, 99.9, 99.0, 90.0]</td>
					<td>The levels of truth sensitivity at which to slice the data. (in percent, that is 1.0 for 1 percent)</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Flags</th>
			</tr>
				<tr>
					<td><a href="#--ignore_all_filters">--ignore_all_filters</a><br />
								&nbsp;<em>-ignoreAllFilters</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If specified, the variant recalibrator will ignore all input filters. Useful to rerun the VQSR from a filtered output file.</td>
				</tr>
				<tr>
					<td><a href="#--output_model">--output_model</a><br />
								&nbsp;<em>-outputModel</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If specified, the variant recalibrator will output the VQSR model fit to the file specified by -modelFile or to stdout</td>
				</tr>
				<tr>
					<td><a href="#--useAlleleSpecificAnnotations">--useAlleleSpecificAnnotations</a><br />
								&nbsp;<em>-AS</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If specified, the variant recalibrator will attempt to use the allele-specific versions of the specified annotations.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Parameters</th>
			</tr>
				<tr>
					<td><a href="#--badLodCutoff">--badLodCutoff</a><br />
					</td>
					<!--<td>double</td> -->
					<td>-5.0</td>
					<td>LOD score cutoff for selecting bad variants</td>
				</tr>
				<tr>
					<td><a href="#--dirichlet">--dirichlet</a><br />
					</td>
					<!--<td>double</td> -->
					<td>0.001</td>
					<td>The dirichlet parameter in the variational Bayes algorithm.</td>
				</tr>
				<tr>
					<td><a href="#--max_attempts">--max_attempts</a><br />
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>Number of attempts to build a model before failing</td>
				</tr>
				<tr>
					<td><a href="#--maxGaussians">--maxGaussians</a><br />
								&nbsp;<em>-mG</em>
					</td>
					<!--<td>int</td> -->
					<td>8</td>
					<td>Max number of Gaussians for the positive model</td>
				</tr>
				<tr>
					<td><a href="#--maxIterations">--maxIterations</a><br />
								&nbsp;<em>-mI</em>
					</td>
					<!--<td>int</td> -->
					<td>150</td>
					<td>Maximum number of VBEM iterations</td>
				</tr>
				<tr>
					<td><a href="#--maxNegativeGaussians">--maxNegativeGaussians</a><br />
								&nbsp;<em>-mNG</em>
					</td>
					<!--<td>int</td> -->
					<td>2</td>
					<td>Max number of Gaussians for the negative model</td>
				</tr>
				<tr>
					<td><a href="#--maxNumTrainingData">--maxNumTrainingData</a><br />
					</td>
					<!--<td>int</td> -->
					<td>2500000</td>
					<td>Maximum number of training data</td>
				</tr>
				<tr>
					<td><a href="#--minNumBadVariants">--minNumBadVariants</a><br />
								&nbsp;<em>-minNumBad</em>
					</td>
					<!--<td>int</td> -->
					<td>1000</td>
					<td>Minimum number of bad variants</td>
				</tr>
				<tr>
					<td><a href="#--MQCapForLogitJitterTransform">--MQCapForLogitJitterTransform</a><br />
								&nbsp;<em>-MQCap</em>
					</td>
					<!--<td>int</td> -->
					<td>0</td>
					<td>Apply logit transform and jitter to MQ values</td>
				</tr>
				<tr>
					<td><a href="#--numKMeans">--numKMeans</a><br />
								&nbsp;<em>-nKM</em>
					</td>
					<!--<td>int</td> -->
					<td>100</td>
					<td>Number of k-means iterations</td>
				</tr>
				<tr>
					<td><a href="#--priorCounts">--priorCounts</a><br />
					</td>
					<!--<td>double</td> -->
					<td>20.0</td>
					<td>The number of prior counts to use in the variational Bayes algorithm.</td>
				</tr>
				<tr>
					<td><a href="#--shrinkage">--shrinkage</a><br />
					</td>
					<!--<td>double</td> -->
					<td>1.0</td>
					<td>The shrinkage parameter in the variational Bayes algorithm.</td>
				</tr>
				<tr>
					<td><a href="#--stdThreshold">--stdThreshold</a><br />
								&nbsp;<em>-std</em>
					</td>
					<!--<td>double</td> -->
					<td>10.0</td>
					<td>Annotation value divergence threshold (number of standard deviations from the means) </td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Flags</th>
			</tr>
				<tr>
					<td><a href="#--trustAllPolymorphic">--trustAllPolymorphic</a><br />
								&nbsp;<em>-allPoly</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Trust that all the input training sets' unfiltered records contain only polymorphic sites to drastically speed up the computation.</td>
				</tr>
						</tbody>
					</table>

							<h3>Argument details</h3>
							<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--aggregate">--aggregate </a>
			 / <small>-aggregate</small>
		</h3>
		<p class="args">
			<b>Additional raw input variants to be used in building the model</b><br />
			These additional calls should be unfiltered and annotated with the error covariates that are intended to be used for modeling.
		</p>
			<p>This argument supports reference-ordered data (ROD) files in the following formats: BCF2, VCF, VCF3</p>
		<p>
			<span class="label label-info ">List[RodBinding[VariantContext]]</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--badLodCutoff">--badLodCutoff </a>
			 / <small>-badLodCutoff</small>
		</h3>
		<p class="args">
			<b>LOD score cutoff for selecting bad variants</b><br />
			Variants scoring lower than this threshold will be used to build the Gaussian model of bad variants.
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">-5.0</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--dirichlet">--dirichlet </a>
			 / <small>-dirichlet</small>
		</h3>
		<p class="args">
			<b>The dirichlet parameter in the variational Bayes algorithm.</b><br />
			
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">0.001</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ignore_all_filters">--ignore_all_filters </a>
			 / <small>-ignoreAllFilters</small>
		</h3>
		<p class="args">
			<b>If specified, the variant recalibrator will ignore all input filters. Useful to rerun the VQSR from a filtered output file.</b><br />
			
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ignore_filter">--ignore_filter </a>
			 / <small>-ignoreFilter</small>
		</h3>
		<p class="args">
			<b>If specified, the variant recalibrator will also use variants marked as filtered by the specified filter name in the input VCF file</b><br />
			For this to work properly, the -ignoreFilter argument should also be applied to the ApplyRecalibration command.
		</p>
		<p>
			<span class="label label-info ">List[String]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--input">--input </a>
			 / <small>-input</small>
		</h3>
		<p class="args">
			<b>One or more VCFs of raw input variants to be recalibrated</b><br />
			These variant calls must be annotated with the annotations that will be used for modeling. If the calls come from multiple samples, 
 they must have been obtained by joint calling the samples, either directly (running HaplotypeCaller on all samples together) or 
 via the GVCF workflow (HaplotypeCaller with -ERC GVCF per-sample then GenotypeGVCFs on the resulting gVCFs) which is more scalable. 
 Note that the ability to pass multiple input files is only intended to facilitate scatter-gather parallelism (to enable e.g. running on VCFs 
 generated per-chromosome), not to combine different callsets. The variant calls in the separate input files should not overlap.
		</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">List[RodBindingCollection[VariantContext]]</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max_attempts">--max_attempts </a>
			 / <small>-max_attempts</small>
		</h3>
		<p class="args">
			<b>Number of attempts to build a model before failing</b><br />
			The statistical model being built by this tool may fail due to simple statistical sampling
 issues. Rather than dying immediately when the initial model fails, this argument allows the
 tool to restart with a different random seed and try to build the model again. The first
 successfully built model will be kept.

 Note that the most common underlying cause of model building failure is that there is insufficient data to
 build a really robust model. This argument provides a workaround for that issue but it is
 preferable to provide this tool with more data (typically by including more samples or more territory)
 in order to generate a more robust model.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">1</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--maxGaussians">--maxGaussians </a>
			 / <small>-mG</small>
		</h3>
		<p class="args">
			<b>Max number of Gaussians for the positive model</b><br />
			This parameter determines the maximum number of Gaussians that should be used when building a positive model
 using the variational Bayes algorithm.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">8</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--maxIterations">--maxIterations </a>
			 / <small>-mI</small>
		</h3>
		<p class="args">
			<b>Maximum number of VBEM iterations</b><br />
			This parameter determines the maximum number of VBEM iterations to be performed in the variational Bayes algorithm.
 The procedure will normally end when convergence is detected.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">150</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--maxNegativeGaussians">--maxNegativeGaussians </a>
			 / <small>-mNG</small>
		</h3>
		<p class="args">
			<b>Max number of Gaussians for the negative model</b><br />
			This parameter determines the maximum number of Gaussians that should be used when building a negative model
 using the variational Bayes algorithm. The actual maximum used is the smaller value between the mG and mNG
 arguments, meaning that if -mG is smaller than -mNG, -mG will be used for both. Note that this number should
 be small (e.g. 4) to achieve the best results.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">2</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--maxNumTrainingData">--maxNumTrainingData </a>
			 / <small>-maxNumTrainingData</small>
		</h3>
		<p class="args">
			<b>Maximum number of training data</b><br />
			The number of variants to use in building the Gaussian mixture model. Training sets larger than this will be randomly downsampled.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">2500000</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--minNumBadVariants">--minNumBadVariants </a>
			 / <small>-minNumBad</small>
		</h3>
		<p class="args">
			<b>Minimum number of bad variants</b><br />
			This parameter determines the minimum number of variants that will be selected from the list of worst scoring
 variants to use for building the Gaussian mixture model of bad variants.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">1000</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--mode">--mode </a>
			 / <small>-mode</small>
		</h3>
		<p class="args">
			<b>Recalibration mode to employ</b><br />
			Use either SNP for recalibrating only SNPs (emitting indels untouched in the output VCF) or INDEL for indels (emitting SNPs untouched in the output VCF).
 There is also a BOTH option for recalibrating both SNPs and indels simultaneously, but this is meant for testing purposes only and should not be used in actual analyses.
		</p>
			<p>
			The --mode argument is an enumerated type (Mode), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">SNP</dt>
					<dd class="enum"></dd>
					<dt class="enum">INDEL</dt>
					<dd class="enum"></dd>
					<dt class="enum">BOTH</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">Mode</span>
				&nbsp;<span class="label">SNP</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--model_file">--model_file </a>
			 / <small>-modelFile</small>
		</h3>
		<p class="args">
			<b>A GATKReport containing the positive and negative model fits</b><br />
			
		</p>
		<p>
			<span class="label label-info ">PrintStream</span>
				&nbsp;<span class="label">stdout</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--MQCapForLogitJitterTransform">--MQCapForLogitJitterTransform </a>
			 / <small>-MQCap</small>
		</h3>
		<p class="args">
			<b>Apply logit transform and jitter to MQ values</b><br />
			MQ is capped at a "max" value (60 for bwa-mem) when the alignment is considered perfect. Typically, a huge
 proportion of the reads in a dataset are perfectly mapped, which yields a distribution of MQ values with a
 blob below the max value and a huge peak at the max value. This does not conform to the expectations of the
 Gaussian mixture model of VQSR and has been observed to yield a ROC curve with a jump.

 This argument aims to mitigate this problem. Using MQCap = X has 2 effects:  (1) MQs are transformed by a scaled
 logit on [0,X] (+ epsilon to avoid division by zero) to make the blob more Gaussian-like and (2) the transformed
 MQ=X are jittered to break the peak into a narrow Gaussian.

 Beware that IndelRealigner, if used, adds 10 to MQ for successfully realigned indels. We recommend to either use
 --read-filter ReassignOriginalMQAfterIndelRealignment with HaplotypeCaller or use a MQCap=max+10 to take that
 into account.

 If this option is not used, or if MQCap is set to 0, MQ will not be transformed.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">0</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--numKMeans">--numKMeans </a>
			 / <small>-nKM</small>
		</h3>
		<p class="args">
			<b>Number of k-means iterations</b><br />
			This parameter determines the number of k-means iterations to perform in order to initialize the means of
 the Gaussians in the Gaussian mixture model.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">100</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--output_model">--output_model </a>
			 / <small>-outputModel</small>
		</h3>
		<p class="args">
			<b>If specified, the variant recalibrator will output the VQSR model fit to the file specified by -modelFile or to stdout</b><br />
			This GATKReport gives information to describe the VQSR model fit. Normalized means for the positive model are
  concatenated as one table and negative model normalized means as another table. Covariances are also concatenated
  for positive and negative models, respectively. Tables of annotation means and standard deviations are provided
  to help describe the normalization. The model fit report can be read in with our R gsalib package. Individual
  model Gaussians can be subset by the value in the "Gaussian" column if desired.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--priorCounts">--priorCounts </a>
			 / <small>-priorCounts</small>
		</h3>
		<p class="args">
			<b>The number of prior counts to use in the variational Bayes algorithm.</b><br />
			
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">20.0</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--recal_file">--recal_file </a>
			 / <small>-recalFile</small>
		</h3>
		<p class="args">
			<b>The output recal file used by ApplyRecalibration</b><br />
			
		</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">VariantContextWriter</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--resource">--resource </a>
			 / <small>-resource</small>
		</h3>
		<p class="args">
			<b>A list of sites for which to apply a prior probability of being correct but which aren't used by the algorithm (training and truth sets are required to run)</b><br />
			Any set of VCF files to use as lists of training, truth, or known sites.
 Training - The program builds the Gaussian mixture model using input variants that overlap with these training sites.
 Truth - The program uses these truth sites to determine where to set the cutoff in VQSLOD sensitivity.
 Known - The program only uses known sites for reporting purposes (to indicate whether variants are already known or novel). They are not used in any calculations by the algorithm itself.
 Bad - A database of known bad variants can be used to supplement the set of worst ranked variants (compared to the Gaussian mixture model) that the program selects from the data to model "bad" variants.
		</p>
			<p>This argument supports reference-ordered data (ROD) files in the following formats: BCF2, VCF, VCF3</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">List[RodBinding[VariantContext]]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--rscript_file">--rscript_file </a>
			 / <small>-rscriptFile</small>
		</h3>
		<p class="args">
			<b>The output rscript file generated by the VQSR to aid in visualization of the input data and learned model</b><br />
			
		</p>
		<p>
			<span class="label label-info ">File</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--shrinkage">--shrinkage </a>
			 / <small>-shrinkage</small>
		</h3>
		<p class="args">
			<b>The shrinkage parameter in the variational Bayes algorithm.</b><br />
			
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">1.0</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--stdThreshold">--stdThreshold </a>
			 / <small>-std</small>
		</h3>
		<p class="args">
			<b>Annotation value divergence threshold (number of standard deviations from the means) </b><br />
			If a variant has annotations more than -std standard deviations away from mean, it won't be used for building
 the Gaussian mixture model.
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">10.0</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--target_titv">--target_titv </a>
			 / <small>-titv</small>
		</h3>
		<p class="args">
			<b>The expected novel Ti/Tv ratio to use when calculating FDR tranches and for display on the optimization curve output figures. (approx 2.15 for whole genome experiments). ONLY USED FOR PLOTTING PURPOSES!</b><br />
			The expected transition / transversion ratio of true novel variants in your targeted region (whole genome, exome, specific
 genes), which varies greatly by the CpG and GC content of the region. See expected Ti/Tv ratios section of the GATK best
 practices documentation (http://www.broadinstitute.org/gatk/guide/best-practices) for more information.
 Normal values are 2.15 for human whole genome values and 3.2 for human whole exomes. Note
 that this parameter is used for display purposes only and isn't used anywhere in the algorithm!
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">2.15</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--tranches_file">--tranches_file </a>
			 / <small>-tranchesFile</small>
		</h3>
		<p class="args">
			<b>The output tranches file used by ApplyRecalibration</b><br />
			
		</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">File</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--trustAllPolymorphic">--trustAllPolymorphic </a>
			 / <small>-allPoly</small>
		</h3>
		<p class="args">
			<b>Trust that all the input training sets' unfiltered records contain only polymorphic sites to drastically speed up the computation.</b><br />
			
		</p>
		<p>
			<span class="label label-info ">Boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--TStranche">--TStranche </a>
			 / <small>-tranche</small>
		</h3>
		<p class="args">
			<b>The levels of truth sensitivity at which to slice the data. (in percent, that is 1.0 for 1 percent)</b><br />
			Add truth sensitivity slices through the call set at the given values. The default values are 100.0, 99.9, 99.0, and 90.0
 which will result in 4 estimated tranches in the final call set: the full set of calls (100% sensitivity at the accessible
 sites in the truth set), a 99.9% truth sensitivity tranche, along with progressively smaller tranches at 99% and 90%.
		</p>
		<p>
			<span class="label label-info ">List[Double]</span>
				&nbsp;<span class="label">[100.0, 99.9, 99.0, 90.0]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--use_annotation">--use_annotation </a>
			 / <small>-an</small>
		</h3>
		<p class="args">
			<b>The names of the annotations which should used for calculations</b><br />
			See the input VCF file's INFO field for a list of all available annotations.
		</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">List[String]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--useAlleleSpecificAnnotations">--useAlleleSpecificAnnotations </a>
			 / <small>-AS</small>
		</h3>
		<p class="args">
			<b>If specified, the variant recalibrator will attempt to use the allele-specific versions of the specified annotations.</b><br />
			Generate a VQSR model using per-allele data instead of the default per-site data, assuming that the input VCF contains allele-specific annotations.
 Annotations should be specified using their full names with AS_ prefix. Non-allele-specific (scalar) annotations will be applied to all alleles.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="http://software.broadinstitute.org/gatk/documentation/">GATK Documentation Index</a> |
        	<a href="index.html">Tool Docs Index</a> |
        	<a href="http://gatkforums.broadinstitute.org/gatk">Support Forum</a>
        </p>

        <p class="version">GATK version 3.6-0-g89b7209 built at 2017/02/09 12:52:48.
        </p>

	</div>

</div>

