

<div class='row' id="top">

	

	

	<section class="col-md-4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_examples_GATKPaperGenotyper.html">GATKPaperGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationTools">Variant Evaluation Tools</a>
							<div id="VariantEvaluationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulationTools">Variant Manipulation Tools</a>
							<div id="VariantManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#AnnotationModules">Annotation Modules</a>
							<div id="AnnotationModules"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InsertSizeRankSum.html">AS_InsertSizeRankSum</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MQMateRankSumTest.html">AS_MQMateRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOriginalMQAfterIndelRealignmentFilter.html">ReassignOriginalMQAfterIndelRealignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ResourceFileCodecs">Resource File Codecs</a>
							<div id="ResourceFileCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
		</aside>
		

	</section>

	<div class="col-md-8">           <h1><small>Showing docs for version 3.6-0 </small></h1>     <hr />

		<h1>ApplyRecalibration</h1>

		<p class="lead">Apply a score cutoff to filter variants based on a recalibration table</p>

			<h3>Category
				<small> Variant Discovery Tools</small>
			</h3>
			<h3>Traversal
				<small>LocusWalker</small>
			</h3>
			<h3>PartitionBy
				<small>LOCUS</small>
			</h3>

		<hr>
		<h2>Overview</h2>
		

 <p>
 This tool performs the second pass in a two-stage process called VQSR; the first pass is performed by the
 <a href='https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html'>VariantRecalibrator</a> tool.
 In brief, the first pass consists of creating a Gaussian mixture model by looking at the distribution of annotation
 values over a high quality subset of the input call set, and then scoring all input variants according to the model.
 The second pass consists of filtering variants based on score cutoffs identified in the first pass.
</p>

 <p>
 Using the tranche file and recalibration table generated by the previous step, the ApplyRecalibration tool looks at each variant's VQSLOD value
 and decides which tranche it falls in. Variants in tranches that fall below the specified truth sensitivity filter level
 have their FILTER field annotated with the corresponding tranche level. This will result in a call set that is filtered
 to the desired level but retains the information necessary to increase sensitivity if needed.</p>

 <p>To be clear, please note that by "filtered", we mean that variants failing the requested tranche cutoff are <b>marked
 as filtered</b> in the output VCF; they are <b>not discarded</b>.</p>

 <p>VQSR is probably the hardest part of the Best Practices to get right, so be sure to read the
 <a href='https://www.broadinstitute.org/gatk/guide/article?id=39'>method documentation</a>,
 <a href='https://www.broadinstitute.org/gatk/guide/article?id=1259'>parameter recommendations</a> and
 <a href='https://www.broadinstitute.org/gatk/guide/article?id=2805'>tutorial</a> to really understand what these
 tools and how to use them for best results on your own data.</p>

 <h3>Input</h3>
 <ul>
 <li>The raw input variants to be filtered.</li>
 <li>The recalibration table file that was generated by the VariantRecalibrator tool.</li>
 <li>The tranches file that was generated by the VariantRecalibrator tool.</li>
 </ul>

 <h3>Output</h3>
 <ul>
 <li>A recalibrated VCF file in which each variant of the requested type is annotated with its VQSLOD and marked as filtered if the score is below the desired quality level.</li>
 </ul>

 <h3>Usage example for filtering SNPs</h3>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T ApplyRecalibration \
   -R reference.fasta \
   -input raw_variants.vcf \
   --ts_filter_level 99.0 \
   -tranchesFile output.tranches \
   -recalFile output.recal \
   -mode SNP \
   -o path/to/output.recalibrated.filtered.vcf
 </pre>

 <h3>Allele-specific usage</h3>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T ApplyRecalibration \
   -R reference.fasta \
   -input raw_variants.withASannotations.vcf \
   -AS \
   --ts_filter_level 99.0 \
   -tranchesFile output.AS.tranches \
   -recalFile output.AS.recal \
   -mode SNP \
   -o path/to/output.recalibrated.ASfiltered.vcf
 </pre>
 Each allele will be annotated by its corresponding entry in the AS_FilterStatus INFO field annotation.  Allele-specific VQSLOD and culprit are also carried through from VariantRecalibrator and stored in the AS_VQSLOD and AS_culprit INFO fields, respectively.
 The site-level filter is set to the most lenient of any of the allele filters.  That is, if one allele passes, the whole site will be PASS.  If no alleles pass, the site-level filter will be set to the lowest sensitivity tranche among all the alleles.

 Note that the .tranches and .recal files should be derived from an allele-specific run of VariantRecalibrator
 Also note that the AS_culprit, AS_FilterStatus, and AS_VQSLOD fields will have placeholder values (NA or NaN) for alleles of a type that have not yet been processed by ApplyRecalibration
 The spanning deletion allele (*) will not be recalibrated because it represents missing data. Its VQSLOD will remain NaN and it's culprit and FilterStatus will be NA.

 <h3>Caveats</h3>

 <ul>
 <li>The tranche values used in the example above are only meant to be a general example. You should determine the level of sensitivity
 that is appropriate for your specific project. Remember that higher sensitivity (more power to detect variants, yay!) comes
 at the cost of specificity (more false negatives, boo!). You have to choose at what point you want to set the tradeoff.</li>
 <li>In order to create the tranche reporting plots (which are only generated for SNPs, not indels!) Rscript needs to be
 in your environment PATH (this is the scripting version of R, not the interactive version).</li>
 </ul>

				<hr>
				<h2>Additional Information</h2>
				<p></p>
				<h3>Read filters</h3>
					<p>These Read Filters are automatically applied to the data by the Engine before processing by ApplyRecalibration.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>
				</ul>
				<h3>Parallelism options</h3>
					<p>This tool can be run in multi-threaded mode using this option.</p>
				<ul>
						<li><a href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_engine_CommandLineGATK.html#-nt">TreeReducible (-nt)</a></li>
				</ul>
				<hr>
				<h2>Command-line Arguments</h2>
				<p></p>
				<h3>Engine arguments</h3>
				<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
					used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
					engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
					certain read filters to exclude some of the data from the analysis.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
				</ul>


					<h3>ApplyRecalibration specific arguments</h3>
					<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
					<table class="table table-striped table-bordered table-condensed">
						<thead>
						<tr>
							<th>Argument name(s)</th>
							<th>Default value</th>
							<th>Summary</th>
						</tr>
						</thead>
						<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Inputs</th>
			</tr>
				<tr>
					<td><a href="#--input">--input</a><br />
					</td>
					<!--<td>List[RodBinding[VariantContext]]</td> -->
					<td>NA</td>
					<td>The raw input variants to be recalibrated</td>
				</tr>
				<tr>
					<td><a href="#--recal_file">--recal_file</a><br />
								&nbsp;<em>-recalFile</em>
					</td>
					<!--<td>RodBinding[VariantContext]</td> -->
					<td>NA</td>
					<td>The input recal file used by ApplyRecalibration</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Inputs</th>
			</tr>
				<tr>
					<td><a href="#--tranches_file">--tranches_file</a><br />
								&nbsp;<em>-tranchesFile</em>
					</td>
					<!--<td>File</td> -->
					<td>NA</td>
					<td>The input tranches file describing where to cut the data</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Outputs</th>
			</tr>
				<tr>
					<td><a href="#--out">--out</a><br />
								&nbsp;<em>-o</em>
					</td>
					<!--<td>VariantContextWriter</td> -->
					<td>stdout</td>
					<td>The output filtered and recalibrated VCF file in which each variant is annotated with its VQSLOD value</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Parameters</th>
			</tr>
				<tr>
					<td><a href="#--ignore_filter">--ignore_filter</a><br />
								&nbsp;<em>-ignoreFilter</em>
					</td>
					<!--<td>String[]</td> -->
					<td>NA</td>
					<td>If specified, the recalibration will be applied to variants marked as filtered by the specified filter name in the input VCF file</td>
				</tr>
				<tr>
					<td><a href="#--mode">--mode</a><br />
					</td>
					<!--<td>Mode</td> -->
					<td>SNP</td>
					<td>Recalibration mode to employ: 1.) SNP for recalibrating only SNPs (emitting indels untouched in the output VCF); 2.) INDEL for indels; and 3.) BOTH for recalibrating both SNPs and indels simultaneously.</td>
				</tr>
				<tr>
					<td><a href="#--ts_filter_level">--ts_filter_level</a><br />
					</td>
					<!--<td>Double</td> -->
					<td>NA</td>
					<td>The truth sensitivity level at which to start filtering</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Flags</th>
			</tr>
				<tr>
					<td><a href="#--excludeFiltered">--excludeFiltered</a><br />
								&nbsp;<em>-ef</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Don't output filtered loci after applying the recalibration</td>
				</tr>
				<tr>
					<td><a href="#--ignore_all_filters">--ignore_all_filters</a><br />
								&nbsp;<em>-ignoreAllFilters</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If specified, the variant recalibrator will ignore all input filters. Useful to rerun the VQSR from a filtered output file.</td>
				</tr>
				<tr>
					<td><a href="#--useAlleleSpecificAnnotations">--useAlleleSpecificAnnotations</a><br />
								&nbsp;<em>-AS</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If specified, the tool will attempt to apply a filter to each allele based on the input tranches and allele-specific .recal file.</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Parameters</th>
			</tr>
				<tr>
					<td><a href="#--lodCutoff">--lodCutoff</a><br />
					</td>
					<!--<td>Double</td> -->
					<td>NA</td>
					<td>The VQSLOD score below which to start filtering</td>
				</tr>
						</tbody>
					</table>

							<h3>Argument details</h3>
							<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--excludeFiltered">--excludeFiltered </a>
			 / <small>-ef</small>
		</h3>
		<p class="args">
			<b>Don't output filtered loci after applying the recalibration</b><br />
			
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ignore_all_filters">--ignore_all_filters </a>
			 / <small>-ignoreAllFilters</small>
		</h3>
		<p class="args">
			<b>If specified, the variant recalibrator will ignore all input filters. Useful to rerun the VQSR from a filtered output file.</b><br />
			
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ignore_filter">--ignore_filter </a>
			 / <small>-ignoreFilter</small>
		</h3>
		<p class="args">
			<b>If specified, the recalibration will be applied to variants marked as filtered by the specified filter name in the input VCF file</b><br />
			For this to work properly, the -ignoreFilter argument should also be applied to the VariantRecalibration command.
		</p>
		<p>
			<span class="label label-info ">String[]</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--input">--input </a>
			 / <small>-input</small>
		</h3>
		<p class="args">
			<b>The raw input variants to be recalibrated</b><br />
			These calls should be unfiltered and annotated with the error covariates that are intended to use for modeling.
		</p>
			<p>This argument supports reference-ordered data (ROD) files in the following formats: BCF2, VCF, VCF3</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">List[RodBinding[VariantContext]]</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--lodCutoff">--lodCutoff </a>
			 / <small>-lodCutoff</small>
		</h3>
		<p class="args">
			<b>The VQSLOD score below which to start filtering</b><br />
			
		</p>
		<p>
			<span class="label label-info ">Double</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--mode">--mode </a>
			 / <small>-mode</small>
		</h3>
		<p class="args">
			<b>Recalibration mode to employ: 1.) SNP for recalibrating only SNPs (emitting indels untouched in the output VCF); 2.) INDEL for indels; and 3.) BOTH for recalibrating both SNPs and indels simultaneously.</b><br />
			
		</p>
			<p>
			The --mode argument is an enumerated type (Mode), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">SNP</dt>
					<dd class="enum"></dd>
					<dt class="enum">INDEL</dt>
					<dd class="enum"></dd>
					<dt class="enum">BOTH</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
			<span class="label label-info ">Mode</span>
				&nbsp;<span class="label">SNP</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--out">--out </a>
			 / <small>-o</small>
		</h3>
		<p class="args">
			<b>The output filtered and recalibrated VCF file in which each variant is annotated with its VQSLOD value</b><br />
			
		</p>
		<p>
			<span class="label label-info ">VariantContextWriter</span>
				&nbsp;<span class="label">stdout</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--recal_file">--recal_file </a>
			 / <small>-recalFile</small>
		</h3>
		<p class="args">
			<b>The input recal file used by ApplyRecalibration</b><br />
			
		</p>
			<p>This argument supports reference-ordered data (ROD) files in the following formats: BCF2, VCF, VCF3</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">RodBinding[VariantContext]</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--tranches_file">--tranches_file </a>
			 / <small>-tranchesFile</small>
		</h3>
		<p class="args">
			<b>The input tranches file describing where to cut the data</b><br />
			
		</p>
		<p>
			<span class="label label-info ">File</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ts_filter_level">--ts_filter_level </a>
			 / <small>-ts_filter_level</small>
		</h3>
		<p class="args">
			<b>The truth sensitivity level at which to start filtering</b><br />
			
		</p>
		<p>
			<span class="label label-info ">Double</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--useAlleleSpecificAnnotations">--useAlleleSpecificAnnotations </a>
			 / <small>-AS</small>
		</h3>
		<p class="args">
			<b>If specified, the tool will attempt to apply a filter to each allele based on the input tranches and allele-specific .recal file.</b><br />
			Filter the input file based on allele-specific recalibration data.  See tool docs for site-level and allele-level filtering details.
  Requires a .recal file produced using an allele-specific run of VariantRecalibrator
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>

        <hr>
        <p><a href='#top'><i class='fa fa-chevron-up'></i> Return to top</a></p>
        <hr>
        <p class="see-also">See also 
        	<a href="http://software.broadinstitute.org/gatk/documentation/">GATK Documentation Index</a> |
        	<a href="index.html">Tool Docs Index</a> |
        	<a href="http://gatkforums.broadinstitute.org/gatk">Support Forum</a>
        </p>

        <p class="version">GATK version 3.6-0-g89b7209 built at 2017/02/09 12:52:48.
        </p>

	</div>

</div>

