

<div class='row' id="top">

	

	

	<section class="col-md-4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_GatherBqsrReports.html">GatherBqsrReports</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_examples_GATKPaperGenotyper.html">GATKPaperGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationTools">Variant Evaluation Tools</a>
							<div id="VariantEvaluationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulationTools">Variant Manipulation Tools</a>
							<div id="VariantManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#AnnotationModules">Annotation Modules</a>
							<div id="AnnotationModules"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InsertSizeRankSum.html">AS_InsertSizeRankSum</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MQMateRankSumTest.html">AS_MQMateRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_ClusteredReadPosition.html">ClusteredReadPosition</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOriginalMQAfterIndelRealignmentFilter.html">ReassignOriginalMQAfterIndelRealignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ResourceFileCodecs">Resource File Codecs</a>
							<div id="ResourceFileCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
		</aside>
		

	</section>

	<div class="col-md-8">           <h1><small>Showing docs for version 3.7-0 </small></h1>     <hr />

		<h1>CombineVariants</h1>

		<p class="lead">Combine variant records from different sources</p>

			<h3>Category
				<small> Variant Manipulation Tools</small>
			</h3>
			<h3>Traversal
				<small>LocusWalker</small>
			</h3>
			<h3>PartitionBy
				<small>LOCUS</small>
			</h3>

		<hr>
		<h2>Overview</h2>
		

 <p>CombineVariants reads in variants records from separate ROD (Reference-Ordered Data) sources and combines them into
 a single VCF. Any number of sources can be input. This tool aims to fulfill two main possible use cases, reflected
 by the two combination options (MERGE and UNION), for merging records at the variant level (the first 8 fields of
 the VCF) or at the genotype level (the rest).</p>

 <ul>
 <li><b>MERGE:</b> combines multiple variant records present at the same site in the different input sources into a
 single variant record in the output. If sample names overlap, then they are "uniquified" by default, which means a
 suffix is appended to make them unique. <em>Note that in version 3.3, the automatic uniquifying was disabled
 (unintentionally), and required setting `-genotypeMergeOptions UNIQUIFY` manually.</em></li>

 <li><b>UNION:</b> assumes that each ROD source represents the same set of samples (although this is not enforced).
 It uses the priority list (if provided) to emit a single record instance at every position represented in the input RODs.</li>
 </ul>

 <p>By default, the input sets will be named variants, variants2, variants3, and so on. You can override this by
 providing an explicit name tag for each input, using the syntax " -V:name,format". Each input tagged in this
 way will be labeled as such in the output (i.e., set=name rather than set=variants2). For example, you could specify
 a set of control samples as " -V:control,vcf my_control_samples.vcf", and the resulting VCF records would contain
 the annotation "set=control" in the INFO field. It is strongly recommended to provide explicit names in this way
 when a rod priority list is provided.</p>

 <p>CombineVariants will emit a record for every site that was present in any of your input VCF files, and will annotate
 (in the set attribute in the INFO field) whether the record had a PASS or FILTER status in each input ROD . In effect,
 CombineVariants always produces a union of the input VCFs.  However, any part of the Venn of the merged VCFs
 can be extracted using JEXL expressions on the set attribute using SelectVariants.  If you want to extract just
 the records in common between two VCFs, you would first run CombineVariants on the two files to generate a single
 VCF and then run SelectVariants to extract the common records with `-select 'set == "Intersection"'`, as worked out
 in the detailed example in the documentation guide.</p>

 <h3>Input</h3>
 <p>
 Two or more variant sets to combine.
 </p>

 <h3>Output</h3>
 <p>
 A combined VCF.
 </p>

 <h3>Usage examples</h3>
 &nbsp;
 <h4>Merge two separate callsets</h4>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T CombineVariants \
   -R reference.fasta \
   --variant input1.vcf \
   --variant input2.vcf \
   -o output.vcf \
   -genotypeMergeOptions UNIQUIFY
 </pre>

 <h4>Get the union of calls made on the same samples </h4>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T CombineVariants \
   -R reference.fasta \
   --variant:foo input1.vcf \
   --variant:bar input2.vcf \
   -o output.vcf \
   -genotypeMergeOptions PRIORITIZE \
   -priority foo,bar
 </pre>

 <h3>Caveats</h3>
 <ul>
 <li>This tool is not intended to manipulate GVCFS! To combine GVCF files output for different samples by HaplotypeCaller, use CombineGVCFs.</li>
 <li>To join intermediate VCFs produced by running jobs in parallel by interval (e.g. by chromosome) from the same sample, use CatVariants.</li>
 </ul>

 <h3>Additional notes</h3>
 <ul>
 <li> Using this tool's multi-threaded parallelism capability is particularly useful
 when converting from VCF to BCF2, which can be time-consuming. In this case each thread spends CPU time
 doing the conversion, and the GATK engine is smart enough to merge the partial BCF2 blocks together
 efficiently.  However, since this merge runs in only one thread, you can quickly reach diminishing
 returns with the number of parallel threads.  In our hands, `-nt 4` works well but `-nt 8` tends to be be too much.</li>
 <li>Since GATK 2.1, when merging multiple VCF records at a site, the combined VCF record has the QUAL of the first
 VCF record with a non-MISSING QUAL value.  The previous behavior was to take the max QUAL, which could result
 in strange downstream confusion</li>
 </ul>

				<hr>
				<h2>Additional Information</h2>
				<p></p>
				<h3>Read filters</h3>
					<p>These Read Filters are automatically applied to the data by the Engine before processing by CombineVariants.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>
				</ul>
				<h3>Parallelism options</h3>
					<p>This tool can be run in multi-threaded mode using this option.</p>
				<ul>
						<li><a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/org_broadinstitute_gatk_engine_CommandLineGATK.html#-nt">TreeReducible (-nt)</a></li>
				</ul>
				<h3>Window size</h3>
				<p>This tool uses a sliding window on the reference.</p>
				<ul>
					<li>Window start: <b>-50</b> bp before the locus</li>
					<li>Window stop: <b>50</b> bp after the locus</li>
				</ul>
				<hr>
				<h2>Command-line Arguments</h2>
				<p></p>
				<h3>Engine arguments</h3>
				<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
					used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
					engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
					certain read filters to exclude some of the data from the analysis.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
				</ul>


					<h3>CombineVariants specific arguments</h3>
					<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
					<table class="table table-striped table-bordered table-condensed">
						<thead>
						<tr>
							<th>Argument name(s)</th>
							<th>Default value</th>
							<th>Summary</th>
						</tr>
						</thead>
						<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Inputs</th>
			</tr>
				<tr>
					<td><a href="#--variant">--variant</a><br />
								&nbsp;<em>-V</em>
					</td>
					<!--<td>List[RodBindingCollection[VariantContext]]</td> -->
					<td>NA</td>
					<td>VCF files to merge together</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Outputs</th>
			</tr>
				<tr>
					<td><a href="#--out">--out</a><br />
								&nbsp;<em>-o</em>
					</td>
					<!--<td>VariantContextWriter</td> -->
					<td>stdout</td>
					<td>File to which variants should be written</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Parameters</th>
			</tr>
				<tr>
					<td><a href="#--filteredrecordsmergetype">--filteredrecordsmergetype</a><br />
								&nbsp;<em>-filteredRecordsMergeType</em>
					</td>
					<!--<td>FilteredRecordMergeType</td> -->
					<td>KEEP_IF_ANY_UNFILTERED</td>
					<td>Determines how we should handle records seen at the same site in the VCF, but with different FILTER fields</td>
				</tr>
				<tr>
					<td><a href="#--genotypemergeoption">--genotypemergeoption</a><br />
								&nbsp;<em>-genotypeMergeOptions</em>
					</td>
					<!--<td>GenotypeMergeType</td> -->
					<td>NA</td>
					<td>Determines how we should merge genotype records for samples shared across the ROD files</td>
				</tr>
				<tr>
					<td><a href="#--minimumN">--minimumN</a><br />
								&nbsp;<em>-minN</em>
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>Minimum number of input files the site must be observed in to be included</td>
				</tr>
				<tr>
					<td><a href="#--rod_priority_list">--rod_priority_list</a><br />
								&nbsp;<em>-priority</em>
					</td>
					<!--<td>String</td> -->
					<td>NA</td>
					<td>Ordered list specifying priority for merging</td>
				</tr>
				<tr>
					<td><a href="#--setKey">--setKey</a><br />
					</td>
					<!--<td>String</td> -->
					<td>set</td>
					<td>Key name for the set attribute</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Flags</th>
			</tr>
				<tr>
					<td><a href="#--assumeIdenticalSamples">--assumeIdenticalSamples</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Assume input VCFs have identical sample sets and disjoint calls</td>
				</tr>
				<tr>
					<td><a href="#--excludeNonVariants">--excludeNonVariants</a><br />
								&nbsp;<em>-env</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Exclude sites where no variation is present after merging</td>
				</tr>
				<tr>
					<td><a href="#--filteredAreUncalled">--filteredAreUncalled</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Treat filtered variants as uncalled</td>
				</tr>
				<tr>
					<td><a href="#--mergeInfoWithMaxAC">--mergeInfoWithMaxAC</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Use the INFO content of the record with the highest AC</td>
				</tr>
				<tr>
					<td><a href="#--minimalVCF">--minimalVCF</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Emit a sites-only file</td>
				</tr>
				<tr>
					<td><a href="#--printComplexMerges">--printComplexMerges</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Emit interesting sites requiring complex compatibility merging to file</td>
				</tr>
				<tr>
					<td><a href="#--suppressCommandLineHeader">--suppressCommandLineHeader</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Do not output the command line to the header</td>
				</tr>
						</tbody>
					</table>

							<h3>Argument details</h3>
							<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--assumeIdenticalSamples">--assumeIdenticalSamples </a>
			 / <small>-assumeIdenticalSamples</small>
		</h3>
		<p class="args">
			<b>Assume input VCFs have identical sample sets and disjoint calls</b><br />
			This option allows you to perform a simple merge (concatenation) to combine the VCFs, drastically reducing
 runtime. Note that in many cases where you think you want to use this option, you may want to check out the
 CatVariants tool instead, because CatVariants provides the same functionality, but does so even more efficiently.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--excludeNonVariants">--excludeNonVariants </a>
			 / <small>-env</small>
		</h3>
		<p class="args">
			<b>Exclude sites where no variation is present after merging</b><br />
			Exclude sites that do not contain any called ALT alleles in the merged callset. The evaluation is made after the
 merging procedure is complete.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--filteredAreUncalled">--filteredAreUncalled </a>
			 / <small>-filteredAreUncalled</small>
		</h3>
		<p class="args">
			<b>Treat filtered variants as uncalled</b><br />
			If enabled, this flag causes filtered variants (i.e. variant records where the FILTER field is populated by
 something other than PASS or a dot) to be omitted from the output.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--filteredrecordsmergetype">--filteredrecordsmergetype </a>
			 / <small>-filteredRecordsMergeType</small>
		</h3>
		<p class="args">
			<b>Determines how we should handle records seen at the same site in the VCF, but with different FILTER fields</b><br />
			
		</p>
			<p>
			The --filteredrecordsmergetype argument is an enumerated type (FilteredRecordMergeType), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">KEEP_IF_ANY_UNFILTERED</dt>
					<dd class="enum">Union - leaves the record if any record is unfiltered.</dd>
					<dt class="enum">KEEP_IF_ALL_UNFILTERED</dt>
					<dd class="enum">Requires all records present at site to be unfiltered. VCF files that don't contain the record don't influence this.</dd>
					<dt class="enum">KEEP_UNCONDITIONAL</dt>
					<dd class="enum">If any record is present at this site (regardless of possibly being filtered), then all such records are kept and the filters are reset.</dd>
			</dl>
			</p>
		<p>
			<span class="label label-info ">FilteredRecordMergeType</span>
				&nbsp;<span class="label">KEEP_IF_ANY_UNFILTERED</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--genotypemergeoption">--genotypemergeoption </a>
			 / <small>-genotypeMergeOptions</small>
		</h3>
		<p class="args">
			<b>Determines how we should merge genotype records for samples shared across the ROD files</b><br />
			
		</p>
			<p>
			The --genotypemergeoption argument is an enumerated type (GenotypeMergeType), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">UNIQUIFY</dt>
					<dd class="enum">Make all sample genotypes unique by file. Each sample shared across RODs gets named sample.ROD.</dd>
					<dt class="enum">PRIORITIZE</dt>
					<dd class="enum">Take genotypes in priority order (see the priority argument).</dd>
					<dt class="enum">UNSORTED</dt>
					<dd class="enum">Take the genotypes in any order.</dd>
					<dt class="enum">REQUIRE_UNIQUE</dt>
					<dd class="enum">Require that all samples/genotypes be unique between all inputs.</dd>
			</dl>
			</p>
		<p>
			<span class="label label-info ">GenotypeMergeType</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--mergeInfoWithMaxAC">--mergeInfoWithMaxAC </a>
			 / <small>-mergeInfoWithMaxAC</small>
		</h3>
		<p class="args">
			<b>Use the INFO content of the record with the highest AC</b><br />
			By default, the INFO field of the merged variant record only contains the INFO field attributes for which all
 original overlapping records had the same values. Discordant attributes are therefore discarded. This flag allows you to
 override that behavior and simply copy over the INFO field contents of whichever record had the highest AC value.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--minimalVCF">--minimalVCF </a>
			 / <small>-minimalVCF</small>
		</h3>
		<p class="args">
			<b>Emit a sites-only file</b><br />
			If this flag is enabled, the INFO, FORMAT and sample-level (genotype) fields will not be emitted to the output file.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--minimumN">--minimumN </a>
			 / <small>-minN</small>
		</h3>
		<p class="args">
			<b>Minimum number of input files the site must be observed in to be included</b><br />
			Sites that are present in fewer than this number of inputs will be ignored. This is a convenient way to build
 a collection of common variants and exclude rare variants.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">1</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--out">--out </a>
			 / <small>-o</small>
		</h3>
		<p class="args">
			<b>File to which variants should be written</b><br />
			
		</p>
		<p>
			<span class="label label-info ">VariantContextWriter</span>
				&nbsp;<span class="label">stdout</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--printComplexMerges">--printComplexMerges </a>
			 / <small>-printComplexMerges</small>
		</h3>
		<p class="args">
			<b>Emit interesting sites requiring complex compatibility merging to file</b><br />
			
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--rod_priority_list">--rod_priority_list </a>
			 / <small>-priority</small>
		</h3>
		<p class="args">
			<b>Ordered list specifying priority for merging</b><br />
			Refers to the merging priority behavior described in the tool documentation regarding the choice of which record
 gets emitted when taking the union of variants that contain genotypes. The list must be passed as a
 comma-separated string listing the names of the variant input files. The list must be complete and include all
 variant inputs that are being provided to the tool. Use name tags for best results.
		</p>
		<p>
			<span class="label label-info ">String</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--setKey">--setKey </a>
			 / <small>-setKey</small>
		</h3>
		<p class="args">
			<b>Key name for the set attribute</b><br />
			Key used in the INFO key=value tag emitted describing which set(s) the combined record came from
 (e.g. set=control). This provides the option to override the default naming, so instead of set=control you could
 have it be origin=control, or any other word you want that is not already an INFO field attribute. Set this to
 'null' if you don't want the set attribute emitted at all.
		</p>
		<p>
			<span class="label label-info ">String</span>
				&nbsp;<span class="label">set</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--suppressCommandLineHeader">--suppressCommandLineHeader </a>
			 / <small>-suppressCommandLineHeader</small>
		</h3>
		<p class="args">
			<b>Do not output the command line to the header</b><br />
			By default, this tool writes the command line that was used in the header of the output VCF file. This flag
 enables you to override that behavior . This is most often useful when combining variants for dozens or
 hundreds of smaller VCFs iteratively, to avoid cluttering the header with a lot of command lines.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--variant">--variant </a>
			 / <small>-V</small>
		</h3>
		<p class="args">
			<b>VCF files to merge together</b><br />
			
		</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">List[RodBindingCollection[VariantContext]]</span>
				&nbsp;<span class="label">NA</span>
		</p>

        <hr>


	</div>

</div>

