

<div class='row' id="top">

	

	

	<section class="col-md-4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_GatherBqsrReports.html">GatherBqsrReports</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_examples_GATKPaperGenotyper.html">GATKPaperGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationTools">Variant Evaluation Tools</a>
							<div id="VariantEvaluationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulationTools">Variant Manipulation Tools</a>
							<div id="VariantManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#AnnotationModules">Annotation Modules</a>
							<div id="AnnotationModules"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InsertSizeRankSum.html">AS_InsertSizeRankSum</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MQMateRankSumTest.html">AS_MQMateRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_ClusteredReadPosition.html">ClusteredReadPosition</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOriginalMQAfterIndelRealignmentFilter.html">ReassignOriginalMQAfterIndelRealignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ResourceFileCodecs">Resource File Codecs</a>
							<div id="ResourceFileCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
		</aside>
		

	</section>

	<div class="col-md-8">           <h1><small>Showing docs for version 3.7-0 </small></h1>     <hr />

		<h1>DepthOfCoverage</h1>

		<p class="lead">Assess sequence coverage by a wide array of metrics, partitioned by sample, read group, or library</p>

			<h3>Category
				<small> Diagnostics and Quality Control Tools</small>
			</h3>
			<h3>Traversal
				<small>LocusWalker</small>
			</h3>
			<h3>PartitionBy
				<small>NONE</small>
			</h3>

		<hr>
		<h2>Overview</h2>
		

 <p>
 This tool processes a set of bam files to determine coverage at different levels of partitioning and
 aggregation. Coverage can be analyzed per locus, per interval, per gene, or in total; can be partitioned by
 sample, by read group, by technology, by center, or by library; and can be summarized by mean, median, quartiles,
 and/or percentage of bases covered to or beyond a threshold. Additionally, reads and bases can be filtered by
 mapping or base quality score.
 </p>

 <h3>Input</h3>
 <ul>
     <li>One or more bam files (with proper headers) to be analyzed for coverage statistics</li>
     <li>(Optional) A REFSEQ file to aggregate coverage to the gene level (for information about creating the REFSEQ Rod, please consult the online documentation)</li>
 </ul>

 <h3>Output</h3>
 <p>
 Tables pertaining to different coverage summaries. Suffix on the table files declares the contents:
 </p>
 <ul>
     <li>no suffix: per locus coverage</li>
     <li>_summary: total, mean, median, quartiles, and threshold proportions, aggregated over all bases</li>
     <li>_statistics: coverage histograms (# locus with X coverage), aggregated over all bases</li>
     <li>_interval_summary: total, mean, median, quartiles, and threshold proportions, aggregated per interval</li>
     <li>_interval_statistics: 2x2 table of # of intervals covered to >= X depth in >=Y samples</li>
     <li>_gene_summary: total, mean, median, quartiles, and threshold proportions, aggregated per gene</li>
     <li>_gene_statistics: 2x2 table of # of genes covered to >= X depth in >= Y samples</li>
     <li>_cumulative_coverage_counts: coverage histograms (# locus with >= X coverage), aggregated over all bases</li>
     <li>_cumulative_coverage_proportions: proprotions of loci with >= X coverage, aggregated over all bases</li>
 </ul>

 <h3>Usage example</h3>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T DepthOfCoverage \
   -R reference.fasta \
   -o file_name_base \
   -I input_bams.list
   [-geneList refSeq.sorted.txt] \
   [-pt readgroup] \
   [-ct 4 -ct 6 -ct 10] \
   [-L my_capture_genes.interval_list]
 </pre>

				<hr>
				<h2>Additional Information</h2>
				<p></p>
				<h3>Read filters</h3>
					<p>These Read Filters are automatically applied to the data by the Engine before processing by DepthOfCoverage.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>
				</ul>
				<h3>Parallelism options</h3>
					<p>This tool can be run in multi-threaded mode using this option.</p>
				<ul>
						<li><a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/org_broadinstitute_gatk_engine_CommandLineGATK.html#-nt">TreeReducible (-nt)</a></li>
				</ul>
				<h3>Downsampling settings</h3>
					<p>This tool does not apply any downsampling by default.</p>
				<hr>
				<h2>Command-line Arguments</h2>
				<p></p>
				<h3>Engine arguments</h3>
				<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
					used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
					engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
					certain read filters to exclude some of the data from the analysis.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
				</ul>


					<h3>DepthOfCoverage specific arguments</h3>
					<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
					<table class="table table-striped table-bordered table-condensed">
						<thead>
						<tr>
							<th>Argument name(s)</th>
							<th>Default value</th>
							<th>Summary</th>
						</tr>
						</thead>
						<tbody>
			<tr>
				<th colspan="4" id="row-divider">Optional Outputs</th>
			</tr>
				<tr>
					<td><a href="#--out">--out</a><br />
								&nbsp;<em>-o</em>
					</td>
					<!--<td>Map[DoCOutputType,PrintStream]</td> -->
					<td>None</td>
					<td>An output file created by the walker.  Will overwrite contents if file exists</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Parameters</th>
			</tr>
				<tr>
					<td><a href="#--calculateCoverageOverGenes">--calculateCoverageOverGenes</a><br />
								&nbsp;<em>-geneList</em>
					</td>
					<!--<td>File</td> -->
					<td>NA</td>
					<td>Calculate coverage statistics over this list of genes</td>
				</tr>
				<tr>
					<td><a href="#--countType">--countType</a><br />
					</td>
					<!--<td>CountPileupType</td> -->
					<td>COUNT_READS</td>
					<td>How should overlapping reads from the same fragment be handled?</td>
				</tr>
				<tr>
					<td><a href="#--maxBaseQuality">--maxBaseQuality</a><br />
					</td>
					<!--<td>byte</td> -->
					<td>127</td>
					<td>Maximum quality of bases to count towards depth</td>
				</tr>
				<tr>
					<td><a href="#--maxMappingQuality">--maxMappingQuality</a><br />
					</td>
					<!--<td>int</td> -->
					<td>2147483647</td>
					<td>Maximum mapping quality of reads to count towards depth</td>
				</tr>
				<tr>
					<td><a href="#--minBaseQuality">--minBaseQuality</a><br />
								&nbsp;<em>-mbq</em>
					</td>
					<!--<td>byte</td> -->
					<td>-1</td>
					<td>Minimum quality of bases to count towards depth</td>
				</tr>
				<tr>
					<td><a href="#--minMappingQuality">--minMappingQuality</a><br />
								&nbsp;<em>-mmq</em>
					</td>
					<!--<td>int</td> -->
					<td>-1</td>
					<td>Minimum mapping quality of reads to count towards depth</td>
				</tr>
				<tr>
					<td><a href="#--outputFormat">--outputFormat</a><br />
					</td>
					<!--<td>String</td> -->
					<td>rtable</td>
					<td>The format of the output file</td>
				</tr>
				<tr>
					<td><a href="#--partitionType">--partitionType</a><br />
								&nbsp;<em>-pt</em>
					</td>
					<!--<td>Set[Partition]</td> -->
					<td>[sample]</td>
					<td>Partition type for depth of coverage</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Flags</th>
			</tr>
				<tr>
					<td><a href="#--omitDepthOutputAtEachBase">--omitDepthOutputAtEachBase</a><br />
								&nbsp;<em>-omitBaseOutput</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Do not output depth of coverage at each base</td>
				</tr>
				<tr>
					<td><a href="#--omitIntervalStatistics">--omitIntervalStatistics</a><br />
								&nbsp;<em>-omitIntervals</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Do not calculate per-interval statistics</td>
				</tr>
				<tr>
					<td><a href="#--omitLocusTable">--omitLocusTable</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Do not calculate per-sample per-depth counts of loci</td>
				</tr>
				<tr>
					<td><a href="#--omitPerSampleStats">--omitPerSampleStats</a><br />
								&nbsp;<em>-omitSampleSummary</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Do not output the summary files per-sample</td>
				</tr>
				<tr>
					<td><a href="#--printBaseCounts">--printBaseCounts</a><br />
								&nbsp;<em>-baseCounts</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Add base counts to per-locus output</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Parameters</th>
			</tr>
				<tr>
					<td><a href="#--nBins">--nBins</a><br />
					</td>
					<!--<td>int</td> -->
					<td>499</td>
					<td>Number of bins to use for granular binning</td>
				</tr>
				<tr>
					<td><a href="#--start">--start</a><br />
					</td>
					<!--<td>int</td> -->
					<td>1</td>
					<td>Starting (left endpoint) for granular binning</td>
				</tr>
				<tr>
					<td><a href="#--stop">--stop</a><br />
					</td>
					<!--<td>int</td> -->
					<td>500</td>
					<td>Ending (right endpoint) for granular binning</td>
				</tr>
				<tr>
					<td><a href="#--summaryCoverageThreshold">--summaryCoverageThreshold</a><br />
								&nbsp;<em>-ct</em>
					</td>
					<!--<td>int[]</td> -->
					<td>[15]</td>
					<td>Coverage threshold (in percent) for summarizing statistics</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Flags</th>
			</tr>
				<tr>
					<td><a href="#--ignoreDeletionSites">--ignoreDeletionSites</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Ignore sites consisting only of deletions</td>
				</tr>
				<tr>
					<td><a href="#--includeDeletions">--includeDeletions</a><br />
								&nbsp;<em>-dels</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Include information on deletions</td>
				</tr>
				<tr>
					<td><a href="#--includeRefNSites">--includeRefNSites</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Include sites where the reference is N</td>
				</tr>
				<tr>
					<td><a href="#--printBinEndpointsAndExit">--printBinEndpointsAndExit</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Print the bin values and exit immediately</td>
				</tr>
						</tbody>
					</table>

							<h3>Argument details</h3>
							<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--calculateCoverageOverGenes">--calculateCoverageOverGenes </a>
			 / <small>-geneList</small>
		</h3>
		<p class="args">
			<b>Calculate coverage statistics over this list of genes</b><br />
			Specify a RefSeq file for use in aggregating coverage statistics over genes.

 This argument is incompatible with --calculateCoverageOverGenes and --omitIntervalStatistics. A warning will be logged and no output file will be produced for the gene list if these arguments are enabled together.
		</p>
		<p>
			<span class="label label-info ">File</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--countType">--countType </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>How should overlapping reads from the same fragment be handled?</b><br />
			
		</p>
			<p>
			The --countType argument is an enumerated type (CountPileupType), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">COUNT_READS</dt>
					<dd class="enum">Count all reads independently (even if from the same fragment).</dd>
					<dt class="enum">COUNT_FRAGMENTS</dt>
					<dd class="enum">Count all fragments (even if the reads that compose the fragment are not consistent at that base).</dd>
					<dt class="enum">COUNT_FRAGMENTS_REQUIRE_SAME_BASE</dt>
					<dd class="enum">Count all fragments (but only if the reads that compose the fragment are consistent at that base).</dd>
			</dl>
			</p>
		<p>
			<span class="label label-info ">CountPileupType</span>
				&nbsp;<span class="label">COUNT_READS</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--ignoreDeletionSites">--ignoreDeletionSites </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Ignore sites consisting only of deletions</b><br />
			
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--includeDeletions">--includeDeletions </a>
			 / <small>-dels</small>
		</h3>
		<p class="args">
			<b>Include information on deletions</b><br />
			Consider a spanning deletion as contributing to coverage. Also enables deletion counts in per-base output.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--includeRefNSites">--includeRefNSites </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Include sites where the reference is N</b><br />
			Normally, sites where the reference is N (or another non-canonical base) are skipped. If this option is enabled, these sites will be included in DoC calculations if there is coverage from neighboring reads.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--maxBaseQuality">--maxBaseQuality </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Maximum quality of bases to count towards depth</b><br />
			Bases with quality scores higher than this threshold will be skipped. The default value is the largest number that can be represented as a byte.
		</p>
		<p>
			<span class="label label-info ">byte</span>
				&nbsp;<span class="label">127</span>
				&nbsp;<span class="label label-warning">[ [ 0</span>
				&nbsp;<span class="label label-warning">127 ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--maxMappingQuality">--maxMappingQuality </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Maximum mapping quality of reads to count towards depth</b><br />
			Reads with mapping quality values higher than this threshold will be skipped. The default value is the largest number that can be represented as an integer by the program.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">2147483647</span>
				&nbsp;<span class="label label-warning">[ [ 0</span>
				&nbsp;<span class="label label-warning">2,147,483,647 ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--minBaseQuality">--minBaseQuality </a>
			 / <small>-mbq</small>
		</h3>
		<p class="args">
			<b>Minimum quality of bases to count towards depth</b><br />
			Bases with quality scores lower than this threshold will be skipped. This is set to -1 by default to disable the evaluation and ignore this threshold.
		</p>
		<p>
			<span class="label label-info ">byte</span>
				&nbsp;<span class="label">-1</span>
				&nbsp;<span class="label label-warning">[ [ 0</span>
				&nbsp;<span class="label label-warning">127 ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--minMappingQuality">--minMappingQuality </a>
			 / <small>-mmq</small>
		</h3>
		<p class="args">
			<b>Minimum mapping quality of reads to count towards depth</b><br />
			Reads with mapping quality values lower than this threshold will be skipped. This is set to -1 by default to disable the evaluation and ignore this threshold.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">-1</span>
				&nbsp;<span class="label label-warning">[ [ 0</span>
				&nbsp;<span class="label label-warning">2,147,483,647 ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--nBins">--nBins </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Number of bins to use for granular binning</b><br />
			Sets the number of bins for granular binning
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">499</span>
				&nbsp;<span class="label label-warning">[ [ 0</span>
				&nbsp;<span class="label label-success">[ 1</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--omitDepthOutputAtEachBase">--omitDepthOutputAtEachBase </a>
			 / <small>-omitBaseOutput</small>
		</h3>
		<p class="args">
			<b>Do not output depth of coverage at each base</b><br />
			Disabling the tabulation of total coverage at every base should speed up processing.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--omitIntervalStatistics">--omitIntervalStatistics </a>
			 / <small>-omitIntervals</small>
		</h3>
		<p class="args">
			<b>Do not calculate per-interval statistics</b><br />
			Disabling the tabulation of interval statistics (mean, median, quartiles AND # intervals by sample by coverage) should speed up processing. This option is required in order to use -nt parallelism.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--omitLocusTable">--omitLocusTable </a>
			 / <small>-omitLocusTable</small>
		</h3>
		<p class="args">
			<b>Do not calculate per-sample per-depth counts of loci</b><br />
			Disabling the tabulation of locus statistics (# loci covered by sample by coverage) should speed up processing.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--omitPerSampleStats">--omitPerSampleStats </a>
			 / <small>-omitSampleSummary</small>
		</h3>
		<p class="args">
			<b>Do not output the summary files per-sample</b><br />
			This option simply disables writing separate files for per-sample summary statistics (total, mean, median, quartile coverage per sample). These statistics are still calculated internally, so enabling this option will not improve runtime.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--out">--out </a>
			 / <small>-o</small>
		</h3>
		<p class="args">
			<b>An output file created by the walker.  Will overwrite contents if file exists</b><br />
			
		</p>
		<p>
			<span class="label label-info ">Map[DoCOutputType,PrintStream]</span>
				&nbsp;<span class="label">None</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--outputFormat">--outputFormat </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>The format of the output file</b><br />
			Output file format (e.g. csv, table, rtable); defaults to r-readable table.
		</p>
		<p>
			<span class="label label-info ">String</span>
				&nbsp;<span class="label">rtable</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--partitionType">--partitionType </a>
			 / <small>-pt</small>
		</h3>
		<p class="args">
			<b>Partition type for depth of coverage</b><br />
			By default, coverage is partitioning by sample, but it can be any combination of sample, readgroup and/or library.
		</p>
		<p>
			<span class="label label-info ">Set[Partition]</span>
				&nbsp;<span class="label">[sample]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--printBaseCounts">--printBaseCounts </a>
			 / <small>-baseCounts</small>
		</h3>
		<p class="args">
			<b>Add base counts to per-locus output</b><br />
			Instead of reporting depth, the program will report the base pileup at each locus
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--printBinEndpointsAndExit">--printBinEndpointsAndExit </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Print the bin values and exit immediately</b><br />
			Use this option to calibrate what bins you want before performing full calculations on your data.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--start">--start </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Starting (left endpoint) for granular binning</b><br />
			Sets the low-coverage cutoff for granular binning. All loci with depth < START are counted in the first bin.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">1</span>
				&nbsp;<span class="label label-warning">[ [ 0</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--stop">--stop </a>
			 / <small>NA</small>
		</h3>
		<p class="args">
			<b>Ending (right endpoint) for granular binning</b><br />
			Sets the high-coverage cutoff for granular binning. All loci with depth > STOP are counted in the last bin.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">500</span>
				&nbsp;<span class="label label-warning">[ [ 1</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--summaryCoverageThreshold">--summaryCoverageThreshold </a>
			 / <small>-ct</small>
		</h3>
		<p class="args">
			<b>Coverage threshold (in percent) for summarizing statistics</b><br />
			For summary file outputs, report the percentage of bases covered to an amount equal to or greater than this number  (e.g. % bases >= CT for each sample). Defaults to 15; can take multiple arguments.
		</p>
		<p>
			<span class="label label-info ">int[]</span>
				&nbsp;<span class="label">[15]</span>
		</p>

        <hr>


	</div>

</div>

