

<div class='row' id="top">

	

	

	<section class="col-md-4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_GatherBqsrReports.html">GatherBqsrReports</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_examples_GATKPaperGenotyper.html">GATKPaperGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationTools">Variant Evaluation Tools</a>
							<div id="VariantEvaluationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulationTools">Variant Manipulation Tools</a>
							<div id="VariantManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#AnnotationModules">Annotation Modules</a>
							<div id="AnnotationModules"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InsertSizeRankSum.html">AS_InsertSizeRankSum</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MQMateRankSumTest.html">AS_MQMateRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_ClusteredReadPosition.html">ClusteredReadPosition</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOriginalMQAfterIndelRealignmentFilter.html">ReassignOriginalMQAfterIndelRealignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ResourceFileCodecs">Resource File Codecs</a>
							<div id="ResourceFileCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
		</aside>
		

	</section>

	<div class="col-md-8">           <h1><small>Showing docs for version 3.7-0 </small></h1>     <hr />

		<h1>GenotypeGVCFs</h1>

		<p class="lead">Perform joint genotyping on gVCF files produced by HaplotypeCaller</p>

			<h3>Category
				<small> Variant Discovery Tools</small>
			</h3>
			<h3>Traversal
				<small>LocusWalker</small>
			</h3>
			<h3>PartitionBy
				<small>LOCUS</small>
			</h3>

		<hr>
		<h2>Overview</h2>
		

 <p>
 GenotypeGVCFs merges gVCF records that were produced as part of the Best Practices workflow for variant discovery
 (see Best Practices documentation for more details) using the '-ERC GVCF' or '-ERC BP_RESOLUTION' mode of the
 HaplotypeCaller, or result from combining such gVCF files using CombineGVCFs. This tool performs the multi-sample
 joint aggregation step and merges the records together in a sophisticated manner: at each position of the input
 gVCFs, this tool will combine all spanning records, produce correct genotype likelihoods, re-genotype the newly
 merged record, and then re-annotate it.</p>

 <h3>Input</h3>
 <p>
 One or more HaplotypeCaller gVCFs to genotype.
 </p>

 <h3>Output</h3>
 <p>
 A combined, genotyped VCF.
 </p>

 <h3>Usage example</h3>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T GenotypeGVCFs \
   -R reference.fasta \
   --variant sample1.g.vcf \
   --variant sample2.g.vcf \
   -o output.vcf
 </pre>

 <h3>Caveat</h3>
 <p>Only gVCF files produced by HaplotypeCaller (or CombineGVCFs) can be used as input for this tool. Some other
 programs produce files that they call gVCFs but those lack some important information (accurate genotype likelihoods
 for every position) that GenotypeGVCFs requires for its operation.</p>

 <h3>Special note on ploidy</h3>
 <p>This tool is able to handle any ploidy (or mix of ploidies) intelligently; there is no need to specify ploidy
 for non-diploid organisms.</p>

				<hr>
				<h2>Additional Information</h2>
				<p></p>
				<h3>Read filters</h3>
					<p>These Read Filters are automatically applied to the data by the Engine before processing by GenotypeGVCFs.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>
				</ul>
				<h3>Parallelism options</h3>
					<p>This tool can be run in multi-threaded mode using this option.</p>
				<ul>
						<li><a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/org_broadinstitute_gatk_engine_CommandLineGATK.html#-nt">TreeReducible (-nt)</a></li>
				</ul>
				<h3>Window size</h3>
				<p>This tool uses a sliding window on the reference.</p>
				<ul>
					<li>Window start: <b>-10</b> bp before the locus</li>
					<li>Window stop: <b>10</b> bp after the locus</li>
				</ul>
				<hr>
				<h2>Command-line Arguments</h2>
				<p></p>
				<h3>Engine arguments</h3>
				<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
					used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
					engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
					certain read filters to exclude some of the data from the analysis.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
				</ul>


					<h3>GenotypeGVCFs specific arguments</h3>
					<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
					<table class="table table-striped table-bordered table-condensed">
						<thead>
						<tr>
							<th>Argument name(s)</th>
							<th>Default value</th>
							<th>Summary</th>
						</tr>
						</thead>
						<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Inputs</th>
			</tr>
				<tr>
					<td><a href="#--variant">--variant</a><br />
								&nbsp;<em>-V</em>
					</td>
					<!--<td>List[RodBindingCollection[VariantContext]]</td> -->
					<td>NA</td>
					<td>One or more input gVCF files</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Inputs</th>
			</tr>
				<tr>
					<td><a href="#--dbsnp">--dbsnp</a><br />
								&nbsp;<em>-D</em>
					</td>
					<!--<td>RodBinding[VariantContext]</td> -->
					<td>none</td>
					<td>dbSNP file</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Outputs</th>
			</tr>
				<tr>
					<td><a href="#--out">--out</a><br />
								&nbsp;<em>-o</em>
					</td>
					<!--<td>VariantContextWriter</td> -->
					<td>stdout</td>
					<td>File to which variants should be written</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Parameters</th>
			</tr>
				<tr>
					<td><a href="#--group">--group</a><br />
								&nbsp;<em>-G</em>
					</td>
					<!--<td>List[String]</td> -->
					<td>[StandardAnnotation]</td>
					<td>One or more classes/groups of annotations to apply to variant calls</td>
				</tr>
				<tr>
					<td><a href="#--heterozygosity">--heterozygosity</a><br />
								&nbsp;<em>-hets</em>
					</td>
					<!--<td>Double</td> -->
					<td>0.001</td>
					<td>Heterozygosity value used to compute prior likelihoods for any locus</td>
				</tr>
				<tr>
					<td><a href="#--heterozygosity_stdev">--heterozygosity_stdev</a><br />
								&nbsp;<em>-heterozygosityStandardDeviation</em>
					</td>
					<!--<td>double</td> -->
					<td>0.01</td>
					<td>Standard deviation of eterozygosity for SNP and indel calling.</td>
				</tr>
				<tr>
					<td><a href="#--indel_heterozygosity">--indel_heterozygosity</a><br />
								&nbsp;<em>-indelHeterozygosity</em>
					</td>
					<!--<td>double</td> -->
					<td>1.25E-4</td>
					<td>Heterozygosity for indel calling</td>
				</tr>
				<tr>
					<td><a href="#--sample_ploidy">--sample_ploidy</a><br />
								&nbsp;<em>-ploidy</em>
					</td>
					<!--<td>int</td> -->
					<td>2</td>
					<td>Ploidy per sample. For pooled data, set to (Number of samples in each pool * Sample Ploidy).</td>
				</tr>
				<tr>
					<td><a href="#--standard_min_confidence_threshold_for_calling">--standard_min_confidence_threshold_for_calling</a><br />
								&nbsp;<em>-stand_call_conf</em>
					</td>
					<!--<td>double</td> -->
					<td>10.0</td>
					<td>The minimum phred-scaled confidence threshold at which variants should be called</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Flags</th>
			</tr>
				<tr>
					<td><a href="#--annotateNDA">--annotateNDA</a><br />
								&nbsp;<em>-nda</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Annotate number of alleles observed</td>
				</tr>
				<tr>
					<td><a href="#--includeNonVariantSites">--includeNonVariantSites</a><br />
								&nbsp;<em>-allSites</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Include loci found to be non-variant after genotyping</td>
				</tr>
				<tr>
					<td><a href="#--useNewAFCalculator">--useNewAFCalculator</a><br />
								&nbsp;<em>-newQual</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Use new AF model instead of the so-called exact model</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Parameters</th>
			</tr>
				<tr>
					<td><a href="#--annotation">--annotation</a><br />
								&nbsp;<em>-A</em>
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>One or more specific annotations to recompute.  The single value 'none' removes the default annotations</td>
				</tr>
				<tr>
					<td><a href="#--input_prior">--input_prior</a><br />
								&nbsp;<em>-inputPrior</em>
					</td>
					<!--<td>List[Double]</td> -->
					<td>[]</td>
					<td>Input prior for calls</td>
				</tr>
				<tr>
					<td><a href="#--max_alternate_alleles">--max_alternate_alleles</a><br />
								&nbsp;<em>-maxAltAlleles</em>
					</td>
					<!--<td>int</td> -->
					<td>6</td>
					<td>Maximum number of alternate alleles to genotype</td>
				</tr>
				<tr>
					<td><a href="#--max_genotype_count">--max_genotype_count</a><br />
								&nbsp;<em>-maxGT</em>
					</td>
					<!--<td>int</td> -->
					<td>1024</td>
					<td>Maximum number of genotypes to consider at any site</td>
				</tr>
				<tr>
					<td><a href="#--max_num_PL_values">--max_num_PL_values</a><br />
								&nbsp;<em>-maxNumPLValues</em>
					</td>
					<!--<td>int</td> -->
					<td>100</td>
					<td>Maximum number of PL values to output</td>
				</tr>
						</tbody>
					</table>

							<h3>Argument details</h3>
							<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--annotateNDA">--annotateNDA </a>
			 / <small>-nda</small>
		</h3>
		<p class="args">
			<b>Annotate number of alleles observed</b><br />
			Depending on the value of the --max_alternate_alleles argument, we may genotype only a fraction of the alleles
 being sent on for genotyping. Using this argument instructs the genotyper to annotate (in the INFO field) the
 number of alternate alleles that were originally discovered (but not necessarily genotyped) at the site.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--annotation">--annotation </a>
			 / <small>-A</small>
		</h3>
		<p class="args">
			<b>One or more specific annotations to recompute.  The single value 'none' removes the default annotations</b><br />
			Which annotations to recompute for the combined output VCF file.
		</p>
		<p>
			<span class="label label-info ">List[String]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--dbsnp">--dbsnp </a>
			 / <small>-D</small>
		</h3>
		<p class="args">
			<b>dbSNP file</b><br />
			The rsIDs from this file are used to populate the ID column of the output.  Also, the DB INFO flag will be set when appropriate. Note that dbSNP is not used in any way for the calculations themselves.
		</p>
			<p>This argument supports reference-ordered data (ROD) files in the following formats: BCF2, VCF, VCF3</p>
		<p>
			<span class="label label-info ">RodBinding[VariantContext]</span>
				&nbsp;<span class="label">none</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--group">--group </a>
			 / <small>-G</small>
		</h3>
		<p class="args">
			<b>One or more classes/groups of annotations to apply to variant calls</b><br />
			Which groups of annotations to add to the output VCF file. The single value 'none' removes the default group. See
 the VariantAnnotator -list argument to view available groups. Note that this usage is not recommended because
 it obscures the specific requirements of individual annotations. Any requirements that are not met (e.g. failing
 to provide a pedigree file for a pedigree-based annotation) may cause the run to fail.
		</p>
		<p>
			<span class="label label-info ">List[String]</span>
				&nbsp;<span class="label">[StandardAnnotation]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--heterozygosity">--heterozygosity </a>
			 / <small>-hets</small>
		</h3>
		<p class="args">
			<b>Heterozygosity value used to compute prior likelihoods for any locus</b><br />
			The expected heterozygosity value used to compute prior probability that a locus is non-reference. See
 https://software.broadinstitute.org/gatk/documentation/article?id=8603 for more details.
		</p>
		<p>
			<span class="label label-info ">Double</span>
				&nbsp;<span class="label">0.001</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--heterozygosity_stdev">--heterozygosity_stdev </a>
			 / <small>-heterozygosityStandardDeviation</small>
		</h3>
		<p class="args">
			<b>Standard deviation of eterozygosity for SNP and indel calling.</b><br />
			The standard deviation of the distribution of alt allele fractions. The above heterozygosity parameters give
 the *mean* of this distribution; this parameter gives its spread.
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">0.01</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--includeNonVariantSites">--includeNonVariantSites </a>
			 / <small>-allSites</small>
		</h3>
		<p class="args">
			<b>Include loci found to be non-variant after genotyping</b><br />
			
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--indel_heterozygosity">--indel_heterozygosity </a>
			 / <small>-indelHeterozygosity</small>
		</h3>
		<p class="args">
			<b>Heterozygosity for indel calling</b><br />
			This argument informs the prior probability of having an indel at a site.
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">1.25E-4</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--input_prior">--input_prior </a>
			 / <small>-inputPrior</small>
		</h3>
		<p class="args">
			<b>Input prior for calls</b><br />
			By default, the prior specified with the argument --heterozygosity/-hets is used for variant discovery at a
 particular locus, using an infinite sites model (see e.g. Waterson, 1975 or Tajima, 1996). This model asserts that
 the probability of having a population of k variant sites in N chromosomes is proportional to theta/k, for 1=1:N.
 However, there are instances where using this prior might not be desirable, e.g. for population studies where prior
 might not be appropriate, as for example when the ancestral status of the reference allele is not known.

 This argument allows you to manually specify a list of probabilities for each AC>1 to be used as
 priors for genotyping, with the following restrictions: only diploid calls are supported; you must specify 2 *
 N values where N is the number of samples; probability values must be positive and specified in Double format,
 in linear space (not log10 space nor Phred-scale); and all values must sume to 1.

 For completely flat priors,  specify the same value (=1/(2*N+1)) 2*N times, e.g.
      -inputPrior 0.33 -inputPrior 0.33
 for the single-sample diploid case.
		</p>
		<p>
			<span class="label label-info ">List[Double]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max_alternate_alleles">--max_alternate_alleles </a>
			 / <small>-maxAltAlleles</small>
		</h3>
		<p class="args">
			<b>Maximum number of alternate alleles to genotype</b><br />
			If there are more than this number of alternate alleles presented to the genotyper (either through discovery or
 GENOTYPE_GIVEN_ALLELES), then only this many alleles will be used.  Note that genotyping sites with many
 alternate alleles is both CPU and memory intensive and it scales exponentially based on the number of alternate
 alleles.  Unless there is a good reason to change the default value, we highly recommend that you not play around
 with this parameter.

 See also {@link #MAX_GENOTYPE_COUNT}.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">6</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max_genotype_count">--max_genotype_count </a>
			 / <small>-maxGT</small>
		</h3>
		<p class="args">
			<b>Maximum number of genotypes to consider at any site</b><br />
			If there are more than this number of genotypes at a locus presented to the genotyper, then only this many
 genotypes will be used. This is intended to deal with sites where the combination of high ploidy and high alt
 allele count can lead to an explosion in the number of possible genotypes, with extreme adverse effects on
 runtime performance.

 How does it work? The possible genotypes are simply different ways of partitioning alleles given a specific
 ploidy assumption. Therefore, we remove genotypes from consideration by removing alternate alleles that are the
 least well supported. The estimate of allele support is based on the ranking of the candidate haplotypes coming
 out of the graph building step. Note however that the reference allele is always kept.

 The maximum number of alternative alleles used in the genotyping step will be the lesser of the two:
 1. the largest number of alt alleles, given ploidy, that yields a genotype count no higher than {@link #MAX_GENOTYPE_COUNT}
 2. the value of {@link #MAX_ALTERNATE_ALLELES}

 As noted above, genotyping sites with large genotype counts is both CPU and memory intensive. Unless you have
 a good reason to change the default value, we highly recommend that you not play around with this parameter.

 See also {@link #MAX_ALTERNATE_ALLELES}.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">1024</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max_num_PL_values">--max_num_PL_values </a>
			 / <small>-maxNumPLValues</small>
		</h3>
		<p class="args">
			<b>Maximum number of PL values to output</b><br />
			Determines the maximum number of PL values that will be logged in the output.  If the number of genotypes
 (which is determined by the ploidy and the number of alleles) exceeds the value provided by this argument,
 then output of all of the PL values will be suppressed.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">100</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--out">--out </a>
			 / <small>-o</small>
		</h3>
		<p class="args">
			<b>File to which variants should be written</b><br />
			
		</p>
		<p>
			<span class="label label-info ">VariantContextWriter</span>
				&nbsp;<span class="label">stdout</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--sample_ploidy">--sample_ploidy </a>
			 / <small>-ploidy</small>
		</h3>
		<p class="args">
			<b>Ploidy per sample. For pooled data, set to (Number of samples in each pool * Sample Ploidy).</b><br />
			Sample ploidy - equivalent to number of chromosome copies per pool. For pooled experiments this should be set to
   the number of samples in pool multiplied by individual sample ploidy.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">2</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--standard_min_confidence_threshold_for_calling">--standard_min_confidence_threshold_for_calling </a>
			 / <small>-stand_call_conf</small>
		</h3>
		<p class="args">
			<b>The minimum phred-scaled confidence threshold at which variants should be called</b><br />
			The minimum phred-scaled Qscore threshold to separate high confidence from low confidence calls. Only genotypes with
 confidence >= this threshold are emitted as called sites. A reasonable threshold is 30 for high-pass calling (this
 is the default).
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">10.0</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--useNewAFCalculator">--useNewAFCalculator </a>
			 / <small>-newQual</small>
		</h3>
		<p class="args">
			<b>Use new AF model instead of the so-called exact model</b><br />
			This activates a model for calculating QUAL that was introduced in version 3.7 (November 2016). We expect this
 model will become the default in future versions.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--variant">--variant </a>
			 / <small>-V</small>
		</h3>
		<p class="args">
			<b>One or more input gVCF files</b><br />
			The gVCF files to merge together
		</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">List[RodBindingCollection[VariantContext]]</span>
				&nbsp;<span class="label">NA</span>
		</p>

        <hr>


	</div>

</div>

