## ReduceReads format specifications - RETIRED

By Carneiro

<h3>Note that the ReduceReads algorithm has been deprecated and should no longer be used. This article is only made available for archival purposes.</h3>

<hr></hr><h3>What is a <em>synthetic read</em>?</h3>

<p>When running reduce reads, the algorithm will find regions of low variation in the genome and compress them together. To represent this compressed region, we use a <em>synthetic read</em> that carries all the information necessary to downstream tools to perform likelihood calculations over the reduced data.</p>

<p>They are called Synthetic because they are not read by a sequencer, these reads are automatically generated by the GATK and can be extremely long. In a synthetic read, each base will represent the <em>consensus base</em> for that genomic location. Each base will have it's <em>consensus quality score</em> represented in the equivalent offset in the quality score string.</p>

<h3>Consensus Bases</h3>

<p>ReduceReads has several filtering parameters for consensus regions. Consensus is created based on base qualities, mapping qualities and other adjustable parameters from the command line. All filters are described in the <a rel="nofollow" href="http://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_sting_gatk_walkers_compression_reducereads_ReduceReads.html">technical documentation of reduce reads</a>.</p>

<h3>Consensus Quality Scores</h3>

<p>The consensus quality score of a consensus base is essentially the mean of all bases that passed all the filters and represent an observation of that base. It is represented in the quality score field of the SAM format.</p>

<p><img src="https://us.v-cdn.net/5019796/uploads/FileUpload/12/0389e56946fcbf87bed68f40e69ba4.png" alt="image" class="embedImage-img importedEmbed-img"></img></p>

<p>n is the number of bases that contributed to the consensus base and q_i is the corresponding quality score of each base.</p>

<p>Insertion quality scores and Deletion quality scores (generated by BQSR) will undergo the same process and will be represented the same way.</p>

<h3>Mapping Quality</h3>

<p>The mapping quality of a synthetic read is a value representative of the mapping qualities of all the reads that contributed to it. This is an average of the root mean square of the mapping quality of all reads that contributed to the bases of the synthetic read. It is represented in the mapping quality score field of the SAM format.</p>

<p><img src="https://us.v-cdn.net/5019796/uploads/FileUpload/16/7db81c90bec293f5be83952b74193c.jpg" alt="image" class="embedImage-img importedEmbed-img"></img></p>

<p>where n is the number of reads and x_i is the mapping quality of each read.</p>

<h3>Original Alignments</h3>

<p>A synthetic read may come with up to two extra tags representing its original alignment information. Due to many filters in ReduceReads, reads are hard-clipped to the are of interest. These hard-clips are always represented in the cigar string with the H element and the length of the clipping <strong>in genomic coordinates</strong>. Sometimes hard clipping will make it impossible to retrieve what was the original alignment start / end of a read. In those cases, the read will contain extra tags with integer values representing their original alignment start or end.</p>

<p>Here are the two integer tags:</p>

<ul><li>OP -- original alignment start</li>
<li>OE -- original alignment end</li>
</ul><p>For all other reads, where this can still be obtained through the cigar string (i.e. using getAlignmentStart() or getUnclippedStart()), these tags are not created.</p>

<h3>The RR Tag</h3>

<p>the RR tag is a tag that holds the observed depth (after filters) of every base that contributed to a reduce read. That means all bases that passed the mapping and base quality filters, and had the same observation as the one in the reduced read.</p>

<p>The RR tag carries an array of bytes and for increased compression, it works like this: <br>
the first number represents the depth of the first base in the reduced read.<br>
all subsequent numbers will represent the offset depth from the first base. <br>
Therefore, to calculate the depth of base "i" using the RR array, one must use :</p>

<p><code class="code codeInline" spellcheck="false">RR[0] + RR[i]</code></p>

<p>but make sure i &gt; 0. Here is the code we use to return the depth of the i'th base:</p>

<p><code class="code codeInline" spellcheck="false">return (i==0) ? firstCount : (byte) Math.min(firstCount + offsetCount, Byte.MAX_VALUE);</code></p>

<hr></hr><h3>Using Synthetic Reads with GATK tools</h3>

<p>The GATK is 100% compatible with synthetic reads. You can use Reduced BAM files in combination with non-reduced BAM files in any GATK analysis tools and it will work seamlessly.</p>

<p>Programming in the GATK</p>

<p>If you are programming using the GATK framework, the GATKSAMRecord class carries all the necessary functionality to use synthetic reads transparently with methods like:</p>

<ul><li>public final byte getReducedCount(final int i)</li>
<li>public int getOriginalAlignmentStart()</li>
<li>public int getOriginalAlignmentEnd()</li>
<li>public boolean isReducedRead()</li>
</ul>