

<div class='row' id="top">

	

	

	<section class="col-md-4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_GatherBqsrReports.html">GatherBqsrReports</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_examples_GATKPaperGenotyper.html">GATKPaperGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationTools">Variant Evaluation Tools</a>
							<div id="VariantEvaluationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulationTools">Variant Manipulation Tools</a>
							<div id="VariantManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#AnnotationModules">Annotation Modules</a>
							<div id="AnnotationModules"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InsertSizeRankSum.html">AS_InsertSizeRankSum</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MQMateRankSumTest.html">AS_MQMateRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_ClusteredReadPosition.html">ClusteredReadPosition</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOriginalMQAfterIndelRealignmentFilter.html">ReassignOriginalMQAfterIndelRealignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ResourceFileCodecs">Resource File Codecs</a>
							<div id="ResourceFileCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
		</aside>
		

	</section>

	<div class="col-md-8">

		<h1>BaseRecalibrator</h1>

		<p class="lead">Detect systematic errors in base quality scores</p>

			<h3>Category
				<small> Sequence Data Processing Tools</small>
			</h3>
			<h3>Traversal
				<small>ReadWalker</small>
			</h3>
			<h3>PartitionBy
				<small>READ</small>
			</h3>

		<hr>
		<h2>Overview</h2>
		

 <p>
 Variant calling algorithms rely heavily on the quality scores assigned to the individual base calls in each sequence
 read. These scores are per-base estimates of error emitted by the sequencing machines. Unfortunately the scores
 produced by the machines are subject to various sources of systematic technical error, leading to over- or
 under-estimated base quality scores in the data. Base quality score recalibration (BQSR) is a process in which we
 apply machine learning to model these errors empirically and adjust the quality scores accordingly. This allows us
 to get more accurate base qualities, which in turn improves the accuracy of our variant calls.

 The base recalibration process involves two key steps: first the program builds a model of covariation based on the
 data and a set of known variants (which you can bootstrap if there is none available for your organism), then it
 adjusts the base quality scores in the data based on the model.

 There is an optional but highly recommended step that involves building a second model and generating before/after
 plots to visualize the effects of the recalibration process. This is useful for quality control purposes.

 This tool performs the first step described above: it builds the model of covariation and produces the recalibration
 table. It operates only at sites that are not in dbSNP; we assume that all reference mismatches we see are therefore
 errors and indicative of poor base quality. This tool generates tables based on various user-specified covariates
 (such as read group, reported quality score, cycle, and context). Assuming we are working with a large amount of data,
 we can then calculate an empirical probability of error given the particular covariates seen at this site,
 where p(error) = num mismatches / num observations.

 The output file is a table (of the several covariate values, number of observations, number of mismatches, empirical
 quality score).
 </p>

 <h3>Inputs</h3>
 <p>
 A BAM file containing data that needs to be recalibrated.
 <p>
 A database of known polymorphic sites to mask out.
 </p>

 <h3>Output</h3>
 <p>A GATKReport file with many tables:</p>
 <ul>
     <li>The list of arguments</li>
     <li>The quantized qualities table</li>
     <li>The recalibration table by read group</li>
     <li>The recalibration table by quality score</li>
     <li>The recalibration table for all the optional covariates</li>
 </ul>
<p>
 The GATKReport table format is intended to be easy to read by both humans and computer languages (especially R).
 Check out the documentation of the GATKReport (in the FAQs) to learn how to manipulate this table.
 </p>

 <h3>Usage example</h3>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T BaseRecalibrator \
   -R reference.fasta \
   -I my_reads.bam \
   -knownSites latest_dbsnp.vcf \
   -o recal_data.table
 </pre>

 <h3>Notes</h3>
 <ul><li>This *base* recalibration process should not be confused with *variant* recalibration, which is a s
 ophisticated filtering technique applied on the variant callset produced in a later step of the analysis workflow.</li>
 <li>ReadGroupCovariate and QualityScoreCovariate are required covariates and will be added regardless of whether
 or not they were specified.</li></ul>

				<hr>
				<h2>Additional Information</h2>
				<p></p>
				<h3>Read filters</h3>
					<p>These Read Filters are automatically applied to the data by the Engine before processing by BaseRecalibrator.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a></li>
				</ul>
				<h3>Parallelism options</h3>
					<p>This tool can be run in multi-threaded mode using this option.</p>
				<ul>
						<li><a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/org_broadinstitute_gatk_engine_CommandLineGATK.html#-nct">NanoSchedulable (-nct)</a></li>
				</ul>
				<h3>Downsampling settings</h3>
					<p>This tool does not apply any downsampling by default.</p>
				<hr>
				<h2>Command-line Arguments</h2>
				<p></p>
				<h3>Engine arguments</h3>
				<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
					used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
					engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
					certain read filters to exclude some of the data from the analysis.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
				</ul>


					<h3>BaseRecalibrator specific arguments</h3>
					<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
					<table class="table table-striped table-bordered table-condensed">
						<thead>
						<tr>
							<th>Argument name(s)</th>
							<th>Default value</th>
							<th>Summary</th>
						</tr>
						</thead>
						<tbody>
			<tr>
				<th colspan="4" id="row-divider">Required Outputs</th>
			</tr>
				<tr>
					<td><a href="#--out">--out</a><br />
								&nbsp;<em>-o</em>
					</td>
					<!--<td>File</td> -->
					<td>NA</td>
					<td>The output recalibration table file to create</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Inputs</th>
			</tr>
				<tr>
					<td><a href="#--knownSites">--knownSites</a><br />
					</td>
					<!--<td>List[RodBinding[Feature]]</td> -->
					<td>[]</td>
					<td>A database of known polymorphic sites</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Parameters</th>
			</tr>
				<tr>
					<td><a href="#--covariate">--covariate</a><br />
								&nbsp;<em>-cov</em>
					</td>
					<!--<td>String[]</td> -->
					<td>NA</td>
					<td>One or more covariates to be used in the recalibration. Can be specified multiple times</td>
				</tr>
				<tr>
					<td><a href="#--indels_context_size">--indels_context_size</a><br />
								&nbsp;<em>-ics</em>
					</td>
					<!--<td>int</td> -->
					<td>3</td>
					<td>Size of the k-mer context to be used for base insertions and deletions</td>
				</tr>
				<tr>
					<td><a href="#--maximum_cycle_value">--maximum_cycle_value</a><br />
								&nbsp;<em>-maxCycle</em>
					</td>
					<!--<td>int</td> -->
					<td>500</td>
					<td>The maximum cycle value permitted for the Cycle covariate</td>
				</tr>
				<tr>
					<td><a href="#--mismatches_context_size">--mismatches_context_size</a><br />
								&nbsp;<em>-mcs</em>
					</td>
					<!--<td>int</td> -->
					<td>2</td>
					<td>Size of the k-mer context to be used for base mismatches</td>
				</tr>
				<tr>
					<td><a href="#--solid_nocall_strategy">--solid_nocall_strategy</a><br />
					</td>
					<!--<td>SOLID_NOCALL_STRATEGY</td> -->
					<td>THROW_EXCEPTION</td>
					<td>Defines the behavior of the recalibrator when it encounters no calls in the color space. Options = THROW_EXCEPTION, LEAVE_READ_UNRECALIBRATED, or PURGE_READ</td>
				</tr>
				<tr>
					<td><a href="#--solid_recal_mode">--solid_recal_mode</a><br />
								&nbsp;<em>-sMode</em>
					</td>
					<!--<td>SOLID_RECAL_MODE</td> -->
					<td>SET_Q_ZERO</td>
					<td>How should we recalibrate solid bases in which the reference was inserted? Options = DO_NOTHING, SET_Q_ZERO, SET_Q_ZERO_BASE_N, or REMOVE_REF_BIAS</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Flags</th>
			</tr>
				<tr>
					<td><a href="#--list">--list</a><br />
								&nbsp;<em>-ls</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>List the available covariates and exit</td>
				</tr>
				<tr>
					<td><a href="#--lowMemoryMode">--lowMemoryMode</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Reduce memory usage in multi-threaded code at the expense of threading efficiency</td>
				</tr>
				<tr>
					<td><a href="#--no_standard_covs">--no_standard_covs</a><br />
								&nbsp;<em>-noStandard</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Do not use the standard set of covariates, but rather just the ones listed using the -cov argument</td>
				</tr>
				<tr>
					<td><a href="#--sort_by_all_columns">--sort_by_all_columns</a><br />
								&nbsp;<em>-sortAllCols</em>
					</td>
					<!--<td>Boolean</td> -->
					<td>false</td>
					<td>Sort the rows in the tables of reports</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Parameters</th>
			</tr>
				<tr>
					<td><a href="#--binary_tag_name">--binary_tag_name</a><br />
								&nbsp;<em>-bintag</em>
					</td>
					<!--<td>String</td> -->
					<td>NA</td>
					<td>the binary tag covariate name if using it</td>
				</tr>
				<tr>
					<td><a href="#--bqsrBAQGapOpenPenalty">--bqsrBAQGapOpenPenalty</a><br />
								&nbsp;<em>-bqsrBAQGOP</em>
					</td>
					<!--<td>double</td> -->
					<td>40.0</td>
					<td>BQSR BAQ gap open penalty (Phred Scaled).  Default value is 40.  30 is perhaps better for whole genome call sets</td>
				</tr>
				<tr>
					<td><a href="#--deletions_default_quality">--deletions_default_quality</a><br />
								&nbsp;<em>-ddq</em>
					</td>
					<!--<td>byte</td> -->
					<td>45</td>
					<td>default quality for the base deletions covariate</td>
				</tr>
				<tr>
					<td><a href="#--insertions_default_quality">--insertions_default_quality</a><br />
								&nbsp;<em>-idq</em>
					</td>
					<!--<td>byte</td> -->
					<td>45</td>
					<td>default quality for the base insertions covariate</td>
				</tr>
				<tr>
					<td><a href="#--low_quality_tail">--low_quality_tail</a><br />
								&nbsp;<em>-lqt</em>
					</td>
					<!--<td>byte</td> -->
					<td>2</td>
					<td>minimum quality for the bases in the tail of the reads to be considered</td>
				</tr>
				<tr>
					<td><a href="#--mismatches_default_quality">--mismatches_default_quality</a><br />
								&nbsp;<em>-mdq</em>
					</td>
					<!--<td>byte</td> -->
					<td>-1</td>
					<td>default quality for the base mismatches covariate</td>
				</tr>
				<tr>
					<td><a href="#--quantizing_levels">--quantizing_levels</a><br />
								&nbsp;<em>-ql</em>
					</td>
					<!--<td>int</td> -->
					<td>16</td>
					<td>number of distinct quality scores in the quantized output</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Flags</th>
			</tr>
				<tr>
					<td><a href="#--run_without_dbsnp_potentially_ruining_quality">--run_without_dbsnp_potentially_ruining_quality</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If specified, allows the recalibrator to be used without a dbsnp rod. Very unsafe and for expert users only.</td>
				</tr>
						</tbody>
					</table>

							<h3>Argument details</h3>
							<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--binary_tag_name">--binary_tag_name </a>
			 / <small>-bintag</small>
		</h3>
		<p class="args">
			<b>the binary tag covariate name if using it</b><br />
			The tag name for the binary tag covariate (if using it)
		</p>
		<p>
			<span class="label label-info ">String</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--bqsrBAQGapOpenPenalty">--bqsrBAQGapOpenPenalty </a>
			 / <small>-bqsrBAQGOP</small>
		</h3>
		<p class="args">
			<b>BQSR BAQ gap open penalty (Phred Scaled).  Default value is 40.  30 is perhaps better for whole genome call sets</b><br />
			
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">40.0</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--covariate">--covariate </a>
			 / <small>-cov</small>
		</h3>
		<p class="args">
			<b>One or more covariates to be used in the recalibration. Can be specified multiple times</b><br />
			Note that the ReadGroup and QualityScore covariates are required and do not need to be specified.
 Also, unless --no_standard_covs is specified, the Cycle and Context covariates are standard and are included by default.
 Use the --list argument to see the available covariates.
		</p>
		<p>
			<span class="label label-info ">String[]</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--deletions_default_quality">--deletions_default_quality </a>
			 / <small>-ddq</small>
		</h3>
		<p class="args">
			<b>default quality for the base deletions covariate</b><br />
			A default base qualities to use as a prior (reported quality) in the mismatch covariate model. This value will replace all base qualities in the read for this default value. Negative value turns it off. [default is on]
		</p>
		<p>
			<span class="label label-info ">byte</span>
				&nbsp;<span class="label">45</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--indels_context_size">--indels_context_size </a>
			 / <small>-ics</small>
		</h3>
		<p class="args">
			<b>Size of the k-mer context to be used for base insertions and deletions</b><br />
			The context covariate will use a context of this size to calculate its covariate value for base insertions and deletions. Must be between 1 and 13 (inclusive). Note that higher values will increase runtime and required java heap size.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">3</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--insertions_default_quality">--insertions_default_quality </a>
			 / <small>-idq</small>
		</h3>
		<p class="args">
			<b>default quality for the base insertions covariate</b><br />
			A default base qualities to use as a prior (reported quality) in the insertion covariate model. This parameter is used for all reads without insertion quality scores for each base. [default is on]
		</p>
		<p>
			<span class="label label-info ">byte</span>
				&nbsp;<span class="label">45</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--knownSites">--knownSites </a>
			 / <small>-knownSites</small>
		</h3>
		<p class="args">
			<b>A database of known polymorphic sites</b><br />
			This algorithm treats every reference mismatch as an indication of error. However, real genetic variation is expected to mismatch the reference,
 so it is critical that a database of known polymorphic sites (e.g. dbSNP) is given to the tool in order to mask out those sites.
		</p>
			<p>This argument supports reference-ordered data (ROD) files in the following formats: BCF2, BEAGLE, BED, BEDTABLE, EXAMPLEBINARY, RAWHAPMAP, REFSEQ, SAMPILEUP, SAMREAD, TABLE, VCF, VCF3</p>
		<p>
			<span class="label label-info ">List[RodBinding[Feature]]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--list">--list </a>
			 / <small>-ls</small>
		</h3>
		<p class="args">
			<b>List the available covariates and exit</b><br />
			Note that the --list argument requires a fully resolved and correct command-line to work.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--low_quality_tail">--low_quality_tail </a>
			 / <small>-lqt</small>
		</h3>
		<p class="args">
			<b>minimum quality for the bases in the tail of the reads to be considered</b><br />
			Reads with low quality bases on either tail (beginning or end) will not be considered in the context. This parameter defines the quality below which (inclusive) a tail is considered low quality
		</p>
		<p>
			<span class="label label-info ">byte</span>
				&nbsp;<span class="label">2</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--lowMemoryMode">--lowMemoryMode </a>
			 / <small>-lowMemoryMode</small>
		</h3>
		<p class="args">
			<b>Reduce memory usage in multi-threaded code at the expense of threading efficiency</b><br />
			When you use nct > 1, BQSR uses nct times more memory to compute its recalibration tables, for efficiency
 purposes.  If you have many covariates, and therefore are using a lot of memory, you can use this flag
 to safely access only one table.  There may be some CPU cost, but as long as the table is really big
 the cost should be relatively reasonable.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--maximum_cycle_value">--maximum_cycle_value </a>
			 / <small>-maxCycle</small>
		</h3>
		<p class="args">
			<b>The maximum cycle value permitted for the Cycle covariate</b><br />
			The cycle covariate will generate an error if it encounters a cycle greater than this value.
 This argument is ignored if the Cycle covariate is not used.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">500</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--mismatches_context_size">--mismatches_context_size </a>
			 / <small>-mcs</small>
		</h3>
		<p class="args">
			<b>Size of the k-mer context to be used for base mismatches</b><br />
			The context covariate will use a context of this size to calculate its covariate value for base mismatches. Must be between 1 and 13 (inclusive). Note that higher values will increase runtime and required java heap size.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">2</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--mismatches_default_quality">--mismatches_default_quality </a>
			 / <small>-mdq</small>
		</h3>
		<p class="args">
			<b>default quality for the base mismatches covariate</b><br />
			A default base qualities to use as a prior (reported quality) in the mismatch covariate model. This value will replace all base qualities in the read for this default value. Negative value turns it off. [default is off]
		</p>
		<p>
			<span class="label label-info ">byte</span>
				&nbsp;<span class="label">-1</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--no_standard_covs">--no_standard_covs </a>
			 / <small>-noStandard</small>
		</h3>
		<p class="args">
			<b>Do not use the standard set of covariates, but rather just the ones listed using the -cov argument</b><br />
			The Cycle and Context covariates are standard and are included by default unless this argument is provided.
 Note that the ReadGroup and QualityScore covariates are required and cannot be excluded.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--out">--out </a>
			 / <small>-o</small>
		</h3>
		<p class="args">
			<b>The output recalibration table file to create</b><br />
			After the header, data records occur one per line until the end of the file. The first several items on a line are the
 values of the individual covariates and will change depending on which covariates were specified at runtime. The last
 three items are the data- that is, number of observations for this combination of covariates, number of reference mismatches,
 and the raw empirical quality score calculated by phred-scaling the mismatch rate.
		</p>
		<p>
					<span class="badge badge-important">R</span>
			<span class="label label-info ">File</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--quantizing_levels">--quantizing_levels </a>
			 / <small>-ql</small>
		</h3>
		<p class="args">
			<b>number of distinct quality scores in the quantized output</b><br />
			BQSR generates a quantization table for quick quantization later by subsequent tools. BQSR does not quantize the base qualities, this is done by the engine with the -qq or -BQSR options.
 This parameter tells BQSR the number of levels of quantization to use to build the quantization table.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">16</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--run_without_dbsnp_potentially_ruining_quality">--run_without_dbsnp_potentially_ruining_quality </a>
			 / <small>-run_without_dbsnp_potentially_ruining_quality</small>
		</h3>
		<p class="args">
			<b>If specified, allows the recalibrator to be used without a dbsnp rod. Very unsafe and for expert users only.</b><br />
			This calculation is critically dependent on being able to skip over known polymorphic sites. Please be sure that you know what you are doing if you use this option.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--solid_nocall_strategy">--solid_nocall_strategy </a>
			 / <small>-solid_nocall_strategy</small>
		</h3>
		<p class="args">
			<b>Defines the behavior of the recalibrator when it encounters no calls in the color space. Options = THROW_EXCEPTION, LEAVE_READ_UNRECALIBRATED, or PURGE_READ</b><br />
			BaseRecalibrator accepts a --solid_nocall_strategy <MODE> flag which governs how the recalibrator handles
 no calls in the color space tag. Unfortunately because of the reference inserted bases mentioned above, reads with no calls in
 their color space tag can not be recalibrated.
		</p>
			<p>
			The --solid_nocall_strategy argument is an enumerated type (SOLID_NOCALL_STRATEGY), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">THROW_EXCEPTION</dt>
					<dd class="enum">When a no call is detected throw an exception to alert the user that recalibrating this SOLiD data is unsafe. This is the default option.</dd>
					<dt class="enum">LEAVE_READ_UNRECALIBRATED</dt>
					<dd class="enum">Leave the read in the output bam completely untouched. This mode is only okay if the no calls are very rare.</dd>
					<dt class="enum">PURGE_READ</dt>
					<dd class="enum">Mark these reads as failing vendor quality checks so they can be filtered out by downstream analyses.</dd>
			</dl>
			</p>
		<p>
			<span class="label label-info ">SOLID_NOCALL_STRATEGY</span>
				&nbsp;<span class="label">THROW_EXCEPTION</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--solid_recal_mode">--solid_recal_mode </a>
			 / <small>-sMode</small>
		</h3>
		<p class="args">
			<b>How should we recalibrate solid bases in which the reference was inserted? Options = DO_NOTHING, SET_Q_ZERO, SET_Q_ZERO_BASE_N, or REMOVE_REF_BIAS</b><br />
			BaseRecalibrator accepts a --solid_recal_mode <MODE> flag which governs how the recalibrator handles the
 reads which have had the reference inserted because of color space inconsistencies.
		</p>
			<p>
			The --solid_recal_mode argument is an enumerated type (SOLID_RECAL_MODE), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">DO_NOTHING</dt>
					<dd class="enum">Treat reference inserted bases as reference matching bases. Very unsafe!</dd>
					<dt class="enum">SET_Q_ZERO</dt>
					<dd class="enum">Set reference inserted bases and the previous base (because of color space alignment details) to Q0. This is the default option.</dd>
					<dt class="enum">SET_Q_ZERO_BASE_N</dt>
					<dd class="enum">In addition to setting the quality scores to zero, also set the base itself to 'N'. This is useful to visualize in IGV.</dd>
					<dt class="enum">REMOVE_REF_BIAS</dt>
					<dd class="enum">Look at the color quality scores and probabilistically decide to change the reference inserted base to be the base which is implied by the original color space instead of the reference.</dd>
			</dl>
			</p>
		<p>
			<span class="label label-info ">SOLID_RECAL_MODE</span>
				&nbsp;<span class="label">SET_Q_ZERO</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--sort_by_all_columns">--sort_by_all_columns </a>
			 / <small>-sortAllCols</small>
		</h3>
		<p class="args">
			<b>Sort the rows in the tables of reports</b><br />
			Whether GATK report tables should have rows in sorted order, starting from leftmost column
		</p>
		<p>
			<span class="label label-info ">Boolean</span>
				&nbsp;<span class="label">false</span>
		</p>

        <hr>


        

	</div>

</div>

