

<div class='row' id="top">

	

	

	<section class="col-md-4">
		<aside class="well">
			<a href="index.html"><h4><i class='fa fa-chevron-left'></i> Back to Tool Docs Index</h4></a>
		</aside>
		<aside class="well">
			<h2>Categories</h2>

        <ul class="nav nav-pills nav-stacked" id="sidenav">
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#EngineParametersavailabletoalltools">Engine Parameters (available to all tools)</a>
							<div id="EngineParametersavailabletoalltools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#DiagnosticsandQualityControlTools">Diagnostics and Quality Control Tools</a>
							<div id="DiagnosticsandQualityControlTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_ASEReadCounter.html">ASEReadCounter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_AnalyzeCovariates.html">AnalyzeCovariates</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CallableLoci.html">CallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CheckPileup.html">CheckPileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_CompareCallableLoci.html">CompareCallableLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_contamination_ContEst.html">ContEst</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountBases.html">CountBases</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountIntervals.html">CountIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountLoci.html">CountLoci</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountMales.html">CountMales</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODs.html">CountRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountRODsByRef.html">CountRODsByRef</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReadEvents.html">CountReadEvents</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountReads.html">CountReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_CountTerminusEvent.html">CountTerminusEvent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.html">DepthOfCoverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_diagnosetargets_DiagnoseTargets.html">DiagnoseTargets</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diffengine_DiffObjects.html">DiffObjects</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ErrorRatePerCycle.html">ErrorRatePerCycle</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaStats.html">FastaStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_FindCoveredIntervals.html">FindCoveredIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_FlagStat.html">FlagStat</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_coverage_GCContentByInterval.html">GCContentByInterval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_GatherBqsrReports.html">GatherBqsrReports</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_Pileup.html">Pileup</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_PrintRODs.html">PrintRODs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_missing_QualifyMissingIntervals.html">QualifyMissingIntervals</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_ReadClippingStats.html">ReadClippingStats</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadGroupProperties.html">ReadGroupProperties</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_diagnostics_ReadLengthDistribution.html">ReadLengthDistribution</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_simulatereads_SimulateReadsForVariants.html">SimulateReadsForVariants</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#SequenceDataProcessingTools">Sequence Data Processing Tools</a>
							<div id="SequenceDataProcessingTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_bqsr_BaseRecalibrator.html">BaseRecalibrator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_ClipReads.html">ClipReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_IndelRealigner.html">IndelRealigner</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_LeftAlignIndels.html">LeftAlignIndels</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_PrintReads.html">PrintReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.html">RealignerTargetCreator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_rnaseq_SplitNCigarReads.html">SplitNCigarReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_readutils_SplitSamFile.html">SplitSamFile</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantDiscoveryTools">Variant Discovery Tools</a>
							<div id="VariantDiscoveryTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_ApplyRecalibration.html">ApplyRecalibration</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CalculateGenotypePosteriors.html">CalculateGenotypePosteriors</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_examples_GATKPaperGenotyper.html">GATKPaperGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeGVCFs.html">GenotypeGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.html">HaplotypeCaller</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.html">MuTect2</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RegenotypeVariants.html">RegenotypeVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.html">VariantRecalibrator</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantEvaluationTools">Variant Evaluation Tools</a>
							<div id="VariantEvaluationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_GenotypeConcordance.html">GenotypeConcordance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_ValidateVariants.html">ValidateVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_varianteval_VariantEval.html">VariantEval</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_filters_VariantFiltration.html">VariantFiltration</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#VariantManipulationTools">Variant Manipulation Tools</a>
							<div id="VariantManipulationTools"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_CatVariants.html">CatVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineGVCFs.html">CombineGVCFs</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_CombineVariants.html">CombineVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeResolver.html">HaplotypeResolver</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_LeftAlignAndTrimVariants.html">LeftAlignAndTrimVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_PhaseByTransmission.html">PhaseByTransmission</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_RandomlySplitVariants.html">RandomlySplitVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_phasing_ReadBackedPhasing.html">ReadBackedPhasing</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectHeaders.html">SelectHeaders</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.html">SelectVariants</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_validation_validationsiteselector_ValidationSiteSelector.html">ValidationSiteSelector</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantAnnotator.html">VariantAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToAllelicPrimitives.html">VariantsToAllelicPrimitives</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToBinaryPed.html">VariantsToBinaryPed</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToTable.html">VariantsToTable</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_variantutils_VariantsToVCF.html">VariantsToVCF</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#AnnotationModules">Annotation Modules</a>
							<div id="AnnotationModules"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_BaseQualityRankSumTest.html">AS_BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_FisherStrand.html">AS_FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InbreedingCoeff.html">AS_InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_InsertSizeRankSum.html">AS_InsertSizeRankSum</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MQMateRankSumTest.html">AS_MQMateRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_MappingQualityRankSumTest.html">AS_MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_QualByDepth.html">AS_QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_RMSMappingQuality.html">AS_RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_ReadPosRankSumTest.html">AS_ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AS_StrandOddsRatio.html">AS_StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalance.html">AlleleBalance</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleBalanceBySample.html">AlleleBalanceBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_AlleleCountBySample.html">AlleleCountBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCounts.html">BaseCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseCountsBySample.html">BaseCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_BaseQualityRankSumTest.html">BaseQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_BaseQualitySumPerAlleleBySample.html">BaseQualitySumPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ChromosomeCounts.html">ChromosomeCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ClippingRankSumTest.html">ClippingRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_ClusteredReadPosition.html">ClusteredReadPosition</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_Coverage.html">Coverage</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerAlleleBySample.html">DepthPerAlleleBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_DepthPerSampleHC.html">DepthPerSampleHC</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ExcessHet.html">ExcessHet</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.html">FisherStrand</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_FractionInformativeReads.html">FractionInformativeReads</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GCContent.html">GCContent</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_GenotypeSummaries.html">GenotypeSummaries</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HaplotypeScore.html">HaplotypeScore</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HardyWeinberg.html">HardyWeinberg</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_HomopolymerRun.html">HomopolymerRun</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_InbreedingCoeff.html">InbreedingCoeff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LikelihoodRankSumTest.html">LikelihoodRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_LowMQ.html">LowMQ</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MVLikelihoodRatio.html">MVLikelihoodRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityRankSumTest.html">MappingQualityRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZero.html">MappingQualityZero</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_MappingQualityZeroBySample.html">MappingQualityZeroBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_NBaseCount.html">NBaseCount</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_cancer_OxoGReadCounts.html">OxoGReadCounts</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_PossibleDeNovo.html">PossibleDeNovo</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_QualByDepth.html">QualByDepth</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_RMSMappingQuality.html">RMSMappingQuality</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_ReadPosRankSumTest.html">ReadPosRankSumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SampleList.html">SampleList</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SnpEff.html">SnpEff</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_SpanningDeletions.html">SpanningDeletions</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.html">StrandAlleleCountsBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.html">StrandBiasBySample</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.html">StrandOddsRatio</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TandemRepeatAnnotator.html">TandemRepeatAnnotator</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_TransmissionDisequilibriumTest.html">TransmissionDisequilibriumTest</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_annotator_VariantType.html">VariantType</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReadFilters">Read Filters</a>
							<div id="ReadFilters"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_CountingFilteringIterator$CountingReadFilter.html">CountingFilteringIterator.CountingReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.html">HCMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_LibraryReadFilter.html">LibraryReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityFilter.html">MappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MappingQualityZeroFilter.html">MappingQualityZeroFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MateSameStrandFilter.html">MateSameStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MaxInsertSizeFilter.html">MaxInsertSizeFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_MissingReadGroupFilter.html">MissingReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NoOriginalQualityScoresFilter.html">NoOriginalQualityScoresFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_OverclippedReadFilter.html">OverclippedReadFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_Platform454Filter.html">Platform454Filter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformFilter.html">PlatformFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_PlatformUnitFilter.html">PlatformUnitFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadGroupBlackListFilter.html">ReadGroupBlackListFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadLengthFilter.html">ReadLengthFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadNameFilter.html">ReadNameFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReadStrandFilter.html">ReadStrandFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignMappingQualityFilter.html">ReassignMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOneMappingQualityFilter.html">ReassignOneMappingQualityFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_ReassignOriginalMQAfterIndelRealignmentFilter.html">ReassignOriginalMQAfterIndelRealignmentFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SampleFilter.html">SampleFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_SingleReadGroupFilter.html">SingleReadGroupFilter</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a>
											</li>
								</ul>
							</div>
						</li>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ResourceFileCodecs">Resource File Codecs</a>
							<div id="ResourceFileCodecs"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_beagle_BeagleCodec.html">BeagleCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_BedTableCodec.html">BedTableCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_hapmap_RawHapMapCodec.html">RawHapMapCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_refseq_RefSeqCodec.html">RefSeqCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_sampileup_SAMPileupCodec.html">SAMPileupCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_samread_SAMReadCodec.html">SAMReadCodec</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_utils_codecs_table_TableCodec.html">TableCodec</a>
											</li>
								</ul>
							</div>
						</li>
        		<hr>
						<li><a data-toggle="collapse" data-parent="#sidenav" href="#ReferenceUtilities">Reference Utilities</a>
							<div id="ReferenceUtilities"
								
								<ul>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaAlternateReferenceMaker.html">FastaAlternateReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_fasta_FastaReferenceMaker.html">FastaReferenceMaker</a>
											</li>
											<li>
												<a href="org_broadinstitute_gatk_tools_walkers_qc_QCRef.html">QCRef</a>
											</li>
								</ul>
							</div>
						</li>
        </ul>
		</aside>
		

	</section>

	<div class="col-md-8">

		<h1>UnifiedGenotyper</h1>

		<p class="lead">Call SNPs and indels on a per-locus basis</p>

			<h3>Category
				<small> Variant Discovery Tools</small>
			</h3>
			<h3>Traversal
				<small>LocusWalker</small>
			</h3>
			<h3>PartitionBy
				<small>LOCUS</small>
			</h3>

		<hr>
		<h2>Overview</h2>
		

 <p>
 This tool uses a Bayesian genotype likelihood model to estimate simultaneously the most likely genotypes and
 allele frequency in a population of N samples, emitting a genotype for each sample. The system can either emit
 just the variant sites or complete genotypes (which includes homozygous reference calls) satisfying some
 phred-scaled confidence value.
 </p>

 <h3>Input</h3>
 <p>
 The read data from which to make variant calls.
 </p>

 <h3>Output</h3>
 <p>
 A raw, unfiltered, highly sensitive callset in VCF format.
 </p>

 <h3>Usage examples</h3>
 <h4>Multi-sample SNP calling</h4>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T UnifiedGenotyper \
   -R reference.fasta \
   -I sample1.bam [-I sample2.bam ...] \
   --dbsnp dbSNP.vcf \
   -o snps.raw.vcf \
   -stand_call_conf [50.0] \
   [-L targets.interval_list]
 </pre>

 <h4>Generate calls at all sites</h4>
 <pre>
 java -jar GenomeAnalysisTK.jar \
   -T UnifiedGenotyper \
   -R reference.fasta \
   -I input.bam \
   -o raw_variants.vcf \
   --output_mode EMIT_ALL_SITES
 </pre>

 <h3>Caveats</h3>
 <ul>
 <li>The caller can be very aggressive in calling variants in order to be very sensitive, so the raw output will
 contain many false positives. We use extensive post-calling filters to eliminate most of these FPs. See the documentation on filtering (especially by Variant Quality Score Recalibration) for more details.</li>
 <li><b>This tool has been deprecated in favor of HaplotypeCaller, a much more sophisticated variant caller that
 produces much better calls, especially on indels, and includes features that allow it to scale to much larger
 cohort sizes.</b></li>
 </ul>

 <h3>Special note on ploidy</h3>
 <p>This tool is able to handle almost any ploidy (except very high ploidies in large pooled experiments); the ploidy
 can be specified using the -ploidy argument for non-diploid organisms.</p>

				<hr>
				<h2>Additional Information</h2>
				<p></p>
				<h3>Read filters</h3>
					<p>These Read Filters are automatically applied to the data by the Engine before processing by UnifiedGenotyper.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_filters_MalformedReadFilter.html">MalformedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_BadCigarFilter.html">BadCigarFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_BadMateFilter.html">BadMateFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_UnmappedReadFilter.html">UnmappedReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_NotPrimaryAlignmentFilter.html">NotPrimaryAlignmentFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_FailsVendorQualityCheckFilter.html">FailsVendorQualityCheckFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_DuplicateReadFilter.html">DuplicateReadFilter</a></li>
						<li><a href="org_broadinstitute_gatk_engine_filters_MappingQualityUnavailableFilter.html">MappingQualityUnavailableFilter</a></li>
				</ul>
				<h3>Parallelism options</h3>
					<p>This tool can be run in multi-threaded mode using these options.</p>
				<ul>
						<li><a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/org_broadinstitute_gatk_engine_CommandLineGATK.html#-nt">TreeReducible (-nt)</a></li>
						<li><a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/org_broadinstitute_gatk_engine_CommandLineGATK.html#-nct">NanoSchedulable (-nct)</a></li>
				</ul>
				<h3>Downsampling settings</h3>
					<p>This tool applies the following downsampling settings by default.</p>
					<ul>
						<li>Mode: <b>BY_SAMPLE</b></li>
						<li>To coverage: <b>250</b></li>
					</ul>
				<h3>Window size</h3>
				<p>This tool uses a sliding window on the reference.</p>
				<ul>
					<li>Window start: <b>-200</b> bp before the locus</li>
					<li>Window stop: <b>200</b> bp after the locus</li>
				</ul>
				<hr>
				<h2>Command-line Arguments</h2>
				<p></p>
				<h3>Engine arguments</h3>
				<p>All tools inherit arguments from the GATK Engine' "CommandLineGATK" argument collection, which can be
					used to modify various aspects of the tool's function. For example, the -L argument directs the GATK
					engine to restrict processing to specific genomic intervals; or the -rf argument allows you to apply
					certain read filters to exclude some of the data from the analysis.</p>
				<ul>
						<li><a href="org_broadinstitute_gatk_engine_CommandLineGATK.html">CommandLineGATK</a></li>
				</ul>


					<h3>UnifiedGenotyper specific arguments</h3>
					<p>This table summarizes the command-line arguments that are specific to this tool. For more details on each argument, see the list further down below the table or click on an argument name to jump directly to that entry in the list.</p>
					<table class="table table-striped table-bordered table-condensed">
						<thead>
						<tr>
							<th>Argument name(s)</th>
							<th>Default value</th>
							<th>Summary</th>
						</tr>
						</thead>
						<tbody>
			<tr>
				<th colspan="4" id="row-divider">Optional Inputs</th>
			</tr>
				<tr>
					<td><a href="#--alleles">--alleles</a><br />
					</td>
					<!--<td>RodBinding[VariantContext]</td> -->
					<td>none</td>
					<td>Set of alleles to use in genotyping</td>
				</tr>
				<tr>
					<td><a href="#--comp">--comp</a><br />
					</td>
					<!--<td>List[RodBinding[VariantContext]]</td> -->
					<td>[]</td>
					<td>Comparison VCF file</td>
				</tr>
				<tr>
					<td><a href="#--dbsnp">--dbsnp</a><br />
								&nbsp;<em>-D</em>
					</td>
					<!--<td>RodBinding[VariantContext]</td> -->
					<td>none</td>
					<td>dbSNP file</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Outputs</th>
			</tr>
				<tr>
					<td><a href="#--out">--out</a><br />
								&nbsp;<em>-o</em>
					</td>
					<!--<td>VariantContextWriter</td> -->
					<td>stdout</td>
					<td>File to which variants should be written</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Parameters</th>
			</tr>
				<tr>
					<td><a href="#--annotation">--annotation</a><br />
								&nbsp;<em>-A</em>
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>One or more specific annotations to apply to variant calls</td>
				</tr>
				<tr>
					<td><a href="#--contamination_fraction_to_filter">--contamination_fraction_to_filter</a><br />
								&nbsp;<em>-contamination</em>
					</td>
					<!--<td>double</td> -->
					<td>0.0</td>
					<td>Fraction of contamination to aggressively remove</td>
				</tr>
				<tr>
					<td><a href="#--excludeAnnotation">--excludeAnnotation</a><br />
								&nbsp;<em>-XA</em>
					</td>
					<!--<td>List[String]</td> -->
					<td>[]</td>
					<td>One or more specific annotations to exclude</td>
				</tr>
				<tr>
					<td><a href="#--genotype_likelihoods_model">--genotype_likelihoods_model</a><br />
								&nbsp;<em>-glm</em>
					</td>
					<!--<td>Model</td> -->
					<td>SNP</td>
					<td>Genotype likelihoods calculation model to employ -- SNP is the default option, while INDEL is also available for calling indels and BOTH is available for calling both together</td>
				</tr>
				<tr>
					<td><a href="#--genotyping_mode">--genotyping_mode</a><br />
								&nbsp;<em>-gt_mode</em>
					</td>
					<!--<td>GenotypingOutputMode</td> -->
					<td>DISCOVERY</td>
					<td>Specifies how to determine the alternate alleles to use for genotyping</td>
				</tr>
				<tr>
					<td><a href="#--group">--group</a><br />
								&nbsp;<em>-G</em>
					</td>
					<!--<td>String[]</td> -->
					<td>[Standard, StandardUG]</td>
					<td>One or more classes/groups of annotations to apply to variant calls.  The single value 'none' removes the default group</td>
				</tr>
				<tr>
					<td><a href="#--heterozygosity">--heterozygosity</a><br />
								&nbsp;<em>-hets</em>
					</td>
					<!--<td>Double</td> -->
					<td>0.001</td>
					<td>Heterozygosity value used to compute prior likelihoods for any locus</td>
				</tr>
				<tr>
					<td><a href="#--heterozygosity_stdev">--heterozygosity_stdev</a><br />
								&nbsp;<em>-heterozygosityStandardDeviation</em>
					</td>
					<!--<td>double</td> -->
					<td>0.01</td>
					<td>Standard deviation of eterozygosity for SNP and indel calling.</td>
				</tr>
				<tr>
					<td><a href="#--indel_heterozygosity">--indel_heterozygosity</a><br />
								&nbsp;<em>-indelHeterozygosity</em>
					</td>
					<!--<td>double</td> -->
					<td>1.25E-4</td>
					<td>Heterozygosity for indel calling</td>
				</tr>
				<tr>
					<td><a href="#--max_deletion_fraction">--max_deletion_fraction</a><br />
								&nbsp;<em>-deletions</em>
					</td>
					<!--<td>Double</td> -->
					<td>0.05</td>
					<td>Maximum fraction of reads with deletions spanning this locus for it to be callable</td>
				</tr>
				<tr>
					<td><a href="#--min_base_quality_score">--min_base_quality_score</a><br />
								&nbsp;<em>-mbq</em>
					</td>
					<!--<td>int</td> -->
					<td>17</td>
					<td>Minimum base quality required to consider a base for calling</td>
				</tr>
				<tr>
					<td><a href="#--min_indel_count_for_genotyping">--min_indel_count_for_genotyping</a><br />
								&nbsp;<em>-minIndelCnt</em>
					</td>
					<!--<td>int</td> -->
					<td>5</td>
					<td>Minimum number of consensus indels required to trigger genotyping run</td>
				</tr>
				<tr>
					<td><a href="#--min_indel_fraction_per_sample">--min_indel_fraction_per_sample</a><br />
								&nbsp;<em>-minIndelFrac</em>
					</td>
					<!--<td>double</td> -->
					<td>0.25</td>
					<td>Minimum fraction of all reads at a locus that must contain an indel (of any allele) for that sample to contribute to the indel count for alleles</td>
				</tr>
				<tr>
					<td><a href="#--pair_hmm_implementation">--pair_hmm_implementation</a><br />
								&nbsp;<em>-pairHMM</em>
					</td>
					<!--<td>HMM_IMPLEMENTATION</td> -->
					<td>LOGLESS_CACHING</td>
					<td>The PairHMM implementation to use for -glm INDEL genotype likelihood calculations</td>
				</tr>
				<tr>
					<td><a href="#--pcr_error_rate">--pcr_error_rate</a><br />
								&nbsp;<em>-pcr_error</em>
					</td>
					<!--<td>Double</td> -->
					<td>1.0E-4</td>
					<td>The PCR error rate to be used for computing fragment-based likelihoods</td>
				</tr>
				<tr>
					<td><a href="#--sample_ploidy">--sample_ploidy</a><br />
								&nbsp;<em>-ploidy</em>
					</td>
					<!--<td>int</td> -->
					<td>2</td>
					<td>Ploidy per sample. For pooled data, set to (Number of samples in each pool * Sample Ploidy).</td>
				</tr>
				<tr>
					<td><a href="#--standard_min_confidence_threshold_for_calling">--standard_min_confidence_threshold_for_calling</a><br />
								&nbsp;<em>-stand_call_conf</em>
					</td>
					<!--<td>double</td> -->
					<td>10.0</td>
					<td>The minimum phred-scaled confidence threshold at which variants should be called</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Optional Flags</th>
			</tr>
				<tr>
					<td><a href="#--annotateNDA">--annotateNDA</a><br />
								&nbsp;<em>-nda</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Annotate number of alleles observed</td>
				</tr>
				<tr>
					<td><a href="#--computeSLOD">--computeSLOD</a><br />
								&nbsp;<em>-slod</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>If provided, we will calculate the SLOD (SB annotation)</td>
				</tr>
				<tr>
					<td><a href="#--useNewAFCalculator">--useNewAFCalculator</a><br />
								&nbsp;<em>-newQual</em>
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Use new AF model instead of the so-called exact model</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Parameters</th>
			</tr>
				<tr>
					<td><a href="#--contamination_fraction_per_sample_file">--contamination_fraction_per_sample_file</a><br />
								&nbsp;<em>-contaminationFile</em>
					</td>
					<!--<td>File</td> -->
					<td>NA</td>
					<td>Contamination per sample</td>
				</tr>
				<tr>
					<td><a href="#--indelGapContinuationPenalty">--indelGapContinuationPenalty</a><br />
								&nbsp;<em>-indelGCP</em>
					</td>
					<!--<td>byte</td> -->
					<td>10</td>
					<td>Indel gap continuation penalty, as Phred-scaled probability.  I.e., 30 => 10^-30/10</td>
				</tr>
				<tr>
					<td><a href="#--indelGapOpenPenalty">--indelGapOpenPenalty</a><br />
								&nbsp;<em>-indelGOP</em>
					</td>
					<!--<td>byte</td> -->
					<td>45</td>
					<td>Indel gap open penalty, as Phred-scaled probability.  I.e., 30 => 10^-30/10</td>
				</tr>
				<tr>
					<td><a href="#--input_prior">--input_prior</a><br />
								&nbsp;<em>-inputPrior</em>
					</td>
					<!--<td>List[Double]</td> -->
					<td>[]</td>
					<td>Input prior for calls</td>
				</tr>
				<tr>
					<td><a href="#--max_alternate_alleles">--max_alternate_alleles</a><br />
								&nbsp;<em>-maxAltAlleles</em>
					</td>
					<!--<td>int</td> -->
					<td>6</td>
					<td>Maximum number of alternate alleles to genotype</td>
				</tr>
				<tr>
					<td><a href="#--max_genotype_count">--max_genotype_count</a><br />
								&nbsp;<em>-maxGT</em>
					</td>
					<!--<td>int</td> -->
					<td>1024</td>
					<td>Maximum number of genotypes to consider at any site</td>
				</tr>
				<tr>
					<td><a href="#--max_num_PL_values">--max_num_PL_values</a><br />
								&nbsp;<em>-maxNumPLValues</em>
					</td>
					<!--<td>int</td> -->
					<td>100</td>
					<td>Maximum number of PL values to output</td>
				</tr>
				<tr>
					<td><a href="#--onlyEmitSamples">--onlyEmitSamples</a><br />
					</td>
					<!--<td>Set[String]</td> -->
					<td>[]</td>
					<td>If provided, only these samples will be emitted into the VCF, regardless of which samples are present in the BAM file</td>
				</tr>
				<tr>
					<td><a href="#--output_mode">--output_mode</a><br />
								&nbsp;<em>-out_mode</em>
					</td>
					<!--<td>OutputMode</td> -->
					<td>EMIT_VARIANTS_ONLY</td>
					<td>Which type of calls we should output</td>
				</tr>
			<tr>
				<th colspan="4" id="row-divider">Advanced Flags</th>
			</tr>
				<tr>
					<td><a href="#--allSitePLs">--allSitePLs</a><br />
					</td>
					<!--<td>boolean</td> -->
					<td>false</td>
					<td>Annotate all sites with PLs</td>
				</tr>
						</tbody>
					</table>

							<h3>Argument details</h3>
							<p>Arguments in this list are specific to this tool. Keep in mind that other arguments are available that are shared with other tools (e.g. command-line GATK arguments); see Inherited arguments above.</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--alleles">--alleles </a>
			 / <small>-alleles</small>
		</h3>
		<p class="args">
			<b>Set of alleles to use in genotyping</b><br />
			When --genotyping_mode is set to GENOTYPE_GIVEN_ALLELES mode, the caller will genotype the samples using only the alleles provide in this callset. Note that this is not well tested in HaplotypeCaller, and is definitely not suitable for use with HaplotypeCaller in -ERC GVCF mode. In addition, it does not apply to MuTect2 at all.
		</p>
			<p>This argument supports reference-ordered data (ROD) files in the following formats: BCF2, VCF, VCF3</p>
		<p>
			<span class="label label-info ">RodBinding[VariantContext]</span>
				&nbsp;<span class="label">none</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--allSitePLs">--allSitePLs </a>
			 / <small>-allSitePLs</small>
		</h3>
		<p class="args">
			<b>Annotate all sites with PLs</b><br />
			Experimental argument FOR USE WITH UnifiedGenotyper ONLY: if SNP likelihood model
 is specified, and if EMIT_ALL_SITES output mode is set, when we set this argument then we
 will also emit PLs at all sites. This will give a measure of reference confidence and a
 measure of which alt alleles are more plausible (if any).
 WARNINGS:
 - This feature will inflate VCF file size considerably.
 - All SNP ALT alleles will be emitted with corresponding 10 PL values.
 - An error will be emitted if EMIT_ALL_SITES is not set, or if anything other than diploid
 SNP model is used
 - THIS WILL NOT WORK WITH HaplotypeCaller, GenotypeGVCFs or MuTect2! Use HaplotypeCaller with
 -ERC GVCF then GenotypeGVCFs instead. See the Best Practices documentation for more information.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--annotateNDA">--annotateNDA </a>
			 / <small>-nda</small>
		</h3>
		<p class="args">
			<b>Annotate number of alleles observed</b><br />
			Depending on the value of the --max_alternate_alleles argument, we may genotype only a fraction of the alleles
 being sent on for genotyping. Using this argument instructs the genotyper to annotate (in the INFO field) the
 number of alternate alleles that were originally discovered (but not necessarily genotyped) at the site.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--annotation">--annotation </a>
			 / <small>-A</small>
		</h3>
		<p class="args">
			<b>One or more specific annotations to apply to variant calls</b><br />
			Which annotations to add to the output VCF file. See the VariantAnnotator -list argument to view available annotations.
		</p>
		<p>
			<span class="label label-info ">List[String]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--comp">--comp </a>
			 / <small>-comp</small>
		</h3>
		<p class="args">
			<b>Comparison VCF file</b><br />
			If a call overlaps with a record from the provided comp track, the INFO field will be annotated
 as such in the output with the track name (e.g. -comp:FOO will have 'FOO' in the INFO field).
 Records that are filtered in the comp track will be ignored.
 Note that 'dbSNP' has been special-cased (see the --dbsnp argument).
		</p>
			<p>This argument supports reference-ordered data (ROD) files in the following formats: BCF2, VCF, VCF3</p>
		<p>
			<span class="label label-info ">List[RodBinding[VariantContext]]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--computeSLOD">--computeSLOD </a>
			 / <small>-slod</small>
		</h3>
		<p class="args">
			<b>If provided, we will calculate the SLOD (SB annotation)</b><br />
			Note that calculating the SLOD increases the runtime by an appreciable amount.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--contamination_fraction_per_sample_file">--contamination_fraction_per_sample_file </a>
			 / <small>-contaminationFile</small>
		</h3>
		<p class="args">
			<b>Contamination per sample</b><br />
			This argument specifies a file with two columns "sample" and "contamination" (separated by a tab)
  specifying the contamination level for those samples (where contamination is given as a
  decimal number, not an integer) per line. There should be no header. Samples that do not appear
  in this file will be processed with CONTAMINATION_FRACTION.
		</p>
		<p>
			<span class="label label-info ">File</span>
				&nbsp;<span class="label">NA</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--contamination_fraction_to_filter">--contamination_fraction_to_filter </a>
			 / <small>-contamination</small>
		</h3>
		<p class="args">
			<b>Fraction of contamination to aggressively remove</b><br />
			If this fraction is greater is than zero, the caller will aggressively attempt to remove
 contamination through biased down-sampling of reads (for all samples). Basically, it will ignore the
 contamination fraction of reads for each alternate allele.  So if the pileup contains N
 total bases, then we will try to remove (N * contamination fraction) bases for each alternate
 allele.
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">0.0</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--dbsnp">--dbsnp </a>
			 / <small>-D</small>
		</h3>
		<p class="args">
			<b>dbSNP file</b><br />
			rsIDs from this file are used to populate the ID column of the output.  Also, the DB INFO flag will be set when appropriate.
 dbSNP is not used in any way for the calculations themselves.
		</p>
			<p>This argument supports reference-ordered data (ROD) files in the following formats: BCF2, VCF, VCF3</p>
		<p>
			<span class="label label-info ">RodBinding[VariantContext]</span>
				&nbsp;<span class="label">none</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--excludeAnnotation">--excludeAnnotation </a>
			 / <small>-XA</small>
		</h3>
		<p class="args">
			<b>One or more specific annotations to exclude</b><br />
			Which annotations to exclude from output in the VCF file.  Note that this argument has higher priority than the -A or -G arguments,
 so annotations will be excluded even if they are explicitly included with the other options.
		</p>
		<p>
			<span class="label label-info ">List[String]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--genotype_likelihoods_model">--genotype_likelihoods_model </a>
			 / <small>-glm</small>
		</h3>
		<p class="args">
			<b>Genotype likelihoods calculation model to employ -- SNP is the default option, while INDEL is also available for calling indels and BOTH is available for calling both together</b><br />
			
		</p>
			<p>
			The --genotype_likelihoods_model argument is an enumerated type (Model), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">SNP</dt>
					<dd class="enum"></dd>
					<dt class="enum">INDEL</dt>
					<dd class="enum"></dd>
					<dt class="enum">GENERALPLOIDYSNP</dt>
					<dd class="enum"></dd>
					<dt class="enum">GENERALPLOIDYINDEL</dt>
					<dd class="enum"></dd>
					<dt class="enum">BOTH</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
			<span class="label label-info ">Model</span>
				&nbsp;<span class="label">SNP</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--genotyping_mode">--genotyping_mode </a>
			 / <small>-gt_mode</small>
		</h3>
		<p class="args">
			<b>Specifies how to determine the alternate alleles to use for genotyping</b><br />
			
		</p>
			<p>
			The --genotyping_mode argument is an enumerated type (GenotypingOutputMode), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">DISCOVERY</dt>
					<dd class="enum">The genotyper will choose the most likely alternate allele</dd>
					<dt class="enum">GENOTYPE_GIVEN_ALLELES</dt>
					<dd class="enum">Only the alleles passed by the user should be considered.</dd>
			</dl>
			</p>
		<p>
			<span class="label label-info ">GenotypingOutputMode</span>
				&nbsp;<span class="label">DISCOVERY</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--group">--group </a>
			 / <small>-G</small>
		</h3>
		<p class="args">
			<b>One or more classes/groups of annotations to apply to variant calls.  The single value 'none' removes the default group</b><br />
			If specified, all available annotations in the group will be applied. See the VariantAnnotator -list argument to view available groups.
 Keep in mind that RODRequiringAnnotations are not intended to be used as a group, because they require specific ROD inputs.
		</p>
		<p>
			<span class="label label-info ">String[]</span>
				&nbsp;<span class="label">[Standard, StandardUG]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--heterozygosity">--heterozygosity </a>
			 / <small>-hets</small>
		</h3>
		<p class="args">
			<b>Heterozygosity value used to compute prior likelihoods for any locus</b><br />
			The expected heterozygosity value used to compute prior probability that a locus is non-reference. See
 https://software.broadinstitute.org/gatk/documentation/article?id=8603 for more details.
		</p>
		<p>
			<span class="label label-info ">Double</span>
				&nbsp;<span class="label">0.001</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--heterozygosity_stdev">--heterozygosity_stdev </a>
			 / <small>-heterozygosityStandardDeviation</small>
		</h3>
		<p class="args">
			<b>Standard deviation of eterozygosity for SNP and indel calling.</b><br />
			The standard deviation of the distribution of alt allele fractions. The above heterozygosity parameters give
 the *mean* of this distribution; this parameter gives its spread.
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">0.01</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--indel_heterozygosity">--indel_heterozygosity </a>
			 / <small>-indelHeterozygosity</small>
		</h3>
		<p class="args">
			<b>Heterozygosity for indel calling</b><br />
			This argument informs the prior probability of having an indel at a site.
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">1.25E-4</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--indelGapContinuationPenalty">--indelGapContinuationPenalty </a>
			 / <small>-indelGCP</small>
		</h3>
		<p class="args">
			<b>Indel gap continuation penalty, as Phred-scaled probability.  I.e., 30 => 10^-30/10</b><br />
			
		</p>
		<p>
			<span class="label label-info ">byte</span>
				&nbsp;<span class="label">10</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--indelGapOpenPenalty">--indelGapOpenPenalty </a>
			 / <small>-indelGOP</small>
		</h3>
		<p class="args">
			<b>Indel gap open penalty, as Phred-scaled probability.  I.e., 30 => 10^-30/10</b><br />
			
		</p>
		<p>
			<span class="label label-info ">byte</span>
				&nbsp;<span class="label">45</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--input_prior">--input_prior </a>
			 / <small>-inputPrior</small>
		</h3>
		<p class="args">
			<b>Input prior for calls</b><br />
			By default, the prior specified with the argument --heterozygosity/-hets is used for variant discovery at a
 particular locus, using an infinite sites model (see e.g. Waterson, 1975 or Tajima, 1996). This model asserts that
 the probability of having a population of k variant sites in N chromosomes is proportional to theta/k, for 1=1:N.
 However, there are instances where using this prior might not be desirable, e.g. for population studies where prior
 might not be appropriate, as for example when the ancestral status of the reference allele is not known.

 This argument allows you to manually specify a list of probabilities for each AC>0 to be used as
 priors for genotyping, with the following restrictions: only diploid calls are supported; you must specify 2 *
 N values where N is the number of samples; probability values must be positive and specified in Double format,
 in linear space (not log10 space nor Phred-scale); and all values must sume to 1.

 For completely flat priors,  specify the same value (=1/(2*N+1)) 2*N times, e.g.
      -inputPrior 0.33 -inputPrior 0.33
 for the single-sample diploid case.
		</p>
		<p>
			<span class="label label-info ">List[Double]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max_alternate_alleles">--max_alternate_alleles </a>
			 / <small>-maxAltAlleles</small>
		</h3>
		<p class="args">
			<b>Maximum number of alternate alleles to genotype</b><br />
			If there are more than this number of alternate alleles presented to the genotyper (either through discovery or
 GENOTYPE_GIVEN_ALLELES), then only this many alleles will be used.  Note that genotyping sites with many
 alternate alleles is both CPU and memory intensive and it scales exponentially based on the number of alternate
 alleles.  Unless there is a good reason to change the default value, we highly recommend that you not play around
 with this parameter.

 See also {@link #MAX_GENOTYPE_COUNT}.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">6</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max_deletion_fraction">--max_deletion_fraction </a>
			 / <small>-deletions</small>
		</h3>
		<p class="args">
			<b>Maximum fraction of reads with deletions spanning this locus for it to be callable</b><br />
			If the fraction of reads with deletions spanning a locus is greater than this value, the site will not be considered callable and will be skipped.
 To disable the use of this parameter, set its value to >1.
		</p>
		<p>
			<span class="label label-info ">Double</span>
				&nbsp;<span class="label">0.05</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max_genotype_count">--max_genotype_count </a>
			 / <small>-maxGT</small>
		</h3>
		<p class="args">
			<b>Maximum number of genotypes to consider at any site</b><br />
			If there are more than this number of genotypes at a locus presented to the genotyper, then only this many
 genotypes will be used. This is intended to deal with sites where the combination of high ploidy and high alt
 allele count can lead to an explosion in the number of possible genotypes, with extreme adverse effects on
 runtime performance.

 How does it work? The possible genotypes are simply different ways of partitioning alleles given a specific
 ploidy assumption. Therefore, we remove genotypes from consideration by removing alternate alleles that are the
 least well supported. The estimate of allele support is based on the ranking of the candidate haplotypes coming
 out of the graph building step. Note however that the reference allele is always kept.

 The maximum number of alternative alleles used in the genotyping step will be the lesser of the two:
 1. the largest number of alt alleles, given ploidy, that yields a genotype count no higher than {@link #MAX_GENOTYPE_COUNT}
 2. the value of {@link #MAX_ALTERNATE_ALLELES}

 As noted above, genotyping sites with large genotype counts is both CPU and memory intensive. Unless you have
 a good reason to change the default value, we highly recommend that you not play around with this parameter.

 See also {@link #MAX_ALTERNATE_ALLELES}.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">1024</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--max_num_PL_values">--max_num_PL_values </a>
			 / <small>-maxNumPLValues</small>
		</h3>
		<p class="args">
			<b>Maximum number of PL values to output</b><br />
			Determines the maximum number of PL values that will be logged in the output.  If the number of genotypes
 (which is determined by the ploidy and the number of alleles) exceeds the value provided by this argument,
 then output of all of the PL values will be suppressed.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">100</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--min_base_quality_score">--min_base_quality_score </a>
			 / <small>-mbq</small>
		</h3>
		<p class="args">
			<b>Minimum base quality required to consider a base for calling</b><br />
			The minimum confidence needed in a given base for it to be used in variant calling.  Note that the base quality of a base
 is capped by the mapping quality so that bases on reads with low mapping quality may get filtered out depending on this value.
 Note too that this argument is ignored in indel calling. In indel calling, low-quality ends of reads are clipped off (with fixed threshold of Q20).
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">17</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--min_indel_count_for_genotyping">--min_indel_count_for_genotyping </a>
			 / <small>-minIndelCnt</small>
		</h3>
		<p class="args">
			<b>Minimum number of consensus indels required to trigger genotyping run</b><br />
			A candidate indel is genotyped (and potentially called) if there are this number of reads with a consensus indel at a site.
 Decreasing this value will increase sensitivity but at the cost of larger calling time and a larger number of false positives.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">5</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--min_indel_fraction_per_sample">--min_indel_fraction_per_sample </a>
			 / <small>-minIndelFrac</small>
		</h3>
		<p class="args">
			<b>Minimum fraction of all reads at a locus that must contain an indel (of any allele) for that sample to contribute to the indel count for alleles</b><br />
			Complementary argument to minIndelCnt.  Only samples with at least this fraction of indel-containing reads will contribute
 to counting and overcoming the threshold minIndelCnt.  This parameter ensures that in deep data you don't end
 up summing lots of super rare errors up to overcome the 5 read default threshold.  Should work equally well for
 low-coverage and high-coverage samples, as low coverage samples with any indel containing reads should easily over
 come this threshold.
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">0.25</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--onlyEmitSamples">--onlyEmitSamples </a>
			 / <small>-onlyEmitSamples</small>
		</h3>
		<p class="args">
			<b>If provided, only these samples will be emitted into the VCF, regardless of which samples are present in the BAM file</b><br />
			
		</p>
		<p>
			<span class="label label-info ">Set[String]</span>
				&nbsp;<span class="label">[]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--out">--out </a>
			 / <small>-o</small>
		</h3>
		<p class="args">
			<b>File to which variants should be written</b><br />
			A raw, unfiltered, highly sensitive callset in VCF format.
		</p>
		<p>
			<span class="label label-info ">VariantContextWriter</span>
				&nbsp;<span class="label">stdout</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--output_mode">--output_mode </a>
			 / <small>-out_mode</small>
		</h3>
		<p class="args">
			<b>Which type of calls we should output</b><br />
			Experimental argument FOR USE WITH UnifiedGenotyper ONLY. When using HaplotypeCaller, use -ERC
 instead. When using GenotypeGVCFs, see -allSites.
		</p>
			<p>
			The --output_mode argument is an enumerated type (OutputMode), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">EMIT_VARIANTS_ONLY</dt>
					<dd class="enum">produces calls only at variant sites</dd>
					<dt class="enum">EMIT_ALL_CONFIDENT_SITES</dt>
					<dd class="enum">produces calls at variant sites and confident reference sites</dd>
					<dt class="enum">EMIT_ALL_SITES</dt>
					<dd class="enum">produces calls at any callable site regardless of confidence; this argument is intended only for point
 mutations (SNPs) in DISCOVERY mode or generally when running in GENOTYPE_GIVEN_ALLELES mode; it will by
 no means produce a comprehensive set of indels in DISCOVERY mode</dd>
			</dl>
			</p>
		<p>
			<span class="label label-info ">OutputMode</span>
				&nbsp;<span class="label">EMIT_VARIANTS_ONLY</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--pair_hmm_implementation">--pair_hmm_implementation </a>
			 / <small>-pairHMM</small>
		</h3>
		<p class="args">
			<b>The PairHMM implementation to use for -glm INDEL genotype likelihood calculations</b><br />
			The PairHMM implementation to use for -glm INDEL genotype likelihood calculations. The various implementations balance a tradeoff of accuracy and runtime.
		</p>
			<p>
			The --pair_hmm_implementation argument is an enumerated type (HMM_IMPLEMENTATION), which can have one of the following values:
			<dl class="enum">
					<dt class="enum">EXACT</dt>
					<dd class="enum"></dd>
					<dt class="enum">ORIGINAL</dt>
					<dd class="enum"></dd>
					<dt class="enum">LOGLESS_CACHING</dt>
					<dd class="enum"></dd>
					<dt class="enum">VECTOR_LOGLESS_CACHING</dt>
					<dd class="enum"></dd>
					<dt class="enum">VECTOR_LOGLESS_CACHING_OMP</dt>
					<dd class="enum"></dd>
					<dt class="enum">VECTOR_LOGLESS_CACHING_FPGA_EXPERIMENTAL</dt>
					<dd class="enum"></dd>
					<dt class="enum">FASTEST_AVAILABLE</dt>
					<dd class="enum"></dd>
					<dt class="enum">DEBUG_VECTOR_LOGLESS_CACHING</dt>
					<dd class="enum"></dd>
					<dt class="enum">ARRAY_LOGLESS</dt>
					<dd class="enum"></dd>
			</dl>
			</p>
		<p>
			<span class="label label-info ">HMM_IMPLEMENTATION</span>
				&nbsp;<span class="label">LOGLESS_CACHING</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--pcr_error_rate">--pcr_error_rate </a>
			 / <small>-pcr_error</small>
		</h3>
		<p class="args">
			<b>The PCR error rate to be used for computing fragment-based likelihoods</b><br />
			The PCR error rate is independent of the sequencing error rate, which is necessary because we cannot necessarily
 distinguish between PCR errors vs. sequencing errors.  The practical implication for this value is that it
 effectively acts as a cap on the base qualities.
		</p>
		<p>
			<span class="label label-info ">Double</span>
				&nbsp;<span class="label">1.0E-4</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--sample_ploidy">--sample_ploidy </a>
			 / <small>-ploidy</small>
		</h3>
		<p class="args">
			<b>Ploidy per sample. For pooled data, set to (Number of samples in each pool * Sample Ploidy).</b><br />
			Sample ploidy - equivalent to number of chromosome copies per pool. For pooled experiments this should be set to
   the number of samples in pool multiplied by individual sample ploidy.
		</p>
		<p>
			<span class="label label-info ">int</span>
				&nbsp;<span class="label">2</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--standard_min_confidence_threshold_for_calling">--standard_min_confidence_threshold_for_calling </a>
			 / <small>-stand_call_conf</small>
		</h3>
		<p class="args">
			<b>The minimum phred-scaled confidence threshold at which variants should be called</b><br />
			The minimum phred-scaled confidence threshold at which variants should be called. Only variant sites with QUAL equal
 or greater than this threshold will be called. Note that since version 3.7, we no longer differentiate high confidence
 from low confidence calls at the calling step. The default call confidence threshold is set low intentionally to achieve
 high sensitivity, which will allow false positive calls as a side effect. Be sure to perform some kind of filtering after
 calling to reduce the amount of false positives in your final callset. Note that when HaplotypeCaller is used in GVCF mode
 (using either -ERC GVCF or -ERC BP_RESOLUTION) the call threshold is automatically set to zero. Call confidence thresholding
 will then be performed in the subsequent GenotypeGVCFs command.
		</p>
		<p>
			<span class="label label-info ">double</span>
				&nbsp;<span class="label">10.0</span>
				&nbsp;<span class="label label-warning">[ [ -∞</span>
				&nbsp;<span class="label label-warning">∞ ] ]</span>
		</p>
		<hr style="border-bottom: dotted 1px #C0C0C0;" />
		<h3><a name="--useNewAFCalculator">--useNewAFCalculator </a>
			 / <small>-newQual</small>
		</h3>
		<p class="args">
			<b>Use new AF model instead of the so-called exact model</b><br />
			This activates a model for calculating QUAL that was introduced in version 3.7 (November 2016). We expect this
 model will become the default in future versions.
		</p>
		<p>
			<span class="label label-info ">boolean</span>
				&nbsp;<span class="label">false</span>
		</p>

        <hr>


        

	</div>

</div>

