## Best Practices VQSR parameters updated

By Geraldine_VdAuwera

<p>The Best Practices recommendations for Variant Quality Score Recalibration have been slightly updated to use the new(ish) <a rel="nofollow" href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.php">StrandOddsRatio (SOR)</a> annotation, which <strong>complements</strong> <a rel="nofollow" href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.php">FisherStrand (FS)</a> as indicator of strand bias (only available in GATK version 3.3-0 and above).</p>

<p>While we were at it we also reconciled some inconsistencies between the <a rel="nofollow" href="http://gatkforums.broadinstitute.org/discussion/2805/howto-recalibrate-variant-quality-scores-run-vqsr">tutorial</a> and the <a rel="nofollow" href="http://gatkforums.broadinstitute.org/discussion/1259/which-training-sets-arguments-should-i-use-for-running-vqsr">FAQ</a> document. As a reminder, if you ever find differences between parameters given in the VQSR docs, let us know, but FYI that the FAQ is the ultimate source of <code class="code codeInline" spellcheck="false">truth=true</code>. Note also that the command line example given in <a rel="nofollow" href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_variantrecalibration_VariantRecalibrator.php">VariantRecalibrator tool doc</a> tends to be out of date because it can only be updated with the next release (due to a limitation of the tool doc generation system) and, well, we often forget to do it in time -- so it should <em>never</em> be used as a reference for Best Practice parameter values, <em>as indicated in the caveat right underneath it which no one ever reads</em>.</p>

<p>Speaking of caveats, there's no such thing as too much repetition of the fact that whole genomes and exomes have subtle differences that require some tweaks to your command lines. In the case of VQSR, that means dropping Coverage (DP) from your VQSR command lines if you're working with exomes.</p>

<p>Finally, keep in mind that the values we recommend for tranches are really just examples; if there's one setting you should freely experiment with, that's the one. You can specify as many tranche cuts as you want to get really fine resolution.</p>
