## Changing workflows around calling SNPs and indels

By shlee

<h2>We are streamlining our recommended workflows by removing a preprocessing step.</h2>

<p>As <a rel="nofollow" href="https://www.broadinstitute.org/gatk/blog?id=7712">announced in the GATK v3.6 highlights</a>, <strong>variant calling workflows that use HaplotypeCaller or MuTect2 now omit indel realignment</strong>. This change does not apply to workflows that call variants with UnifiedGenotyper or the original MuTect. We still recommend indel realignment for these legacy workflows. Recall that <a rel="nofollow" href="https://www.broadinstitute.org/gatk/guide/article?id=7156">indel realignment uses RealignerTargetCreator and IndelRealigner</a> and comes after duplicate marking and before base quality score recalibration (BQSR).</p>

<p>In light of these changes, let’s take a brisk stroll through the implications for variant detection. In particular, let’s focus on insertion and deletion events (indels).</p>

<p><a name="passage1" id="passage1"></a></p>

<hr></hr><h2>Why am I blogging about this now?</h2>

<p>I recently studied IndelRealigner and HaplotypeCaller, one after the other, to update a tutorial and to give workshop talks in the UK. <strong>This juxtaposition has brought into focus how our capabilities are changing with (i) improving sequencing technology and (ii) improving algorithmic methods.</strong> For a comprehensive overview of the state of different sequencing technologies, see the <a rel="nofollow" href="http://www.nature.com/nrg/journal/v17/n6/full/nrg.2016.49.html">2016 Goodwin et al. <em>Nature Reviews</em> article</a>. The points in this blogpost pertain in particular to short read data from sequencing by synthesis. Longer <em>short reads</em>, deeper coverage and increased library complexity with PCR-free preparations give us higher quality sequence data. The graph in <a rel="nofollow" href="https://flxlexblog.wordpress.com/2015/06/17/developments-in-high-throughput-sequencing-june-2015-edition/">this external blog</a> shows short read lengths varying from 25 bases in 2006 to 250 bases in 2014. Reads longer than 70 basepairs enable the mapping optimizations BWA-MEM offers, i.e. maximal exact matches. Adding to this, sophisticated haplotype-based variant detectors, such as <a rel="nofollow" href="https://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.php">HaplotypeCaller</a> and <a rel="nofollow" href="https://github.com/ekg/freebayes">FreeBayes</a>, improve our ability to resolve variants. In particular, they are enabling better detection of indels.</p>

<p>Obviously, the added benefit from greater read lengths is more than just the proportional increase in bases. This <a rel="nofollow" href="http://www.biotechniques.com/news/Going-to-Great-Read-Lengths/biotechniques-341722.html#.VyOXYD8_4yM">external article</a> recaps how <strong>coverage alone cannot compensate for the shortcomings of short read lengths in closing gaps</strong>. Longer reads reduce mapping ambiguity for regions with interspersed repetitive sequences, for some paralogous regions and for gapped alignments. In the case of gapped alignment, the aligner may split the read into multiple alignment records for distant or discordant events, resolve the split with an indel for local events and clip segments that it cannot map. All this splitting and clipping carries some burden of aligner artifact that confounds variant detection. For example, in the case of an insertion in the middle of a read, the inserted sequence distracts from global then local mapping.</p>

<p>In this way, short read mapping limits detection of variants that are dependent on gapped alignment. Towards overcoming these limitations, we added new algorithmic solutions to our workflows --local realignment in 2009 and local assembly in 2012. <a rel="nofollow" href="#illustrationA">Illustration A</a> describes a locus with a 20 basepair insertion that we cannot call with just aligner mapped reads. <strong>Realigning reads using IndelRealigner or assembling reads using HaplotypeCaller allows us to call the insertion.</strong> That indel realignment has been a part of pre-processing workflows for seven years and will continue to be a part of workflows still dependent on locus-based callers is a testament to the improvements it brings. And if you feel apprehensive about omitting it from your HaplotypeCaller and MuTect2 workflows, we empathize. These changes are about improving efficiency in the face of incremental returns. If you find substantial changes, then I encourage you to share details with us.</p>

<p>Let’s consider another illustration that showcases HaplotypeCaller's more robust capability. <a rel="nofollow" href="#illustrationB">Illustration B</a> shows how <strong>IndelRealigner is unable to realign reads around a 30 basepair insertion</strong>, despite inclusion of the region as a target and the presence of a read with the insertion. <strong>HaplotypeCaller reassembles and calls this insertion with or without indel realignment</strong>. In point of fact, for the example data, the two HaplotypeCaller workflows call identical genotypes for all sites. However, <a rel="nofollow" href="#illustrationA">Illustration A</a> points to some subtle differences in the variant annotation metrics. I will talk about the origins of these and their implications.</p>

<p><strong>For those interested in an anecdote from MuTect2</strong>, which uses HaplotypeCaller’s approach but without ploidy assumptions to accommodate tumor data, see our <a rel="nofollow" href="https://drive.google.com/open?id=0BwTg3aXzGxEDdXBtWUhnQ2NJUTg">MuTect2 workshop presentation</a>. Slides 14–15 outline a case where MuTect2 resolves a 120 basepair deletion haplotype for a tumor.</p>

<p><a name="passage2" id="passage2"></a></p>

<hr></hr><h2>Did indel realignment still have some value-add?</h2>

<p><a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/FileUpload/67/d60943af33446000bcc10ebae5f68d.png"><img src="https://us.v-cdn.net/5019796/uploads/FileUpload/67/d60943af33446000bcc10ebae5f68d.png" height="120" alt="image" style="float: right;" class="embedImage-img importedEmbed-img"></img></a>  If indel realignment is redundant to HaplotypeCaller’s functions, why were we recommending both until now? Well, <strong>although HaplotypeCaller throws out realignment’s local placement of reads, indel realignment still influences outcomes.</strong> I’m told these influences should be subtle for high quality data but make a difference for lower quality data. Let’s talk about the influences conceptually and examine some initial differences for our example data snippet. Breaking down the effects into two categories gives us: <strong>(1)</strong> the effects of increasing MAPQ by ten for realigned reads and <strong>(2)</strong> the effects of the physical realignment.</p>

<h3>(1) The effects of increasing MAPQ by 10 for realigned reads</h3>

<p><a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/FileUpload/a5/c36ff31d4e7ca8eaec8484789daaf5.png"><img src="https://us.v-cdn.net/5019796/uploads/FileUpload/a5/c36ff31d4e7ca8eaec8484789daaf5.png" height="120" alt="image" style="float: left;" class="embedImage-img importedEmbed-img"></img></a> <strong>(1a)</strong> Increases in MAPQ add previously ignored reads to the pool of primary alignment records BaseRecalibrator (BQSR) and HaplotypeCaller consider. BQSR gains previously MAPQ0 reads now MAPQ10 and HaplotypeCaller gains previously MAPQ10–19 reads now over the default MAPQ20 threshold of the <a rel="nofollow" href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_haplotypecaller_HCMappingQualityFilter.php">HCMappingQualityFilter</a>. <strong>(1b)</strong> HaplotypeCaller considers all indel realignment target regions, and except for regions that HaplotypeCaller cannot resolve cycles for in its graph assembly, e.g. a tandem repeat region, HaplotypeCaller acts on these regions. Indel realigned reads that cross above HaplotypeCaller's MAPQ threshold alter the allelic and total depth counts for a variant site and can affect genotyping and genotype probability and quality scores as well as other annotation metrics. If the added read(s) are informative towards haplotypes in a region with few informative reads, although <a rel="nofollow" href="https://www.broadinstitute.org/gatk/guide/article?id=6005">they do not impact genotyping any more than uninformative reads</a>, they may have a disproportionate impact on metrics that use exclusively informative reads, e.g. sample-level AD &amp; DP (allelic depth and total depth), QD (QUAL by depth where QUAL is the phred-scaled score for the assertion made in ALT) and FS &amp; SOR (Fisher strand bias and strand odds ratio). <strong>(1c)</strong> Variant sites that include these increased MAPQ reads directly increase MQ and MQRankSum annotation values (root mean square of mapping quality and ranksum test of ALT vs. REF read mapping qualities). <strong>(1d)</strong> Finally, these altered annotation metrics (QD, FS, SOR, MQ &amp; MQRankSum for the snippet) impact cohort level annotation metrics, the variant quality score recalibration (<a rel="nofollow" href="https://www.broadinstitute.org/gatk/guide/article?id=39">VQSR</a>) model and the resulting VQSLOD score for the site. These changes to the distribution of the variant scores impact variant site filtering.</p>

<p>For our data snippet, indel realignment adds zero reads to either pool. The lowest MAPQ of reads indel realignment then increases is at MAPQ26 and in fact over 94% of increases are for MAPQ60 reads. The 194 (~0.07% of total primary) indel realigned reads increase cumulative MAPQ by 0.012% and the impact on MQ for the 20 basepair insertion site in <a rel="nofollow" href="#illustrationA">Illustration A</a> is small (MQ 59.39 vs. 62.81). This site has plenty of informative reads. Interestingly, indel realignment adds four more reads that HaplotypeCaller considers for this locus and adds a total of five additional reads for two loci for our snippet's entire one megabasepair region. Of the four additional reads, only one is realigned. This brings us to the next section's topic.</p>

<h3>(2) The effects of the physical realignment of reads</h3>

<p>Indel realignment shifts read positions and this impacts the outcomes of both BQSR and HaplotypeCaller. <strong>(2a)</strong> Shifted reads and their mismatching positions around indels will change the bases <a rel="nofollow" href="http://gatkforums.broadinstitute.org/gatk/discussion/44/base-quality-score-recalibration-bqsr">BQSR</a> modeling uses and potentially recalibrates. Indel realignment generally reduces false positive mismatches around indels. If these mismatches contribute to a covariate pattern, and indel realignment removes these mismatches, then indel realignment removes the support towards these patterns and in turn enables BQSR to calibrate using cleaner models. These contributing false mismatches avoid being unfairly penalized. However, BQSR looks for fairly obvious failures stemming from sequencer flowcell technology such as those <a rel="nofollow" href="https://sequencing.qcfail.com/articles/position-specific-failures-of-flowcells/">this external blog</a> describes neatly. Any discernible patterns from misaligned reads around indels will be small compared to that of flowcell failures. This is one area that could use more data on the extent of the impact of removing indel realignment. What I am told is that effects will focus more on SNPs than indels in that the new workflow lowers the confidence in mainly what will become reference matches after HaplotypeCaller realignment. For these instances, at the worst, you may end up calling a false heterozygous SNP instead of homozygous reference.</p>

<p>Additionally, changes to read positions for a locus impact <strong>(2b)</strong> the genomic window HaplotypeCaller considers for graph assembly and realignment, because the <a rel="nofollow" href="http://gatkforums.broadinstitute.org/gatk/discussion/4147">entropy</a> distribution changes, and <strong>(2c)</strong> the reads the window includes. For <a rel="nofollow" href="#illustrationA">Ilustration A</a>'s locus, indel realignment tightens the active region size from 208 to 152 basepairs and also adds four additional reads for consideration. Such occurences alter site annotation values and can influence downstream variant calling.</p>

<p>For our data snippet (again, high quality 30X coverage PCR-free 2x150 WGS), the effects of removing indel realignment are subtle. However, for data that deviate towards lower quality and for data that benefit greatly from BQSR, the effects could be more substantial. Given this, we ask researchers who are considering our new workflows to take into account properties of their data and consider the impact omitting indel realignment may have for their analyses. A first step towards this could be to examine the MAPQ distribution of the primary alignments.</p>

<hr></hr><h2>Let's wrap up with some wider picture considerations.</h2>

<p>I've covered some ground on the influences of removing indel realignment from a workflow. One side note is that <strong>called sites from different variant callers can have different allele variants</strong>. For example, IndelRealigner and HaplotypeCaller can differ in how they resolve the variation around a given locus. This raises the question to what extent differential representation, based on different underlying algorithmic solutions, contribute to the discordant calls observed between different workflows. If the extent is great, then, given the potential for differential phenotypic classification, the clinical impact can be significant. A <a rel="nofollow" href="https://www.genomeweb.com/informatics/genomic-variant-data-sharing-gains-support-collaboration-seen-key-interpretation">recent article</a> describes how the burgeoning movement to use sequencing in patient diagnostics and disease research is hampered by discordant classifications. By the same reasoning, then the ability to physically phase proximal variants, that determines their <em>cis</em> or <em>trans</em> relationship, has direct impact on phenotypic classification. For example, resolving phasing using tools such as <a rel="nofollow" href="http://gatkforums.broadinstitute.org/gatk/discussion/45">ReadBackedPhasing</a> can tell us whether a variant that shifts the coding frame is expressed on the same molecule as another variant that restores the coding frame.</p>

<p><strong>You should care about resolving indels as best you can even if your analyses focus only on SNPs.</strong> Correctly resolving indels reduces the false positive mismatches from alignment that accrue in the vicinity of indels. Also, we care about resolving larger indels for the biological insights they will bring. One example of the biological significance of indels lies in the transposable elements that litter our genomes that in transposing create insertion and deletion events. These elements are currently considered structural given their large size. They are a part of our individual variation and have been proposed to drive speciation and evolution and are implicated in disease including cancer (see <a rel="nofollow" href="https://en.wikipedia.org/wiki/Endogenous_retrovirus">endogenous retrovirus</a>). The <a rel="nofollow" href="http://www.repeatmasker.org/species/hg.html">RepeatMasker web resource</a> catalogues our repetitive elements, to which transposable elements belong, that together comprise over half our human genome.</p>

<p>Finally, <strong>technological capabilities are changing the definition of an indel</strong>. Back yonder, detection of indels was limited to small indels of a couple to fewer than ten basepairs. GATK workflows assigned resolving anything larger to the realm of structural variant detection. Today, lengthening sequencing reads together with improving global and local alignment algorithms are revealing indels reaching tens to low hundreds of basepairs in our workflows.</p>

<p><a name="illustrationA" id="illustrationA"></a></p>

<blockquote class="UserQuote blockquote"><div class="blockquote-content">
  <p class="blockquote-line">To see for yourself the points made in Illustrations A &amp; B, download <a rel="nofollow" href="https://www.broadinstitute.org/gatk/guide/article?id=7156">Tutorial#7156</a>’s example data <a rel="nofollow" href="https://drive.google.com/open?id=0BzI1CyccGsZia2xmYWdGVVJGb2c">tutorial_7156.tar.gz</a> and follow instructions in <a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/FileUpload/f7/882d5d9cac21e970cb8c3abf145aed.txt">Blog#7847_README.txt</a>.</p>
</div></blockquote>

<hr></hr><h2>Illustration A</h2>

<h3>Both IndelRealigner and HaplotypeCaller allow for better indel detection over UnifiedGenotyper alone.</h3>

<p>Here I consider a 20 basepair insertion, at 10:96,424,673, for a 2x150 PCR-free human WGS library with deep coverage (~30x average) shown in <a rel="nofollow" href="http://gatkforums.broadinstitute.org/gatk/discussion/7156#section2">Tutorial#7156</a>'s <a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/FileUpload/07/5c266d45fec220139c065ed8d540fb.png">second IGV screenshot</a>. The region’s sequence is of low complexity and the insertion site also carries a SNP in <em>trans</em>. To capture IndelRealigner’s (IR) capabilities, I used UnifiedGenotyper (UG) as the variant caller, as its genotyping engine is equivalent to that of HaplotypeCaller’s (HC). Below are the calls for the site for the different workflows.</p>

<p><strong>snippet_UG.vcf</strong></p>

<pre class="code codeBlock" spellcheck="false">#CHROM    POS    ID    REF    ALT    QUAL    FILTER    INFO    FORMAT    NA12878
10  96424673    .   G   C   824.77  .   AC=1;AF=0.500;AN=2;BaseQRankSum=1.767;DP=44;Dels=0.00;ExcessHet=3.0103;FS=1.268;HaplotypeScore=13.7364;MLEAC=1;MLEAF=0.500;MQ=59.06;MQ0=0;MQRankSum=1.770;QD=18.74;ReadPosRankSum=-0.988;SOR=0.480  GT:AD:DP:GQ:PL  0/1:17,27:44:99:853,0,512
</pre>

<p><strong>snippet_IR_UG.vcf</strong></p>

<pre class="code codeBlock" spellcheck="false">10  96424673    .   G   C   630.77  .   AC=1;AF=0.500;AN=2;BaseQRankSum=1.259;DP=40;Dels=0.00;ExcessHet=3.0103;FS=0.000;HaplotypeScore=18.8916;MLEAC=1;MLEAF=0.500;MQ=62.57;MQ0=0;MQRankSum=-3.725;QD=15.77;ReadPosRankSum=0.234;SOR=0.836  GT:AD:DP:GQ:PL  0/1:18,22:40:99:659,0,575
10  96424673    .   G   GTTTGTTTGTTTCTTTCTTTC   2415.73 .   AC=1;AF=0.500;AN=2;BaseQRankSum=-1.390;DP=40;ExcessHet=3.0103;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=62.57;MQ0=0;MQRankSum=3.552;QD=10.35;ReadPosRankSum=-2.066;SOR=0.770  GT:AD:DP:GQ:PL  0/1:17,18:40:99:2453,0,2077
</pre>

<p><strong>snippet_HC.vcf</strong></p>

<pre class="code codeBlock" spellcheck="false">10  96424673    .   G   C,GTTTGTTTGTTTCTTTCTTTC 900.73  .   AC=1,1;AF=0.500,0.500;AN=2;DP=40;ExcessHet=3.0103;FS=0.000;MLEAC=1,1;MLEAF=0.500,0.500;MQ=59.39;QD=33.49;SOR=2.712  GT:AD:DP:GQ:PL  1/2:0,12,8:20:99:938,323,569,472,0,742
</pre>

<p><strong>snippet_IR_HC.vcf</strong></p>

<pre class="code codeBlock" spellcheck="false">10  96424673    .   G   C,GTTTGTTTGTTTCTTTCTTTC 1028.73 .   AC=1,1;AF=0.500,0.500;AN=2;DP=44;ExcessHet=3.0103;FS=0.000;MLEAC=1,1;MLEAF=0.500,0.500;MQ=62.81;QD=33.49;SOR=1.981  GT:AD:DP:GQ:PL  1/2:0,13,9:22:99:1066,367,607,516,0,782
</pre>

<p>The first record shows UnifiedGenotyper cannot call the insertion with just the aligner mapped reads. The insertion is present in ~5 reads in the original alignment and after IndelRealigner+UnifiedGenotyper is supported by 18 total reads (see <a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/FileUpload/fe/dd603f62d298bae1c45cd2a9b36f75.png">full context</a>). HaplotypeCaller calls the insertion with or without indel realignment. The former uses nine supporting reads out of 22 <a rel="nofollow" href="https://www.broadinstitute.org/gatk/guide/article?id=6005">informative reads</a> while the latter uses eight supporting reads out of 20 informative reads. Additional minor differences manifest in the annotation metrics. For this particular site, these differences are in <code class="code codeInline" spellcheck="false">QUAL</code>, INFO’s <code class="code codeInline" spellcheck="false">DP</code>, <code class="code codeInline" spellcheck="false">MQ</code> and <code class="code codeInline" spellcheck="false">SOR</code>, and FORMAT’s <code class="code codeInline" spellcheck="false">PL</code> and the described <code class="code codeInline" spellcheck="false">AD</code> and <code class="code codeInline" spellcheck="false">DP</code>. Despite these minor differences, the two HaplotypeCaller workflows call identical alleles and genotypes for all sites in the one megabasepair interval.</p>

<p><a rel="nofollow" href="#passage1">back to <em>Why am I blogging about this now?</em></a><br><a rel="nofollow" href="#passage2">back to <em>What was the value-add of indel realignment?</em></a><br><a name="illustrationB" id="illustrationB"></a></p>

<hr></hr><p><a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/FileUpload/53/d309a12672459368f357cebae0ef2e.png"><img src="https://us.v-cdn.net/5019796/uploads/FileUpload/53/d309a12672459368f357cebae0ef2e.png" height="270" alt="image" style="float: right;" class="embedImage-img importedEmbed-img"></img></a></p>

<h2>Illustration B</h2>

<h3>HaplotypeCaller boosts indel detection further.</h3>

<p>Here I consider a 30 basepair insertion for the same library in a locus with ~25x depth (shown in <strong>B.1</strong>). The IndelRealigner+UnifiedGenotyper workflow cannot resolve the 30 basepair insertion, despite inclusion of the locus as a target for realignment and a read with the full insertion placed by the aligner. HaplotypeCaller calls the insertion allele and the region shows five informative reads. Furthermore, as I downsample the coverage of the library, the 6% gap between the number of indels HaplotypeCaller and the IndelRealigner+UnifiedGenotyper workflows detect widens, and the numbers drop in parallel (<strong>B.2</strong>). <a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/FileUpload/12/a863c639780f8b829b1361b27b467b.png"><img src="https://us.v-cdn.net/5019796/uploads/FileUpload/12/a863c639780f8b829b1361b27b467b.png" height="180" alt="image" style="float: left;" class="embedImage-img importedEmbed-img"></img></a> I am cautious not to equate greater or matching <em>number</em> of indels with the validity or the concordance of calls. Measuring the full extent of differences requires further comparison of <em>sites</em>, <em>alleles</em>, <em>genotypes</em> and <em>quality</em> of calls. Even still, what is apparent from the illustration is that HaplotypeCaller pushes the envelope of indel detection, a step ahead of IndelRealigner.</p>

<p>HaplotypeCaller calls the insertion with or without indel realignment and the records are shown below. For this particular site, the annotation metrics are identical.</p>

<p><strong>snippet_HC.vcf</strong></p>

<pre class="code codeBlock" spellcheck="false">#CHROM    POS    ID    REF    ALT    QUAL    FILTER    INFO    FORMAT    NA12878
10  96321641    .   A   ATATTCCTTCAATACCCTACCTATTGAAGGG 167.73  .   AC=1;AF=0.500;AN=2;BaseQRankSum=1.438;ClippingRankSum=0.000;DP=23;ExcessHet=3.0103;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=10.48;ReadPosRankSum=-1.190;SOR=0.301   GT:AD:DP:GQ:PL  0/1:11,5:16:99:205,0,2170
</pre>

<p><strong>snippet_IR_HC.vcf</strong></p>

<pre class="code codeBlock" spellcheck="false">10  96321641    .   A   ATATTCCTTCAATACCCTACCTATTGAAGGG 167.73  .   AC=1;AF=0.500;AN=2;BaseQRankSum=1.438;ClippingRankSum=0.000;DP=23;ExcessHet=3.0103;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=60.00;MQRankSum=0.000;QD=10.48;ReadPosRankSum=-1.190;SOR=0.301   GT:AD:DP:GQ:PL  0/1:11,5:16:99:205,0,2170
</pre>

<p><a rel="nofollow" href="#passage1">back to <em>Why am I blogging about this now?</em></a><br><a rel="nofollow" href="#passage2">back to <em>What was the value-add of indel realignment?</em></a></p>

<hr></hr><h3>Links to external references</h3>

<ul><li>Goodwin S, McPherson JD, McCombie WR. Coming of age: ten years of next-generation sequencing technologies. Nat Rev Genet. 2016;17(6):333-351. <a href="http://dx.doi.org/10.1038/nrg.2016.49" rel="nofollow">http://dx.doi.org/10.1038/nrg.2016.49</a>.</li>
<li>Blog <em>In between lines of code</em>'s post titled "Developments in high throughput sequencing -- June 2015 edition" by Lex Nederbragt posted June 18, 2015. Accessed May 2016 at <a rel="nofollow" href="https://flxlexblog.wordpress.com/2015/06/17/developments-in-high-throughput-sequencing-june-2015-edition/">https://flxlexblog.wordpress.com/2015/06/17/developments-in-high-throughput-sequencing-june-2015-edition/</a>.</li>
<li><em>BioTechniques</em> journal article titled <em>Going to Great Read Lengths</em> by Janelle Weaver and dated October 27, 2014. Accessed May 2016 at <a rel="nofollow" href="http://www.biotechniques.com/news/Going-to-Great-Read-Lengths/biotechniques-341722.html#.Vyz2DT8_4yN">http://www.biotechniques.com/news/Going-to-Great-Read-Lengths/biotechniques-341722.html#.Vyz2DT8_4yN</a>.</li>
<li><em>GenomeWeb</em> article titled "Genomic Variant Data Sharing Gains Support; Collaboration Seen as Key to Interpretation Challenge" by Turna Ray and dated May 2, 2016. Accessed May 2016 at <a rel="nofollow" href="https://www.genomeweb.com/informatics/genomic-variant-data-sharing-gains-support-collaboration-seen-key-interpretation">https://www.genomeweb.com/informatics/genomic-variant-data-sharing-gains-support-collaboration-seen-key-interpretation</a>.</li>
<li><em>QCFAIL.com</em> site blog titled "Position specific failures of flowcells" by Simon Andrews dated January 31, 2016. Accessed May 2016 at <a rel="nofollow" href="https://sequencing.qcfail.com/articles/position-specific-failures-of-flowcells/">https://sequencing.qcfail.com/articles/position-specific-failures-of-flowcells/</a>.</li>
<li><em>Repeatmasker</em> web resource titled "Human [Homo sapiens] Genomic Dataset" for hg38 by the Institute for Systems Biology based on Repeat Library date-stamped db20140131. Accessed May 2016 at <a rel="nofollow" href="http://www.repeatmasker.org/species/hg.html">http://www.repeatmasker.org/species/hg.html</a>.</li>
</ul>