## A dog, a ship and an algorithm that measures relatedness

By shlee

<h3>What is beagle.</h3>

<p>Beagle is a type of dog known for its even temper and intelligence. It is also the name given to the ship Darwin sailed to the Galapagos (the <a rel="nofollow" href="https://www.aboutdarwin.com/literature/Beagles.html"><em>H.M.S. Beagle</em></a>), where he developed his theory of natural selection from observing finches. It is also the name of a genomics software package known for phasing and imputing genotypes. <em>Beagle</em> also calls genotypes and detects <a rel="nofollow" href="https://isogg.org/wiki/Identical_by_descent">identity-by-descent</a> (IBD), i.e. it can find segments of identical DNA that indicate two individuals are related.</p>

<p>I will be writing a series of posts where I share with you how I take 23andMe raw data to locate IBD segments using Beagle v4.1 (<a rel="nofollow" href="https://faculty.washington.edu/browning/beagle/beagle.html">website</a>; <a rel="nofollow" href="http://www.genetics.org/content/194/2/459">doi:10.1534/genetics.113.150029</a>). For a review of the statistical methods and other theory underlying IBD, see <a rel="nofollow" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3664843/">doi:10.1534/genetics.112.148825</a>. To see a skipper’s dog on a ship at sea, watch <a rel="nofollow" href="https://www.youtube.com/watch?v=icnjC_gJOLQ#t=9m58s">Irving Johnson’s footage of the Peking barque</a>.</p>

<hr></hr><h3>Why a GATK blog series that is mostly not about GATK?</h3>

<p>I became interested in genomics after doing my 23andMe in 2013. I am in this field as a newb--some might say <em>upstart</em>. My team is giving me the freedom to write this series as a <em>twenty percent project</em>, as a reward for my hard work.</p>

<p>It appears to me that we are at the start of an era of genomics, as evidenced by the <em>Precision Medicine Initiative</em>. The initiative’s <a rel="nofollow" href="https://www.nih.gov/research-training/allofus-research-program"><em>All of Us</em></a> program will begin to recruit study participants this year (anyone in the U.S. can <a rel="nofollow" href="https://service.govdelivery.com/accounts/USNIH/subscriber/new?topic_id=USNIH_95">enroll</a>) and aims to build a genomics resource that reflects the diversity of the U.S. population (<a rel="nofollow" href="http://www.nature.com/articles/ncomms7596">doi:10.1038/ncomms7596</a>). I became aware of this project when I attended a Town Hall meeting a week ago, because other teams under the same umbrella as the GATK team work on it.</p>

<p>I think this series of posts is useful for two reasons.</p>

<p>First, GATK and Picard are not the only programs that offer useful tools in genomic analyses. There are a multitude of bioinformatics software tools. Each can be said to occupy some niche in the ecosystem of tools that help us understand genomics. These versioned software programs enable reproducible data transformations. By experimenting on projects like this one,  I hope to become more aware of the available tools. By focusing on one specific question, I can see how the tools fit together for a particular use case. Through this exercise, my perspective of the field will widen.</p>

<p>Second, the steps that I will go through to preprocess data will illustrate some considerations when preparing genotype data for analyses. If you haven’t noticed from our forum or experienced first-hand from error messages, the Picard and GATK are sticklers for adherence to <a rel="nofollow" href="https://samtools.github.io/hts-specs/">SAM and VCF specifications</a>. This is to ensure compatibility of data across the many genomics applications. Our reasons for this upfront cost are justified. We prevent the pain that unconventional formats, even in experienced hands, have a tendency to spawn.</p>

<hr></hr><h3>Let’s go over some background.</h3>

<p>Genotyping data is not just about calling two alleles for a diploid genomic site, aka locus. We can also capture <em>phasing</em> information, i.e. which set of alleles are on one chromosome molecule, or <em>haplotype</em>, and which are on the other. For example, you might have an <code class="code codeInline" spellcheck="false">AT</code> genotype for site 1 and a <code class="code codeInline" spellcheck="false">GC</code> for site 2. If we know that <code class="code codeInline" spellcheck="false">A</code> and <code class="code codeInline" spellcheck="false">C</code> are from mom and <code class="code codeInline" spellcheck="false">T</code> and <code class="code codeInline" spellcheck="false">G</code> are from dad, then we know the phasing of these two genotypes in relation to each other. I illustrate this concept in a diagram later.</p>

<p>Extend phasing to more loci, and what we get are blocks of sequence with a pattern of variant alleles that travel together and that we refer to as a <a rel="nofollow" href="https://en.wikipedia.org/wiki/Linkage_disequilibrium">linkage disequilibrium</a> (LD) block. These blocks can be of variable size because they can be broken up in meiotic recombination <sup>1</sup>, which happens on average roughly twice per chromosome. To phase short-read sequence and SNP array data, we impute an individual’s LD blocks using known population LD blocks. The alternative approach is to physically phase variants using expensive technology that sequences long molecules, e.g. from PacBio, or to incorporate long range phasing information, e.g. using 10X Genomics technology or algorithms such as phASER with RNA splice isoforms (<a rel="nofollow" href="http://biorxiv.org/content/early/2016/02/12/039529">doi:10.1101/039529</a>).</p>

<p>Similarity and relatedness are two different concepts. You may have heard each of the following.</p>

<ol><li>Humans are <a rel="nofollow" href="(http://education.seattlepi.com/animals-share-human-dna-sequences-6693.html)">98.8% similar to chimpanzees and 84% similar to dogs</a>.</li>
<li>Humans are 99.9% genetically identical.</li>
<li>Humans share <a rel="nofollow" href="http://genetics.thetech.org/original_news/news38">99.5% DNA with any other human</a>.</li>
<li>Siblings share between <a rel="nofollow" href="https://isogg.org/wiki/Autosomal_DNA_statistics">83.81% and 87.47% of SNPs</a>.</li>
<li>Siblings are on average <a rel="nofollow" href="https://isogg.org/wiki/Autosomal_DNA_statistics">50% identical in DNA to each other</a>.<br>
The first two instances compare the entirety of syntenic genomic sequence. The third incorporates copy number variants (CNVs). The fourth compares only genomic positions that are commonly variant in humans and ignores nonvariant DNA. The last refers to shared LD blocks.</li>
</ol><p><a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/FileUpload/0b/6cdede3bf9ebc50fc4dbf63b67af1a.png"><img src="https://us.v-cdn.net/5019796/uploads/FileUpload/0b/6cdede3bf9ebc50fc4dbf63b67af1a.png" width="450" alt="image" style="float: right;" class="embedImage-img importedEmbed-img"></img></a> As we decrease the <a rel="nofollow" href="https://isogg.org/wiki/Autosomal_DNA_statistics">degrees of relatedness between family members</a>, the expected average percent of identical DNA segments drops precipitously, by halves. My sister Sue and I expect ~50% identical DNA (half-identical on 50% and fully identical on 25%). However, my 23andMe cousin Susan, who lives in Arkansas and who connected with me via 23andMe in 2013, only shares 0.11% of her DNA with me, stemming from a single half identical segment on chromosome 5 <sup>2</sup>. Beagle and algorithms like it enable us to identify and score such segments of potential shared haplotypes given unphased genotype data and a <em>phased reference population resource</em>. The algorithms impute phases using this resource.</p>

<p>What these algorithms do not capture is how Susan, Sue and I will share many alleles at sites of common variation across our genomes due to shared ethnicity. I want to clarify that despite the similarity of our names (my name is Soo Hee, pronounced <em>So-hee</em>), we are each real and distinct individuals.</p>

<p><a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/FileUpload/e9/6b9d53c9c83ecc5f45f2e2f5a5e112.png"></a><a name="top" id="top"></a> </p><div style="text-align: center;"><img src="https://us.v-cdn.net/5019796/uploads/FileUpload/e9/6b9d53c9c83ecc5f45f2e2f5a5e112.png" width="450" alt="image" class="embedImage-img importedEmbed-img"></img></div>

<p>The 23andMe data that I will use in this series of posts is from my sister Sue and myself. We are clearly related not just by the happenstance of our births but by 54.7% and 44 segments according to 23andMe. Given this, the results of IBD analysis will be unsurprising and serves as a control in how well we can impute phasing information using a <em>public</em> population resource file that, to be clear, does not represent our Korean ethnic background. This resource is the <a rel="nofollow" href="http://www.internationalgenome.org/home"><em>1000 Genomes Project</em></a> phase 3 genotypes (<a rel="nofollow" href="http://www.nature.com/nature/journal/v526/n7571/full/nature15393.html">doi:10.1038/nature15393</a>) that I will refer to as <em>1KGP</em>. Of the resource’s 2504 individuals, who are all forward-thinking and generous <em>volunteers</em>, approximately 500 are East Asian. This supergroup includes ~300 Chinese, ~100 Vietnamese and ~100 Japanese, but no Koreans. Phasing confidence improves when the reference panel is ethnically matched to the individual. So ideally, we would want to match Sue and my data against a panel that includes Koreans. I should mention that the <em>1000 Genomes Project</em> is continuing to fill gaps in population representation via the <em>International Genome Sample Resource</em> (<a rel="nofollow" href="http://www.internationalgenome.org/about">IGSR</a>).</p>

<p>Our expectation is that Beagle’s <em>Refined IBD</em> approach (<a rel="nofollow" href="http://www.genetics.org/content/194/2/459">doi:10.1534/genetics.113.150029</a>) using the 1KGP resource will be robust to our genetic heterogeneity. I’m told there are better tools out there, e.g. <a rel="nofollow" href="http://faculty.washington.edu/browning/ibdseq.html">IBDSeq</a>, but that these require a reference panel from the same population as the analyzed samples. I did ask around if such phased data representing my peoples should exist but word is that none will be as carefully phased and <em>reliable</em> as the 1KGP resource.</p>

<p>To illustrate my last statement, let me digress a bit. Sue is getting married tomorrow, on April 8th, to Dave (<strong>pictured</strong>). So first I’d like to congratulate them. Sue and Dave’s ethnic origins are far apart--9,241 kilometers to be exact or a ~10 hour flight. Besides lending support to <a rel="nofollow" href="http://www.nature.com/articles/srep32550">HLA-mediated attraction</a>, their marriage has me asking this: what reference resource should we use to impute phasing for samples from Sue and Dave’s future children who would be <a rel="nofollow" href="https://en.wikipedia.org/wiki/Genetic_admixture">admixed</a> <a rel="nofollow" href="http://www.npr.org/sections/codeswitch/2014/12/15/370416571/half-asian-half-white-no-hapa">hapas</a>?</p>

<p><a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/FileUpload/4a/fe578e8c66ee4f08dc5958faecab11.jpg"></a><a name="top"></a> </p><div style="text-align: center;"><img src="https://us.v-cdn.net/5019796/uploads/FileUpload/4a/fe578e8c66ee4f08dc5958faecab11.jpg" width="320" alt="image" class="embedImage-img importedEmbed-img"></img></div>

<p>Another reason to take this journey is to flex our file processing muscles. To start, posts will focus on specific pre-processing aims, and I use a variety of tools, both GATK and non-GATK, to achieve these. My selection criteria is that of a <a rel="nofollow" href="https://www.wsj.com/articles/how-you-make-decisions-says-a-lot-about-how-happy-you-are-1412614997">satisficer</a>, i.e. I settle on the first tool or approach that achieves what I need. Given this, there may be more efficient ways to process the data, and hopefully those in the know who are reading will share these ways.</p>

<hr></hr><h3>I’m curious.</h3>

<p>I'm curious to find out how our field’s better IBD algorithm performs using a reference resource that underrepresents the ethnic background of the samples. I am a minority in this country whose parents immigrated when I was a baby. Sue was born while we lived in Fargo, North Dakota. Growing up in the Midwest and in Washington state, I have gotten used to the majority of faces not looking like me. What I have learned is that unlike the proportion my ethnicity represents in the population, to enable well-powered genomics studies whose results will be applicable to me, my ethnicity needs to be disproportionately <em>overrepresented</em> in programs like <em>All of Us</em>. I want my genetics represented because I want to benefit from the findings from genomics studies towards a longer and healthier life.</p>

<p>Gleaning equally useful information from studies as those in the ethnic majority means that those of us in the minority, including those of us who are African, Hispanic, mixed and Native Americans, need to <em>over</em> represent. I know there are other efforts, e.g. <a rel="nofollow" href="http://www.genomeasia100k.com/"><em>Genome Asia 100K</em></a> that plans for 100,000 Asian genomes. However, consider context. The <em>Genome Asia 100K</em> represents diversity within Asia, where populations are relatively homogenous. The resource, and findings from the resource, because of their population context, cannot apply widely to admixed populations such as that of America. As I explained earlier, context matters for IBD imputation in that, for Sue and my ethnically homogenous samples, the IBDSeq algorithm that uses a homogenous reference panel will out perform Beagle with the diverse reference panel.</p>

<p>Why is IBD analysis important to research? For one, validating pedigrees by measuring relatedness is an integral quality control step in population genomics studies involving extended families. Second, it can help genetic linkage mapping. For example, haplotype phase inference can help fill in missing genotypes. Third, it can help decrease bias. We see this application in the routine removal of closely related samples in population reference resources.</p>

<p>We can learn the importance of representing diversity from <a rel="nofollow" href="https://darwinsdogs.org/"><em>Darwin’s Dogs</em></a>. Last I heard, in December 2016, at the Broad company retreat, the project is only interested in sequencing <em>mutts</em>. You have to fill out a 100-question survey and they will let you know if they are interested in sequencing your dog’s genome. One thing they are interested in is associating behavior, e.g. OCD in dogs, to complex variation, e.g. multiple common variants. Sue and Dave have signed up Pepper (<strong>pictured above</strong>), who is of indeterminate mixed breed.</p>

<p>From where I’m standing, the contribution of genomics towards understanding disease appears so far to have come from (i) rare de novo mutations and (ii) isolated founder effects in extended but small families that control for genetic backdrop. <em>Darwin’s Dogs</em> explains a third approach in <a rel="nofollow" href="https://darwinsdogs.org/?pg=blog&amp;id=3">their blog</a>. They are enabled to take this third approach in their studies because of findings in a 2016 article titled <em>Complex disease and phenotype mapping in the domestic dog</em> (<a rel="nofollow" href="http://www.nature.com/articles/ncomms10460">doi:10.1038/ncomms10460</a>). The key enabling factor is knowing the number of dogs to power an analysis, e.g. either 400 dogs of a single breed (200 with and 200 without the trait) or 1000 dogs of many breeds (500 with and 500 without). I would surmise that the latter scenario would yield qualitatively different results that would be more applicable to dogs in general.</p>

<hr></hr><p>[1]  The average frequency of recombination events across a chromosome is captured in a metric geneticists call the centiMorgan.<br>
[2]  The algorithm 23andMe uses has a 7 cM (<a rel="nofollow" href="https://isogg.org/wiki/CentiMorgan">centiMorgan</a>) and 700 SNPs <a rel="nofollow" href="https://customercare.23andme.com/hc/en-us/articles/202907200-DNA-Relatives-How-it-works#threshold">threshold for relatedness</a> to ensure high quality matches, so it is highly likely we share common recent ancestors.</p>
