## Goodbye note to the GATK community

By shlee

<p><a rel="nofollow" href="https://us.v-cdn.net/5019796/uploads/editor/9b/27ayny8h6kl8.png"><img src="https://us.v-cdn.net/5019796/uploads/editor/9b/27ayny8h6kl8.png" width="360" alt="image" style="float: right;" class="embedImage-img importedEmbed-img"></img></a></p>

<p>The other day I decided to disassemble the bathroom doorknob. Efforts included chipping away layers of paint and recruiting some muscle to remove the screws the chipping had revealed. When I levered out the latch system and took it apart, I noticed two things. First, parts of it had a beautiful copper color. Second, the internal spring was broken into three parts. The latter explained the sticky latch you had to jiggle to stay closed and a door that had started to pop open randomly as if possessed.</p>

<p>I made the fix with a new spring.</p>

<p>It may be a compulsion to want to fix broken things. I think it stems from the same curiosity that makes you want to take things apart to understand how they work [<a rel="nofollow" href="#ref1">1</a>]. When I joined the GATK some 3.5 years ago as a technical writer, this compulsion surfaced and drove me to the effort that resulted in the pieces I wrote. Below, at the end of this post, is <a rel="nofollow" href="#table">a sampling of my most viewed articles</a>.</p>

<p>I would like to thank you for allowing me to serve you these past years. I have learned much in the process. The knowledge I have gained in genomics comes not only from these writing projects but also just as much from answering your questions on the forum. From holding to answering just one forum question a day, I am proud to have earned over <a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/badge/like-250">250 likes</a> and points aplenty for the forum’s <a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/profile/shlee">five-star ranking</a>.</p>

<p>My last day with the DSP Communications Team is April 1, which is today [<a rel="nofollow" href="#ref2">2</a>]. Rest assured, <a rel="nofollow" href="https://software.broadinstitute.org/gatk/blog?id=23384">my teammates</a> and our wonderful <a rel="nofollow" href="https://github.com/broadinstitute/gatk/graphs/contributors">methods developers</a> will continue to take excellent care of you.</p>

<p>Looking back, 2018 was a busy year. Geraldine asked I help out at the July Cambridge UK workshop and also at the December Taiwan workshop [<a rel="nofollow" href="#ref3">3</a>]. Each workshop brings with it a torrent of activity creating and updating materials. It is always insightful and rewarding to interact firsthand with researchers, to hear about sticking points and to see reactions to the tutorials I develop and write.</p>

<p>Since returning from the December workshop, I have been submarined pouring effort into finalizing the <a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11684">gCNV tutorial</a> in time for my departure. I hope you find it useful. This tutorial has been the most challenging to develop so far in that exploring the results involved more creative solutions than usual, as you will see in the tutorial’s companion Jupyter Notebook reports <a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11685/notebook-concordance-of-na19017-chr20-gcnv-calls/p1?new=1">here</a> and  <a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11686/notebook-correlate-gcnv-callset-metrics-and-annotations/p1?new=1">here</a> [<a rel="nofollow" href="#ref4">4</a>].</p>

<p>Before I start searching for a new job, this month I will spend some time visiting friends and family and remembering <a rel="nofollow" href="https://www.hopkinsmedicine.org/news/newsroom/news-releases/paul-englund-renowned-biochemist-who-studied-parasites-dies">my Ph.D. advisor</a> at his memorial. If you would like to lend your support, I would love to have your endorsement on <a rel="nofollow" href="https://www.linkedin.com/in/sooheelee">LinkedIn</a> [<a rel="nofollow" href="#ref5">5</a>]. If you need to get in touch with me, please ping me on GitHub, in the <a rel="nofollow" href="https://github.com/broadinstitute/gatk">broadinstitute/gatk</a> repository. My handle is <a rel="nofollow" href="https://github.com/sooheelee">@sooheelee</a> and I will be checking in intermittently.</p>

<p>It has been a privilege.</p>

<p>Yours truly,</p>

<p>Soo Hee</p>

<hr></hr><h3>Footnotes</h3>

<p><a name="ref1" id="ref1"></a> <br>
[1] This curiosity should not be surprising in someone who once walked the life of a Ph.D. biochemist. And it should be expected from someone whose folks include a plant pathologist (Dad studied in North Dakota) and a WWII pilot turned aeronautical engineer (Mr. Cummings served in the Army Air Corp; he is turning 95 this May and I will be seeing him for his birthday). Each of my families tells me I’m molded from the same clay as my fathers.<br><a name="ref2" id="ref2"></a> <br>
[2] No, this is not an April Fools' joke.<br><a name="ref3" id="ref3"></a> <br>
[3] There were two Taiwan workshops in 2018. The video footage of the December 2018 Taiwan workshop is not posted anywhere else, and so here is the link: <a rel="nofollow" href="https://drive.google.com/drive/folders/1-uMoz-ui5IteriKngee7Vic9AWAcnfcL">https://drive.google.com/drive/folders/1-uMoz-ui5IteriKngee7Vic9AWAcnfcL</a>. <br><a name="ref4" id="ref4"></a> <br>
[4] I have become a fan of pandas <a rel="nofollow" href="https://en.wikipedia.org/wiki/Pandas_(software)">the software</a> but also <a rel="nofollow" href="https://www.nationalgeographic.com/animals/2018/08/giant-baby-panda-cubs-china-conservation/#/panda-hidden-forest.jpg">the animal</a>. <br><a name="ref5" id="ref5"></a> <br>
[5] <a rel="nofollow" href="https://www.linkedin.com/in/sooheelee">Connect with me</a>, and, if you feel like it, please endorse my skills in <em>Genomics</em>.</p>

<hr></hr><p><a name="table" id="table"></a></p>

<h3>A sampling of my most popular articles grouped by year and sorted by number of views</h3>

<table><thead><tr><th><strong>Year</strong></th>
  <th align="center"><strong>Views</strong></th>
  <th align="center"><strong>Article# and link</strong></th>
  <th><strong>Title</strong></th>
</tr></thead><tbody><tr><td>2015</td>
  <td align="center">26.1K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/6484">6484</a></td>
  <td>(How to) Generate an unmapped BAM from FASTQ or aligned BAM</td>
</tr><tr><td>.</td>
  <td align="center">17.2K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/6483">6483</a></td>
  <td>(How to) Map and clean up short read sequence data efficiently</td>
</tr><tr><td>2016</td>
  <td align="center">17.9K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/6747">6747</a></td>
  <td>(How to) Mark duplicates with MarkDuplicates or MarkDuplicatesWithMateCigar</td>
</tr><tr><td>.</td>
  <td align="center">8.2K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/7857">7857</a></td>
  <td>Reference Genome Components</td>
</tr><tr><td>.</td>
  <td align="center">7.8K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/8017">8017</a></td>
  <td>(How to) Map reads to a reference with alternate contigs like GRCh38</td>
</tr><tr><td>.</td>
  <td align="center">4.9K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/7847">7847</a></td>
  <td>Changing workflows around calling SNPs and indels</td>
</tr><tr><td>.</td>
  <td align="center">4.4K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/7156">7156</a></td>
  <td>(howto) Perform local realignment around indels</td>
</tr><tr><td>.</td>
  <td align="center">3.5K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/7899">7899</a></td>
  <td>Reference implementation: PairedEndSingleSampleWf pipeline</td>
</tr><tr><td>.</td>
  <td align="center">2.2K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/6926">6926</a></td>
  <td>Spanning or overlapping deletions (* allele)</td>
</tr><tr><td>.</td>
  <td align="center">2.0K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/8180">8180</a></td>
  <td>9 Takeaways to help you get started with GRCh38</td>
</tr><tr><td>.</td>
  <td align="center">1.9K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/7859">7859</a></td>
  <td>(How to) Simulate reads using a reference genome ALT contig</td>
</tr><tr><td>.</td>
  <td align="center">1.1K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/7019">7019</a></td>
  <td>Sam flags down a boat</td>
</tr><tr><td>2017</td>
  <td align="center">22.7K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/9143">9143</a>*</td>
  <td>(How to) Call somatic copy number variants using GATK4 CNV</td>
</tr><tr><td>.</td>
  <td align="center">2.9K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/9183">9183</a>*</td>
  <td>(How to) Call somatic SNVs and indels using MuTect2</td>
</tr><tr><td>.</td>
  <td align="center">2.2K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/10172">10172</a></td>
  <td>(How to) Run the GATK4 Docker locally and take a look inside</td>
</tr><tr><td>.</td>
  <td align="center">1.7K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/10911">10911</a></td>
  <td>Differences between GATK3 MuTect2 and GATK4 Mutect2</td>
</tr><tr><td>.</td>
  <td align="center">1.1K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/10060">10060</a></td>
  <td>(How to) Run FlagStatSpark on a cloud Spark cluster</td>
</tr><tr><td>2018</td>
  <td align="center">18.5K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11136">11136</a></td>
  <td>(How to) Call somatic mutations using GATK4 Mutect2</td>
</tr><tr><td>.</td>
  <td align="center">3.0K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11682">11682</a></td>
  <td>(How to part I) Sensitively detect copy ratio alterations and allelic segments</td>
</tr><tr><td>.</td>
  <td align="center">2.6K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11127">11127</a></td>
  <td>Somatic calling is NOT simply a difference between two callsets</td>
</tr><tr><td>.</td>
  <td align="center">2.0K</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11683">11683</a></td>
  <td>(How to part II) Sensitively detect copy ratio alterations and allelic segments</td>
</tr><tr><td>.</td>
  <td align="center">938</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/12350">12350</a></td>
  <td>(How to) Filter on genotype using VariantFiltration</td>
</tr><tr><td>.</td>
  <td align="center">740</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11315">11315</a></td>
  <td>Off-label workflow to simply call differences in two samples</td>
</tr><tr><td>.</td>
  <td align="center">~</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/23216">23216</a></td>
  <td>(How to) Filter variants either with VQSR or by hard-filtering</td>
</tr><tr><td>2019</td>
  <td align="center">~</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11684">11684</a></td>
  <td>(How to) Call common and rare germline copy number variants</td>
</tr><tr><td>.</td>
  <td align="center">~</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11685">11685</a></td>
  <td>(Notebook) Concordance of NA19017 chr20 gCNV calls</td>
</tr><tr><td>.</td>
  <td align="center">~</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11686">11686</a></td>
  <td>(Notebook) Correlate gCNV callset metrics and annotations</td>
</tr><tr><td>.</td>
  <td align="center">~</td>
  <td align="center"><a rel="nofollow" href="https://gatkforums.broadinstitute.org/gatk/discussion/11687">11687</a></td>
  <td>After gCNV calling considerations</td>
</tr></tbody></table><p>*Uses older versions of tools that have been replaced.<br>
~Published in the last three months.</p>

<hr></hr>